<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ittai Eres" />

<meta name="date" content="2018-03-13" />

<title>Functional Enrichments of Hi-C Bins</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Hi-C</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Functional Enrichments of Hi-C Bins</h1>
<h4 class="author"><em>Ittai Eres</em></h4>
<h4 class="date"><em>2018-03-13</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-05-10</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> e81cf9e</p>
<!-- Add your analysis here -->
<div id="reading-in-the-data-reformatting-for-enrichment-analyses" class="section level3">
<h3>Reading in the data, reformatting for enrichment analyses</h3>
<p>Here I merely read in the filtered data and create a function to reformat it for a variety of overlap analyses external to R.</p>
<pre class="r"><code>#Now, read in filtered data from linear_modeling_QC.Rmd.
data.filtered &lt;- fread(&quot;~/Desktop/Hi-C/data.4.filtered.txt&quot;, header=TRUE, data.table=FALSE, stringsAsFactors = FALSE, showProgress=FALSE)
#meta.data &lt;- data.frame(&quot;SP&quot;=c(&quot;H&quot;, &quot;H&quot;, &quot;C&quot;, &quot;C&quot;, &quot;H&quot;, &quot;H&quot;, &quot;C&quot;, &quot;C&quot;), &quot;SX&quot;=c(&quot;F&quot;, &quot;M&quot;, &quot;M&quot;, &quot;F&quot;, &quot;M&quot;, &quot;F&quot;, &quot;M&quot;, &quot;F&quot;), &quot;Batch&quot;=c(1, 1, 1, 1, 2, 2, 2, 2))

#Now, prepare a bedPE file from my filtered data! Here I construct a function to do this in case I ever want to do it again. The H1 and H2 columns are good to use for this.
bedpe.prep &lt;- function(homersigdf, res){ #Also is the BEDPE format!
  H1 &lt;- homersigdf$H1
  H2 &lt;- homersigdf$H2
  H1_coords &lt;- as.numeric(gsub(&quot;.*-&quot;, &quot;&quot;, H1))
  H2_coords &lt;- as.numeric(gsub(&quot;.*-&quot;, &quot;&quot;, H2))
  H1_chrs &lt;- gsub(&quot;-.*&quot;, &quot;&quot;, H1)
  H2_chrs &lt;- gsub(&quot;-.*&quot;, &quot;&quot;, H2)
  H1_first &lt;- ifelse(H1_coords&lt;H2_coords, H1_coords, H2_coords)
  H2_second &lt;- ifelse(H1_coords&lt;H2_coords, H2_coords, H1_coords)
  result &lt;- data.frame(chr.1 = H1_chrs, start.1 = H1_first, end.1 = H1_first+res, chr.2 = H2_chrs, start.2 = H2_second, end.2 = H2_second+res)
  return(result)
}
test &lt;- bedpe.prep(data.filtered, 10000)
test &lt;- format(test, scientific=FALSE)
fwrite(test, &quot;~/Desktop/pgltoolswork/new_updated/filt.bins.bedPE&quot;, quote = FALSE, sep = &quot;\t&quot;, col.names = TRUE, row.names=FALSE)</code></pre>
</div>
<div id="functional-enrichments-read-in-of-files-and-creation-of-comprehensive-dataframe." class="section level3">
<h3>Functional enrichments: read-in of files and creation of comprehensive dataframe.</h3>
<p>Looking at functional categories enriched in my specific bins. Here I pull out several histone mark peak files and chromHMM annotations, overlapping them with my significant Hi-C bins and adding in information from my linear modeling on the contacts as well as RNA-seq differential expression analyses.</p>
<pre class="r"><code>#####First, obtain data frames from the gene_expression.Rmd file, so I can use the promoter information we have from our in-house metaorthoexon trios file.
gene.hic &lt;- readRDS(&quot;~/Desktop/Hi-C/gene.hic.filt.RDS&quot;)

###################ChromHMM Enrichments: I obtain the chromHMM bin annotations.
HMM.info.1 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/chromHMM_bin1.bed&quot;, sep=&quot;\t&quot;, header=FALSE)
HMM.info.2 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/chromHMM_bin2.bed&quot;, sep=&quot;\t&quot;, header=FALSE)

#Fix things up, obtain single identifications for each bin based on highest abundance of annotations.
HMM.info.1$HID &lt;- paste(HMM.info.1$V1, HMM.info.1$V2, sep=&quot;-&quot;)
HMM.info.2$HID &lt;- paste(HMM.info.2$V1, HMM.info.2$V2, sep=&quot;-&quot;)
group_by(HMM.info.1, HID, V7) %&gt;% summarise(overlap=sum(V13)) %&gt;% group_by(HID) %&gt;% summarise(chromstate=V7[which.max(overlap)]) -&gt; bin1.HMM.anno #Summarizes each bin as a single chromHMM annotation. This does it by which element covers majority of the bin, rather than just frequency of elements as this would work: group_by(HMM.info.1, HID) %&gt;% summarise(chromstate=names(sort(-table(V7)))[1]).
group_by(HMM.info.2, HID, V7) %&gt;% summarise(overlap=sum(V13)) %&gt;% group_by(HID) %&gt;% summarise(chromstate=V7[which.max(overlap)]) -&gt; bin2.HMM.anno
colnames(bin1.HMM.anno) &lt;- c(&quot;H1&quot;, &quot;bin1.HMM&quot;)
colnames(bin2.HMM.anno) &lt;- c(&quot;H2&quot;, &quot;bin2.HMM&quot;)

group_by(HMM.info.1, HID, V7) %&gt;% summarise(overlap=sum(V13)) -&gt; bin1.HMM.overlaps #Gets me specific annotations&#39; summed overlaps on each bin. Now to convert to wide-form df:
bin1.HMMs &lt;- spread(bin1.HMM.overlaps, V7, overlap) #2nd argument is new column names (annotations), 3rd argument is values (length of overlap)
group_by(HMM.info.2, HID, V7) %&gt;% summarise(overlap=sum(V13)) -&gt; bin2.HMM.overlaps #Same situation for second set of bins.
bin2.HMMs &lt;- spread(bin2.HMM.overlaps, V7, overlap)
colnames(bin1.HMMs) &lt;- c(&quot;H1&quot;, &quot;Nothing&quot;, &quot;TssA&quot;, &quot;TssBiv&quot;, &quot;BivFlnk&quot;, &quot;EnhBiv&quot;, &quot;ReprPC&quot;, &quot;ReprPCWk&quot;, &quot;Quies&quot;, &quot;TssAFlnk&quot;, &quot;TxFlnk&quot;, &quot;Tx&quot;, &quot;TxWk&quot;, &quot;EnhG&quot;, &quot;Enh&quot;, &quot;ZNF/Rpts&quot;, &quot;Het&quot;)
colnames(bin2.HMMs) &lt;- c(&quot;H2&quot;, &quot;Nothing&quot;, &quot;TssA&quot;, &quot;TssBiv&quot;, &quot;BivFlnk&quot;, &quot;EnhBiv&quot;, &quot;ReprPC&quot;, &quot;ReprPCWk&quot;, &quot;Quies&quot;, &quot;TssAFlnk&quot;, &quot;TxFlnk&quot;, &quot;Tx&quot;, &quot;TxWk&quot;, &quot;EnhG&quot;, &quot;Enh&quot;, &quot;ZNF/Rpts&quot;, &quot;Het&quot;)

#Add in linear modeling information from the data.filtered df of my homer hits. Combine data frames here for analysis of the overall proportions by assigning each bin to a given chromHMM annotation; as well as for quantitatively assessing differences in the overlaps of these elements at particular FDRs.
chromenrich &lt;- select(data.filtered, H1, H2, sp_BH_pval, sp_beta)
chromrich &lt;- left_join(chromenrich, bin1.HMM.anno, by=&quot;H1&quot;)
chromrich &lt;- left_join(chromrich, bin2.HMM.anno, by=&quot;H2&quot;)
chromrich &lt;- chromrich[order(chromrich$sp_BH_pval),] #Order by FDR. Provides a df of the Hi-C bin hits and their chromHMM annotations.
HMM.quant &lt;- left_join(chromenrich, bin1.HMMs, by=&quot;H1&quot;)
HMM.quant &lt;- left_join(HMM.quant, bin2.HMMs, by=&quot;H2&quot;, suffix=c(&quot;.1&quot;, &quot;.2&quot;))
chromrich$pairID &lt;- paste(chromrich$H1, chromrich$H2, sep=&quot;_&quot;)
HMM.quant$pairID &lt;- paste(HMM.quant$H1, HMM.quant$H2, sep=&quot;_&quot;)
chromrich &lt;- left_join(chromrich, HMM.quant[,c(-1:-4)], by=&quot;pairID&quot;)

####Now, add in other chromatin mark peak files and info about their overlap with Hi-C bins:
#####################CTCF Enrichments. Also works for other quantitative enrichments that just look at overlapping bases.
CTCF.info &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/CTCF_not_ispan.bed&quot;, sep=&quot;\t&quot;, header=FALSE)
CTCF.info$pairID &lt;- paste(paste(CTCF.info$V1, CTCF.info$V2, sep=&quot;-&quot;), paste(CTCF.info$V4, CTCF.info$V5, sep=&quot;-&quot;), sep=&quot;_&quot;)

######Start of other quantitative enrichments.
DHS.bin1 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/DHS_bin1.bed&quot;, sep=&quot;\t&quot;, header=FALSE)
DHS.bin2 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/DHS_bin2.bed&quot;, sep=&quot;\t&quot;, header=FALSE)
Hac.bin1 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/H3K27ac_bin1.bed&quot;, sep=&quot;\t&quot;, header=FALSE)
Hac.bin2 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/H3K27ac_bin2.bed&quot;, sep=&quot;\t&quot;, header=FALSE)
Hme1.bin1 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/H3K4me1_bin1.bed&quot;, sep=&quot;\t&quot;, header=FALSE)
Hme1.bin2 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/H3K4me1_bin2.bed&quot;, sep=&quot;\t&quot;, header=FALSE)
Hme3.bin1 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/H3K4me3_bin1.bed&quot;, sep=&quot;\t&quot;, header=FALSE)
Hme3.bin2 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/H3K4me3_bin2.bed&quot;, sep=&quot;\t&quot;, header=FALSE)
HAR.bin1 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/HAR_bin1.bed&quot;, sep=&quot;\t&quot;, header=FALSE)
HAR.bin2 &lt;- read.csv(&quot;~/Desktop/pgltoolswork/new_updated/intersects/HAR_bin2.bed&quot;, sep=&quot;\t&quot;, header=FALSE)

DHS.bin1$HID &lt;- paste(DHS.bin1$V1, DHS.bin1$V2, sep=&quot;-&quot;)
DHS.bin2$HID &lt;- paste(DHS.bin2$V1, DHS.bin2$V2, sep=&quot;-&quot;)
group_by(DHS.bin1, HID) %&gt;% summarise(bin1.DHSval=sum(V14)) -&gt; bin1.DHS
group_by(DHS.bin2, HID) %&gt;% summarise(bin2.DHSval=sum(V14)) -&gt; bin2.DHS

Hac.bin1$HID &lt;- paste(Hac.bin1$V1, Hac.bin1$V2, sep=&quot;-&quot;)
Hac.bin2$HID &lt;- paste(Hac.bin2$V1, Hac.bin2$V2, sep=&quot;-&quot;)
group_by(Hac.bin1, HID) %&gt;% summarise(bin1.Hacval=sum(V14)) -&gt; bin1.Hac
group_by(Hac.bin2, HID) %&gt;% summarise(bin2.Hacval=sum(V14)) -&gt; bin2.Hac

Hme1.bin1$HID &lt;- paste(Hme1.bin1$V1, Hme1.bin1$V2, sep=&quot;-&quot;)
Hme1.bin2$HID &lt;- paste(Hme1.bin2$V1, Hme1.bin2$V2, sep=&quot;-&quot;)
group_by(Hme1.bin1, HID) %&gt;% summarise(bin1.Hme1val=sum(V14)) -&gt; bin1.Hme1
group_by(Hme1.bin2, HID) %&gt;% summarise(bin2.Hme1val=sum(V14)) -&gt; bin2.Hme1

Hme3.bin1$HID &lt;- paste(Hme3.bin1$V1, Hme3.bin1$V2, sep=&quot;-&quot;)
Hme3.bin2$HID &lt;- paste(Hme3.bin2$V1, Hme3.bin2$V2, sep=&quot;-&quot;)
group_by(Hme3.bin1, HID) %&gt;% summarise(bin1.Hme3val=sum(V14)) -&gt; bin1.Hme3
group_by(Hme3.bin2, HID) %&gt;% summarise(bin2.Hme3val=sum(V14)) -&gt; bin2.Hme3

HAR.bin1$HID &lt;- paste(HAR.bin1$V1, HAR.bin1$V2, sep=&quot;-&quot;)
HAR.bin2$HID &lt;- paste(HAR.bin2$V1, HAR.bin2$V2, sep=&quot;-&quot;)
group_by(HAR.bin1, HID) %&gt;% summarise(bin1.HARval=sum(V7)) -&gt; bin1.HAR
group_by(HAR.bin2, HID) %&gt;% summarise(bin2.HARval=sum(V7)) -&gt; bin2.HAR

colnames(bin1.DHS)[1] &lt;- colnames(bin1.Hac)[1] &lt;- colnames(bin1.Hme1)[1] &lt;- colnames(bin1.Hme3)[1] &lt;- colnames(bin1.HAR)[1] &lt;- &quot;H1&quot;
colnames(bin2.DHS)[1] &lt;- colnames(bin2.Hac)[1] &lt;- colnames(bin2.Hme1)[1] &lt;- colnames(bin2.Hme3)[1] &lt;- colnames(bin2.HAR)[1] &lt;- &quot;H2&quot;

#Sequential left joins to make huge table on paired data!
left_join(chromrich, bin1.DHS, by = &quot;H1&quot;) %&gt;% left_join(., bin2.DHS, by=&quot;H2&quot;) %&gt;% left_join(., bin1.Hac, by=&quot;H1&quot;) %&gt;% left_join(., bin2.Hac, by=&quot;H2&quot;) %&gt;% left_join(., bin1.Hme1, by=&quot;H1&quot;) %&gt;% left_join(., bin2.Hme1, by=&quot;H2&quot;) %&gt;% left_join(., bin1.Hme3, by=&quot;H1&quot;) %&gt;% left_join(., bin2.Hme3, by=&quot;H2&quot;) %&gt;% left_join(., bin1.HAR, by=&quot;H1&quot;) %&gt;% left_join(., bin2.HAR, by=&quot;H2&quot;) -&gt; full.annotations

full.annotations$CTCF.interrupt &lt;- ifelse(full.annotations$pairID %in% CTCF.info$pairID, &quot;no&quot;, &quot;yes&quot;)

#Add a column to the df indicating which interactions include a promoter and which don&#39;t. Here I am defining a promoter based off the meta ortho exon trios file (true.promoter). I used to secondarily look at chromHMM promoters (promoter) but there&#39;s not much to be gleaned there.
promoter.IDs &lt;- gene.hic$HID
full.annotations$true.promoter &lt;- ifelse((full.annotations$H1 %in% promoter.IDs)|(full.annotations$H2 %in% promoter.IDs), &quot;yes&quot;, &quot;no&quot;)
full.annotations$promoter.bin1 &lt;- ifelse(full.annotations$H1 %in% promoter.IDs, &quot;yes&quot;, &quot;no&quot;)
full.annotations$promoter.bin2 &lt;- ifelse(full.annotations$H2 %in% promoter.IDs, &quot;yes&quot;, &quot;no&quot;)
#full.annotations$HMM.promoter &lt;- ifelse(chromrich$bin1.HMM==&quot;1_TssA&quot;|chromrich$bin1.HMM==&quot;2_TssAFlnk&quot;|chromrich$bin1.HMM==&quot;3_TxFlnk&quot;|chromrich$bin2.HMM==&quot;1_TssA&quot;|chromrich$bin2.HMM==&quot;2_TssAFlnk&quot;|chromrich$bin2.HMM==&quot;3_TxFlnk&quot;, &quot;yes&quot;, &quot;no&quot;)

#Just out of interest, see how many of the genes that are/not DE had promoters picked up by our Hi-C:
nonprom.IDs &lt;- filter(gene.hic, adj.P.Val&gt;0.05) %&gt;% select(., HID)
prom.IDs &lt;- filter(gene.hic, adj.P.Val&lt;=0.05) %&gt;% select(., HID)

nrow(prom.IDs)</code></pre>
<pre><code>[1] 1627</code></pre>
<pre class="r"><code>sum((prom.IDs$HID %in% full.annotations$H1)|(prom.IDs$HID %in% full.annotations$H2))/nrow(prom.IDs)</code></pre>
<pre><code>[1] 0.9452981</code></pre>
<pre class="r"><code>nrow(nonprom.IDs)</code></pre>
<pre><code>[1] 6547</code></pre>
<pre class="r"><code>sum((nonprom.IDs$HID %in% full.annotations$H1)|(nonprom.IDs$HID %in% full.annotations$H2))/nrow(nonprom.IDs)</code></pre>
<pre><code>[1] 0.9517336</code></pre>
<pre class="r"><code>#Not interesting.

#Grab info from gene.hic to add some information to this table about DE and DC status:
DEbins &lt;- gene.hic$HID[which(gene.hic$adj.P.Val&lt;=0.05)]
full.annotations$DE &lt;- ifelse((full.annotations$H1 %in% DEbins)|(full.annotations$H2 %in% DEbins), &quot;yes&quot;, &quot;no&quot;)
full.annotations$DC &lt;- ifelse(full.annotations$sp_BH_pval&lt;=0.05, &quot;yes&quot;, &quot;no&quot;)
full.annotations$differential &lt;- ifelse(full.annotations$DC==&quot;no&quot;&amp;full.annotations$DE==&quot;no&quot;, &quot;no&quot;, ifelse(full.annotations$DC==&quot;yes&quot;&amp;full.annotations$DE==&quot;yes&quot;, &quot;both&quot;, ifelse(full.annotations$DC==&quot;yes&quot;, &quot;contact&quot;, &quot;expression&quot;)))
#I&#39;ll use this downstream in this file to look at enrichments of specific sets of histone marks in my data. This is now ordered by FDR for DC as well.

#A function to deduplicate the hits, removing bins that are repeated at higher FDR, which can be used on full.annotations after a variety of filters are applied.
deduplicator &lt;- function(df){
  duplicators &lt;- data.frame(matrix(ncol=2, nrow=nrow(df)*2))
  duplicators[seq(1, nrow(df)*2, 2), 1] &lt;- df$H1
  duplicators[seq(2, nrow(df)*2, 2), 1] &lt;- df$H2
  duplicators$X2 &lt;- rep(1:nrow(df), 1, each=2)
  repeats &lt;- which(duplicated(duplicators$X1))
  bin1.dups &lt;- repeats[which(repeats%%2==1)] #Pull out bin 1 duplicates, based on odd row identity.
  bin2.dups &lt;- repeats[which(repeats%%2==0)] #Pull out bin 2 duplicates, based on even row identity.
  dup1.indices &lt;- duplicators[bin1.dups, 2] #Pull out the original row indices for bin1 dups.
  dup2.indices &lt;- duplicators[bin2.dups, 2] #Pull out the original row indices for bin2 dups.
  df[dup1.indices, c(5, 8:23, 40, 42, 44, 46, 48)] &lt;- NA #Set duplicated bin1 values to NA (aside from bin identity, already checked this function works as desired).
  df[dup2.indices, c(6, 24:39, 41, 43, 45, 47, 49)] &lt;- NA #Set duplicated bin2 values to NA (aside from bin identity).
  return(df) #Return the cleaned df!
}

###Now, prepare files of annotations, first filtering for given questions about promoters, and then filtering out duplicates in those data frames. Save those out afterwards as a variety of different files to calculate chromHMM annotations proportions on them on midway2.
#Promoter annotations, DE/DC or not.
full.proms &lt;- filter(full.annotations, true.promoter==&quot;yes&quot;) %&gt;% deduplicator(.)
proms.DE &lt;- filter(full.annotations, true.promoter==&quot;yes&quot;, DE==&quot;yes&quot;) %&gt;% deduplicator(.)
proms.not.DE &lt;- filter(full.annotations, true.promoter==&quot;yes&quot;, DE==&quot;no&quot;) %&gt;% deduplicator(.)
proms.DC &lt;- filter(full.annotations, true.promoter==&quot;yes&quot;, DC==&quot;yes&quot;) %&gt;% deduplicator(.)
proms.not.DC &lt;- filter(full.annotations, true.promoter==&quot;yes&quot;, DC==&quot;no&quot;) %&gt;% deduplicator(.)

#Also do exclusive sets for differential status on the promoter bins:
proms.DE.only &lt;- filter(full.annotations, true.promoter==&quot;yes&quot;, differential==&quot;expression&quot;) %&gt;% deduplicator(.)
proms.DC.only &lt;- filter(full.annotations, true.promoter==&quot;yes&quot;, differential==&quot;contact&quot;) %&gt;% deduplicator(.)
proms.both.only &lt;- filter(full.annotations, true.promoter==&quot;yes&quot;, differential==&quot;both&quot;) %&gt;% deduplicator(.)
proms.neither &lt;- filter(full.annotations, true.promoter==&quot;yes&quot;, differential==&quot;no&quot;) %&gt;% deduplicator(.)

#Non promoter annotations, DE/DC or not. Note that there are no DE non-promoters, DUH, meaning there are none that fall into the &quot;both&quot; differential category. Also means full set of non-promoters qualifies as not DE.
full.no.proms &lt;- filter(full.annotations, true.promoter==&quot;no&quot;) %&gt;% deduplicator(.)
no.proms.DC &lt;- filter(full.annotations, true.promoter==&quot;no&quot;, DC==&quot;yes&quot;) %&gt;% deduplicator(.)
no.proms.not.DC &lt;- filter(full.annotations, true.promoter==&quot;no&quot;, DC==&quot;no&quot;) %&gt;% deduplicator

#Now, write these files out.
saveRDS(full.proms, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/full.promoters.RDS&quot;)
saveRDS(proms.DE, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/promoters.DE.RDS&quot;)
saveRDS(proms.not.DE, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/promoters.not.DE.RDS&quot;)
saveRDS(proms.DC, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/promoters.DC.RDS&quot;)
saveRDS(proms.not.DC, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/promoters.not.DC.RDS&quot;)
saveRDS(proms.neither, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/promoters.neither.RDS&quot;)

saveRDS(full.no.proms, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/full.no.promoters.RDS&quot;)
saveRDS(no.proms.DC, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/no.promoters.DC.RDS&quot;)
saveRDS(no.proms.not.DC, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/no.promoters.not.DC.RDS&quot;)

saveRDS(proms.DE.only, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/proms.DE.only.RDS&quot;)
saveRDS(proms.DC.only, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/proms.DC.only.RDS&quot;)
saveRDS(proms.both.only, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/proms.both.only.RDS&quot;)
saveRDS(proms.neither, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/proms.neither.only.RDS&quot;)

#Last thing I am interested in also writing out and working with is a df that only takes the bins making contact with promoter bins (not the promoters themselves). Construct that here:
prom.1s &lt;- filter(full.annotations, promoter.bin1==&quot;yes&quot;) %&gt;% select(., H2, sp_BH_pval, sp_beta, bin2.HMM, pairID, Nothing.2, TssA.2, TssBiv.2, BivFlnk.2, EnhBiv.2, ReprPC.2, ReprPCWk.2, Quies.2, TssAFlnk.2, TxFlnk.2, Tx.2, TxWk.2, EnhG.2, Enh.2, &quot;ZNF/Rpts.2&quot;, Het.2, bin2.DHSval, bin2.Hacval, bin2.Hme1val, bin2.Hme3val, bin2.HARval, DE, DC, differential)
prom.2s &lt;- filter(full.annotations, promoter.bin2==&quot;yes&quot;) %&gt;% select(., H1, sp_BH_pval, sp_beta, bin1.HMM, pairID, Nothing.1, TssA.1, TssBiv.1, BivFlnk.1, EnhBiv.1, ReprPC.1, ReprPCWk.1, Quies.1, TssAFlnk.1, TxFlnk.1, Tx.1, TxWk.1, EnhG.1, Enh.1, &quot;ZNF/Rpts.1&quot;, Het.1, bin1.DHSval, bin1.Hacval, bin1.Hme1val, bin1.Hme3val, bin1.HARval, DE, DC, differential)

#Combine them into one df for running analyses on!
colnames(prom.2s) &lt;- colnames(prom.1s)
promoter.contacts &lt;- rbind(prom.1s, prom.2s)

#Prepare separate dfs of the different classes for running tabulations on. This initial df (promoter.contacts) is also the one I should use for running a number of downstream histone mark analyses and asking targeted questions about the bins making contacts with promoters...
promoter.contacts &lt;- promoter.contacts[order(promoter.contacts$sp_BH_pval),] #Order by FDR here, then wait to remove duplicates until differential filtering is done.

mates.proms.full &lt;- promoter.contacts[!duplicated(promoter.contacts$H2),] #Prep full promoter file, minus duplicate bins.
mates.proms.DE &lt;- filter(promoter.contacts, DE==&quot;yes&quot;)
mates.proms.DE &lt;- mates.proms.DE[!duplicated(mates.proms.DE$H2),] #DE file, minus duplicates
mates.proms.not.DE &lt;- filter(promoter.contacts, DE==&quot;no&quot;)
mates.proms.not.DE &lt;- mates.proms.not.DE[!duplicated(mates.proms.not.DE$H2),] #notDE file, minus duplicates
mates.proms.DC &lt;- filter(promoter.contacts, DC==&quot;yes&quot;)
mates.proms.DC &lt;- mates.proms.DC[!duplicated(mates.proms.DC$H2),] #DC file minus dups
mates.proms.not.DC &lt;- filter(promoter.contacts, DC==&quot;no&quot;)
mates.proms.not.DC &lt;- mates.proms.not.DC[!duplicated(mates.proms.not.DC$H2),] #notDC file, minus dups

#Classes with exclusive differential identifications.
mates.proms.DE.only &lt;- filter(promoter.contacts, differential==&quot;expression&quot;)
mates.proms.DE.only &lt;- mates.proms.DE.only[!duplicated(mates.proms.DE.only$H2),]
mates.proms.DC.only &lt;- filter(promoter.contacts, differential==&quot;contact&quot;)
mates.proms.DC.only &lt;- mates.proms.DC.only[!duplicated(mates.proms.DC.only$H2),]
mates.proms.DCDE.only &lt;- filter(promoter.contacts, differential==&quot;both&quot;)
mates.proms.DCDE.only &lt;- mates.proms.DCDE.only[!duplicated(mates.proms.DCDE.only$H2),]
mates.proms.neither.only &lt;- filter(promoter.contacts, differential==&quot;no&quot;)
mates.proms.neither.only &lt;- mates.proms.neither.only[!duplicated(mates.proms.neither.only$H2),]

#Now, write these files out! Note that chromHMM tabulator function will have to be adjusted for these.
saveRDS(mates.proms.full, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/mates.proms.full.RDS&quot;)
saveRDS(mates.proms.DE, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/mates.proms.DE.RDS&quot;)
saveRDS(mates.proms.not.DE, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/mates.proms.not.DE.RDS&quot;)
saveRDS(mates.proms.DC, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/mates.proms.DC.RDS&quot;)
saveRDS(mates.proms.not.DC, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/mates.proms.not.DC.RDS&quot;)
saveRDS(mates.proms.DE.only, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/mates.proms.DE.only.RDS&quot;)
saveRDS(mates.proms.DC.only, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/mates.proms.DC.only.RDS&quot;)
saveRDS(mates.proms.DCDE.only, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/mates.proms.DCDE.only.RDS&quot;)
saveRDS(mates.proms.neither.only, file=&quot;~/Desktop/enrichdfs/metaproms/justmeta/mates.proms.neither.only.RDS&quot;)</code></pre>
</div>
<div id="chromhmm-visualization" class="section level3">
<h3>ChromHMM Visualization</h3>
<p>Now, I load in the chromHMM proportion tables from midway2, and visualize them to examine differences in dynamics of regulatory regions for promoter and non-promoter contacts.</p>
<pre class="r"><code>#Just a function for calculating proportions of chromHMM annotations in bin sets of the significant hits. Primarily used on midway2, created and tested here. No particular function at this point right now--loading of proportion tables begins shortly after.
anno.prop.calc &lt;- function(df, col1, col2, startbin, pval.col){
  categories &lt;- unique(c(levels(as.factor(df[,col1])), levels(as.factor(df[,col2]))))
  result &lt;- as.data.frame(matrix(ncol=length(categories), nrow=length((startbin+1):(nrow(df)+1))))
  colnames(result) &lt;- categories
  result$pval &lt;- NA
  firstbit &lt;- table(c(as.character(df[1:startbin, col1]), as.character(df[1:startbin, col2])))
  result[1, which(colnames(result) %in% names(firstbit))] &lt;- firstbit
  result[1, &quot;pval&quot;] &lt;- df[startbin, pval.col]
  for(row in (startbin+1):nrow(df)){
    tmp.table &lt;- table(c(as.character(df[1:row, col1]), as.character(df[1:row, col2])))
    result[row-(startbin-1), which(colnames(result) %in% names(tmp.table))] &lt;- tmp.table
    result[row-(startbin-1), &quot;pval&quot;] &lt;- df[row, pval.col]
  }
  return(result)
}

###Create vectors for the names of the chromHMM marks and their standardized colors, for passing to scale_fill_manual in ggplot2 calls below.
HMM.marks &lt;- c(&quot;Nothing&quot;, &quot;TssA&quot;, &quot;TssBiv&quot;, &quot;BivFlnk&quot;, &quot;EnhBiv&quot;,&quot;ReprPC&quot;, &quot;ReprPCWk&quot;, &quot;Quies&quot;, &quot;TssAFlnk&quot;, &quot;TxFlnk&quot;, &quot;Tx&quot;, &quot;TxWk&quot;, &quot;EnhG&quot;, &quot;Enh&quot;, &quot;ZNF/Rpts&quot;, &quot;Het&quot;)
HMM.colors &lt;- c(rgb(0, 0, 0, maxColorValue = 255), rgb(255, 0, 0, maxColorValue = 255), rgb(205, 92, 92, maxColorValue = 255),   rgb(233, 150, 122, maxColorValue = 255), rgb(189, 183, 107, maxColorValue = 255), rgb(128, 128, 128, maxColorValue = 255), rgb(192, 192, 192, maxColorValue = 255), rgb(255, 255, 255, maxColorValue = 255), rgb(255, 69, 0, maxColorValue = 255), rgb(50, 205, 50, maxColorValue = 255), rgb(0, 128, 0, maxColorValue = 255), rgb(0, 100, 0, maxColorValue = 255), rgb(194, 225, 5, maxColorValue = 255), rgb(255, 255, 0, maxColorValue = 255), rgb(102, 205, 170, maxColorValue = 255), rgb(138, 145, 208, maxColorValue = 255))

#######Read in proportion tables from running on midway2! First I&#39;ll do this on the full set of pairs that contain a promoter bin, and then on just the set of bins that are themselves contacting a promoter (and not the promoters themselves).
prom.props &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/full.promoters.RDS.result.RDS&quot;)
nonprom.props &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/full.no.promoters.RDS.result.RDS&quot;)

colnames(nonprom.props) &lt;- colnames(prom.props) &lt;- c(&quot;Nothing&quot;, &quot;TssA&quot;, &quot;TssBiv&quot;, &quot;BivFlnk&quot;, &quot;EnhBiv&quot;, &quot;ReprPC&quot;, &quot;ReprPCWk&quot;, &quot;Quies&quot;, &quot;TssAFlnk&quot;, &quot;TxFlnk&quot;, &quot;Tx&quot;, &quot;TxWk&quot;, &quot;EnhG&quot;, &quot;Enh&quot;, &quot;ZNF/Rpts&quot;, &quot;Het&quot;, &quot;pval&quot;)

max(which(prom.props$pval&lt;=0.3))</code></pre>
<pre><code>[1] 5418</code></pre>
<pre class="r"><code>max(which(nonprom.props$pval&lt;=0.3))</code></pre>
<pre><code>[1] 33956</code></pre>
<pre class="r"><code>prom.props$bin &lt;- 1:nrow(prom.props)
nonprom.props$bin &lt;- 1:nrow(nonprom.props)
prom.props &lt;- prom.props[,-17]
nonprom.props &lt;- nonprom.props[,-17]

ggprom &lt;- melt(prom.props[1:10000,], id=&quot;bin&quot;)
ggnoprom &lt;- melt(nonprom.props[1:10000,], id=&quot;bin&quot;) #Only take the first ~50k rows of this because it takes FOREVER otherwise and there are only ~36k promoters anyway.

#Visualize chromHMM annotation dynamics across FDR for promoter bins and non-promoter bins.
ggplot(data=ggprom, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;Proportion of chromHMM Annotations, Promoter Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + theme(plot.title=element_text(hjust=11)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 4017 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggnoprom, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;Proportion of chromHMM Annotations, non-Promoter Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + theme(plot.title=element_text(hjust=1)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 27119 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Maybe only the first 1000 are actually interesting...
ggprom &lt;- melt(prom.props[1:1000,], id=&quot;bin&quot;)
ggnoprom &lt;- melt(nonprom.props[1:1000,], id=&quot;bin&quot;) #Only take the first ~50k rows of this because it takes FOREVER otherwise and there are only ~36k promoters anyway.

#Visualize chromHMM annotation dynamics across FDR for promoter bins and non-promoter bins.
ggplot(data=ggprom, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;Proportion of chromHMM Annotations, Promoter Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 3725 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggnoprom, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;Proportion of chromHMM Annotations, non-Promoter Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + theme(plot.title=element_text(hjust=1)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 8099 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##Mabe it&#39;s even less than 1000 actually. Do it entirely by FDR instead? Reload and reformat the dfs to see pvals again.
prom.props &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/full.promoters.RDS.result.RDS&quot;)
nonprom.props &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/full.no.promoters.RDS.result.RDS&quot;)
colnames(nonprom.props) &lt;- colnames(prom.props) &lt;- c(&quot;Nothing&quot;, &quot;TssA&quot;, &quot;TssBiv&quot;, &quot;BivFlnk&quot;, &quot;EnhBiv&quot;, &quot;ReprPC&quot;, &quot;ReprPCWk&quot;, &quot;Quies&quot;, &quot;TssAFlnk&quot;, &quot;TxFlnk&quot;, &quot;Tx&quot;, &quot;TxWk&quot;, &quot;EnhG&quot;, &quot;Enh&quot;, &quot;ZNF/Rpts&quot;, &quot;Het&quot;, &quot;pval&quot;)
prom.props$pval[100]</code></pre>
<pre><code>[1] 0.0005873684</code></pre>
<pre class="r"><code>nonprom.props$pval[100]</code></pre>
<pre><code>[1] 0.0001820104</code></pre>
<pre class="r"><code>#The p-values are drastically different at these ranks unfortunately. But do one more visualization, even closer, 250 total. First reformat the dfs to work with ggplot:
prom.props$bin &lt;- 1:nrow(prom.props)
nonprom.props$bin &lt;- 1:nrow(nonprom.props)
prom.props &lt;- prom.props[,-17]
nonprom.props &lt;- nonprom.props[,-17]
ggprom &lt;- melt(prom.props[1:250,], id=&quot;bin&quot;)
ggnoprom &lt;- melt(nonprom.props[1:250,], id=&quot;bin&quot;)

#Visualize chromHMM annotation dynamics across FDR for promoter bins and non-promoter bins.
ggplot(data=ggprom, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;Proportion of chromHMM Annotations, Promoter Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + theme(plot.title=element_text(hjust=11)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 1942 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggnoprom, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;Proportion of chromHMM Annotations, non-Promoter Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + theme(plot.title=element_text(hjust=1)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 2149 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#These aren&#39;t great right now because the colors don&#39;t even match for the annotations between the two...highly annoying, useless.

#Read in, format properly, and visualize these same dynamics for promoter bins, split between those that were or were not DE/DC.
prom.DC.props &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/promoters.DC.RDS.result.RDS&quot;)
prom.DE.props &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/promoters.DE.RDS.result.RDS&quot;)
prom.noDE &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/promoters.not.DE.RDS.result.RDS&quot;)
prom.noDC &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/promoters.not.DC.RDS.result.RDS&quot;)

prom.DC.props$bin &lt;- 1:nrow(prom.DC.props)
prom.DE.props$bin &lt;- 1:nrow(prom.DE.props)
prom.noDE$bin &lt;- 1:nrow(prom.noDE)
prom.noDC$bin &lt;- 1:nrow(prom.noDC)
prom.DC.props &lt;- prom.DC.props[,-17]
prom.DE.props &lt;- prom.DE.props[,-17]
prom.noDE &lt;- prom.noDE[,-17]
prom.noDC &lt;- prom.noDC[,-17]

ggpromDC &lt;- melt(prom.DC.props, id=&quot;bin&quot;)
ggpromDE &lt;- melt(prom.DE.props[1:4000,], id=&quot;bin&quot;)
ggprom.noDE &lt;- melt(prom.noDE[1:4000,], id=&quot;bin&quot;)
ggprom.noDC &lt;- melt(prom.noDC[1:1250,], id=&quot;bin&quot;)

ggplot(data=ggpromDC, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM Annotations, DC Promoter Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 3957 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggprom.noDC, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM Annotations, non-DC Promoter Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 4450 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggpromDE, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM Annotations, DE Promoter Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 7273 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggprom.noDE, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM Annotations, non-DE Promoter Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 3434 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Again unfortunately p-values line up terribly, for instance max at the DC promoters is 5% and nonDC promoters 15% at the same rank (1200). At the smaller ranges (~250), it&#39;s even worse (.1% to 7%). In the DE case, at 4000 rank it&#39;s 65% and 29% DE to no DE, and around areas of extreme change (~250) it&#39;s 4.6% to .17%, and at 100 rank it&#39;s .47% to .06%...
#Because nothing is conclusive here, just do 10%, 5%, and 1% FDR below.

#####Repeat of what I just did, but now on only promoter-contacting bins (and not the promoters themselves).
#First, make plot of the full range of dynamics, then separated out by differential status, and then by DC and DE statuses.
prom.props &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/mates.proms.full.RDS.result.RDS&quot;)
both.props &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/mates.proms.DCDE.only.RDS.result.RDS&quot;)
neither.props &lt;-  readRDS(&quot;~/Desktop/enrichdfs/newresults/mates.proms.neither.only.RDS.result.RDS&quot;)
expression.props &lt;-  readRDS(&quot;~/Desktop/enrichdfs/newresults/mates.proms.DE.only.RDS.result.RDS&quot;)
contact.props &lt;-  readRDS(&quot;~/Desktop/enrichdfs/newresults/mates.proms.DC.only.RDS.result.RDS&quot;)

colnames(prom.props) &lt;- colnames(both.props) &lt;- colnames(neither.props) &lt;- colnames(expression.props) &lt;- colnames(contact.props) &lt;- c(&quot;Nothing&quot;, &quot;TssA&quot;, &quot;TssBiv&quot;, &quot;BivFlnk&quot;, &quot;EnhBiv&quot;, &quot;ReprPC&quot;, &quot;ReprPCWk&quot;, &quot;Quies&quot;, &quot;TssAFlnk&quot;, &quot;TxFlnk&quot;, &quot;Tx&quot;, &quot;TxWk&quot;, &quot;EnhG&quot;, &quot;Enh&quot;, &quot;ZNF/Rpts&quot;, &quot;Het&quot;, &quot;pval&quot;)

prom.props$bin &lt;- 1:nrow(prom.props)
both.props$bin &lt;- 1:nrow(both.props)
neither.props$bin &lt;- 1:nrow(neither.props)
expression.props$bin &lt;- 1:nrow(expression.props)
contact.props$bin &lt;- 1:nrow(contact.props)

prom.props &lt;- prom.props[,-17]
both.props &lt;- both.props[,-17]
neither.props &lt;- neither.props[,-17]
expression.props &lt;- expression.props[,-17]
contact.props &lt;- contact.props[,-17]

ggprom &lt;- melt(prom.props[1:10000,], id=&quot;bin&quot;)
ggneither &lt;- melt(neither.props[1:8000,], id=&quot;bin&quot;)
ggboth &lt;- melt(both.props, id=&quot;bin&quot;)
ggexpression &lt;- melt(expression.props, id=&quot;bin&quot;)
ggcontact &lt;- melt(contact.props, id=&quot;bin&quot;)

#Visualize chromHMM annotation dynamics across FDR for promoter bins, and then by differential status.
ggplot(data=ggprom, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM, Promoter-Contacting Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 5612 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggboth, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM, DC+DE Promoter-Contacting Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 2069 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggexpression, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM, DE Promoter-Contacting Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 14857 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggcontact, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM, DC Promoter-Contacting Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 3121 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggneither, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM, non-DC/DE Promoter-Contacting Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 9839 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-15.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Now, read in, format properly, and visualize these same dynamics for promoter bins, split between those that were or were not DE/DC.
prom.DC.props &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/mates.proms.DC.RDS.result.RDS&quot;)
prom.DE.props &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/mates.proms.DE.RDS.result.RDS&quot;)
prom.noDE &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/mates.proms.not.DE.RDS.result.RDS&quot;)
prom.noDC &lt;- readRDS(&quot;~/Desktop/enrichdfs/newresults/mates.proms.not.DC.RDS.result.RDS&quot;)

prom.DC.props$bin &lt;- 1:nrow(prom.DC.props)
prom.DE.props$bin &lt;- 1:nrow(prom.DE.props)
prom.noDE$bin &lt;- 1:nrow(prom.noDE)
prom.noDC$bin &lt;- 1:nrow(prom.noDC)
prom.DC.props &lt;- prom.DC.props[,-17]
prom.DE.props &lt;- prom.DE.props[,-17]
prom.noDE &lt;- prom.noDE[,-17]
prom.noDC &lt;- prom.noDC[,-17]

ggpromDC &lt;- melt(prom.DC.props, id=&quot;bin&quot;)
ggpromDE &lt;- melt(prom.DE.props[1:4000,], id=&quot;bin&quot;)
ggprom.noDE &lt;- melt(prom.noDE[1:4000,], id=&quot;bin&quot;)
ggprom.noDC &lt;- melt(prom.noDC[1:1250,], id=&quot;bin&quot;)

ggplot(data=ggpromDC, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM, DC Promoter-Contacting Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 3659 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggprom.noDC, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM, non-DC Promoter-Contacting Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 6847 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-17.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggpromDE, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM, DE Promoter-Contacting Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 9189 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=ggprom.noDE, aes(x=bin, group=bin, fill=variable, y=value)) + geom_col(position=&quot;fill&quot;) + ggtitle(&quot;chromHMM, non-DE Promoter-Contacting Bins&quot;) + xlab(&quot;Decreasing Significance from Hi-C LM&quot;) + ylab(&quot;Proportion of hits&quot;) + guides(fill=guide_legend(title=&quot;Annotations&quot;)) + scale_fill_manual(values=HMM.colors, labels=HMM.marks, drop=FALSE)</code></pre>
<pre><code>Warning: Removed 4591 rows containing missing values (position_stack).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-19.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Now, also look at specific FDR levels for enrichments of a particular overlap! Prepare a df that has both the promoter and nonpromoter values for these, with duplicates within classes being filtered out (set to NA via deduplicator).
head(mates.proms.full) #Created before, full set of promoter-contacting bins minus duplicates.</code></pre>
<pre><code>                  H2   sp_BH_pval   sp_beta bin2.HMM
1      chr7-43900000 7.759749e-05 -5.335645 15_Quies
2     chr13-46400000 9.808859e-05 -4.502457   5_TxWk
19100 chr9-129170000 1.320264e-04  2.770650 15_Quies
3      chr7-43940000 1.360426e-04 -4.735619   5_TxWk
19101 chr1-109770000 1.580912e-04  2.461795     4_Tx
19102  chr11-2430000 1.756877e-04 -2.594482 15_Quies
                             pairID Nothing.2 TssA.2 TssBiv.2 BivFlnk.2
1       chr7-39620000_chr7-43900000        NA   1400       NA        NA
2     chr13-45330000_chr13-46400000        NA     NA       NA        NA
19100 chr9-129170000_chr9-129600000        NA     NA       NA        NA
3       chr7-39620000_chr7-43940000        NA   1000       NA        NA
19101 chr1-109770000_chr1-109980000        NA     NA       NA        NA
19102   chr11-2430000_chr11-2880000        NA     NA       NA        NA
      EnhBiv.2 ReprPC.2 ReprPCWk.2 Quies.2 TssAFlnk.2 TxFlnk.2 Tx.2 TxWk.2
1           NA       NA         NA    4200        600       NA 1000   2600
2           NA       NA         NA      NA         NA       NA   NA   8800
19100       NA       NA         NA    5800         NA       NA   NA   3000
3           NA       NA       1200    2800        800       NA   NA   3600
19101       NA       NA         NA      NA       1000       NA 5600    800
19102       NA       NA         NA   10000         NA       NA   NA     NA
      EnhG.2 Enh.2 ZNF/Rpts.2 Het.2 bin2.DHSval bin2.Hacval bin2.Hme1val
1         NA   200         NA    NA           0           0         1159
2         NA  1200         NA    NA           0           0            0
19100     NA  1200         NA    NA         300           0         1442
3         NA   600         NA    NA           0           0            0
19101   1200  1400         NA    NA         300           0            0
19102     NA    NA         NA    NA           0           0            0
      bin2.Hme3val bin2.HARval  DE  DC differential
1             2163           0  no yes      contact
2                0           0 yes yes         both
19100         5076           0  no yes      contact
3                0           0  no yes      contact
19101            0           0 yes yes         both
19102            0           0  no yes      contact</code></pre>
<pre class="r"><code>nonprom.annos &lt;- filter(full.annotations, true.promoter==&quot;no&quot;) %&gt;% deduplicator(.)
dubs.nonprom &lt;- data.frame(H1=rep(nonprom.annos$H1, 2), H2=rep(nonprom.annos$H2, 2), sp_BH_pval=rep(nonprom.annos$sp_BH_pval, 2), sp_beta=rep(nonprom.annos$sp_beta, 2), bin1.HMM=rep(nonprom.annos$bin1.HMM, 2), bin2.HMM=rep(nonprom.annos$bin2.HMM, 2), pairID=rep(nonprom.annos$pairID, 2), Nothing=c(nonprom.annos$Nothing.1, nonprom.annos$Nothing.2), TssA=c(nonprom.annos$TssA.1, nonprom.annos$TssA.2), TssBiv=c(nonprom.annos$TssBiv.1, nonprom.annos$TssBiv.2), BivFlnk=c(nonprom.annos$TssBiv.1, nonprom.annos$TssBiv.2), EnhBiv=c(nonprom.annos$EnhBiv.1, nonprom.annos$EnhBiv.2), ReprPc=c(nonprom.annos$ReprPC.1, nonprom.annos$ReprPC.2), ReprPCWk=c(nonprom.annos$ReprPCWk.1, nonprom.annos$ReprPCWk.2), Quies=c(nonprom.annos$Quies.1, nonprom.annos$Quies.2), TssAFlnk=c(nonprom.annos$TssAFlnk.1, nonprom.annos$TssAFlnk.2), TxFlnk=c(nonprom.annos$TxFlnk.1, nonprom.annos$TxFlnk.2), Tx=c(nonprom.annos$Tx.1, nonprom.annos$Tx.2), TxWk=c(nonprom.annos$TxWk.1, nonprom.annos$TxWk.2), EnhG=c(nonprom.annos$EnhG.1, nonprom.annos$EnhG.2), Enh=c(nonprom.annos$Enh.1, nonprom.annos$Enh.2), ZNF_Rpts=c(nonprom.annos$`ZNF/Rpts.1`, nonprom.annos$`ZNF/Rpts.2`), Het=c(nonprom.annos$Het.1, nonprom.annos$Het.2), DHSval=c(nonprom.annos$bin1.DHSval, nonprom.annos$bin2.DHSval), Hacval=c(nonprom.annos$bin1.Hacval, nonprom.annos$bin2.Hacval), Hme1val=c(nonprom.annos$bin1.Hme1val, nonprom.annos$bin2.Hme1val), Hme3val=c(nonprom.annos$bin1.Hme3val, nonprom.annos$bin2.Hme3val), HARval=c(nonprom.annos$bin1.HARval, nonprom.annos$bin2.HARval), DE=rep(nonprom.annos$DE, 2), DC=rep(nonprom.annos$DC, 2), differential=rep(nonprom.annos$differential, 2))#, CTCF.interrupt=rep(nonprom.annos$CTCF.interrupt, 2), true.promoter=rep(nonprom.annos$true.promoter, 2), promoter.bin1=rep(nonprom.annos$promoter.bin1, 2), promoter.bin2=rep(nonprom.annos$promoter.bin2, 2))
dubs.nonprom &lt;- dubs.nonprom[,c(-1, -5)]
colnames(mates.proms.full) &lt;- colnames(dubs.nonprom)
mates.proms.full$true.promoter &lt;-  &quot;yes&quot; #Create a variable to separate them before combining
dubs.nonprom$true.promoter &lt;- &quot;no&quot;

#Now, for examining specific chromHMM marks, will want to combine these two--true.promoter is the variable that separates them out!
chrom.annos &lt;- rbind(dubs.nonprom, mates.proms.full)
chrom.annos &lt;- chrom.annos[order(chrom.annos$sp_BH_pval),] #Order by FDR. Duplicates have already been removed from each set. Set sizes are:
sum(chrom.annos$true.promoter==&quot;yes&quot;)</code></pre>
<pre><code>[1] 27001</code></pre>
<pre class="r"><code>sum(chrom.annos$true.promoter==&quot;no&quot;)</code></pre>
<pre><code>[1] 511952</code></pre>
<pre class="r"><code>#Do the same thing for creating the other necessary DFs in comparison to the promoter-contacting set of bins. First DC/not and DE/not, then differential statuses.
#Also do exclusive sets for differential status on the promoter bins:
nonprom.DC &lt;- filter(full.annotations, true.promoter==&quot;no&quot;, DC==&quot;yes&quot;) %&gt;% deduplicator(.)
nonprom.notDC &lt;- filter(full.annotations, true.promoter==&quot;no&quot;, DC==&quot;no&quot;) %&gt;% deduplicator(.)
dubs.nonprom.DC &lt;- data.frame(H1=rep(nonprom.DC$H1, 2), H2=rep(nonprom.DC$H2, 2), sp_BH_pval=rep(nonprom.DC$sp_BH_pval, 2), sp_beta=rep(nonprom.DC$sp_beta, 2), bin1.HMM=rep(nonprom.DC$bin1.HMM, 2), bin2.HMM=rep(nonprom.DC$bin2.HMM, 2), pairID=rep(nonprom.DC$pairID, 2), Nothing=c(nonprom.DC$Nothing.1, nonprom.DC$Nothing.2), TssA=c(nonprom.DC$TssA.1, nonprom.DC$TssA.2), TssBiv=c(nonprom.DC$TssBiv.1, nonprom.DC$TssBiv.2), BivFlnk=c(nonprom.DC$TssBiv.1, nonprom.DC$TssBiv.2), EnhBiv=c(nonprom.DC$EnhBiv.1, nonprom.DC$EnhBiv.2), ReprPc=c(nonprom.DC$ReprPC.1, nonprom.DC$ReprPC.2), ReprPCWk=c(nonprom.DC$ReprPCWk.1, nonprom.DC$ReprPCWk.2), Quies=c(nonprom.DC$Quies.1, nonprom.DC$Quies.2), TssAFlnk=c(nonprom.DC$TssAFlnk.1, nonprom.DC$TssAFlnk.2), TxFlnk=c(nonprom.DC$TxFlnk.1, nonprom.DC$TxFlnk.2), Tx=c(nonprom.DC$Tx.1, nonprom.DC$Tx.2), TxWk=c(nonprom.DC$TxWk.1, nonprom.DC$TxWk.2), EnhG=c(nonprom.DC$EnhG.1, nonprom.DC$EnhG.2), Enh=c(nonprom.DC$Enh.1, nonprom.DC$Enh.2), ZNF_Rpts=c(nonprom.DC$`ZNF/Rpts.1`, nonprom.DC$`ZNF/Rpts.2`), Het=c(nonprom.DC$Het.1, nonprom.DC$Het.2), DHSval=c(nonprom.DC$bin1.DHSval, nonprom.DC$bin2.DHSval), Hacval=c(nonprom.DC$bin1.Hacval, nonprom.DC$bin2.Hacval), Hme1val=c(nonprom.DC$bin1.Hme1val, nonprom.DC$bin2.Hme1val), Hme3val=c(nonprom.DC$bin1.Hme3val, nonprom.DC$bin2.Hme3val), HARval=c(nonprom.DC$bin1.HARval, nonprom.DC$bin2.HARval), DE=rep(nonprom.DC$DE, 2), DC=rep(nonprom.DC$DC, 2), differential=rep(nonprom.DC$differential, 2))
dubs.nonprom.notDC &lt;- data.frame(H1=rep(nonprom.notDC$H1, 2), H2=rep(nonprom.notDC$H2, 2), sp_BH_pval=rep(nonprom.notDC$sp_BH_pval, 2), sp_beta=rep(nonprom.notDC$sp_beta, 2), bin1.HMM=rep(nonprom.notDC$bin1.HMM, 2), bin2.HMM=rep(nonprom.notDC$bin2.HMM, 2), pairID=rep(nonprom.notDC$pairID, 2), Nothing=c(nonprom.notDC$Nothing.1, nonprom.notDC$Nothing.2), TssA=c(nonprom.notDC$TssA.1, nonprom.notDC$TssA.2), TssBiv=c(nonprom.notDC$TssBiv.1, nonprom.notDC$TssBiv.2), BivFlnk=c(nonprom.notDC$TssBiv.1, nonprom.notDC$TssBiv.2), EnhBiv=c(nonprom.notDC$EnhBiv.1, nonprom.notDC$EnhBiv.2), ReprPc=c(nonprom.notDC$ReprPC.1, nonprom.notDC$ReprPC.2), ReprPCWk=c(nonprom.notDC$ReprPCWk.1, nonprom.notDC$ReprPCWk.2), Quies=c(nonprom.notDC$Quies.1, nonprom.notDC$Quies.2), TssAFlnk=c(nonprom.notDC$TssAFlnk.1, nonprom.notDC$TssAFlnk.2), TxFlnk=c(nonprom.notDC$TxFlnk.1, nonprom.notDC$TxFlnk.2), Tx=c(nonprom.notDC$Tx.1, nonprom.notDC$Tx.2), TxWk=c(nonprom.notDC$TxWk.1, nonprom.notDC$TxWk.2), EnhG=c(nonprom.notDC$EnhG.1, nonprom.notDC$EnhG.2), Enh=c(nonprom.notDC$Enh.1, nonprom.notDC$Enh.2), ZNF_Rpts=c(nonprom.notDC$`ZNF/Rpts.1`, nonprom.notDC$`ZNF/Rpts.2`), Het=c(nonprom.notDC$Het.1, nonprom.notDC$Het.2), DHSval=c(nonprom.notDC$bin1.DHSval, nonprom.notDC$bin2.DHSval), Hacval=c(nonprom.notDC$bin1.Hacval, nonprom.notDC$bin2.Hacval), Hme1val=c(nonprom.notDC$bin1.Hme1val, nonprom.notDC$bin2.Hme1val), Hme3val=c(nonprom.notDC$bin1.Hme3val, nonprom.notDC$bin2.Hme3val), HARval=c(nonprom.notDC$bin1.HARval, nonprom.notDC$bin2.HARval), DE=rep(nonprom.notDC$DE, 2), DC=rep(nonprom.notDC$DC, 2), differential=rep(nonprom.notDC$differential, 2))
nonproms.DC.df &lt;- rbind(dubs.nonprom.DC, dubs.nonprom.notDC)

#Prep promoters DC df, DE df, and differential DF. This is definitely redundant code throughout all three chunks here, so eventually fix that!!!
DC.proms &lt;- filter(promoter.contacts, DC==&quot;yes&quot;)
DC.proms &lt;- DC.proms[!duplicated(DC.proms$H2),]
nonDC.proms &lt;- filter(promoter.contacts, DC==&quot;no&quot;)
nonDC.proms &lt;- nonDC.proms[!duplicated(nonDC.proms$H2),]
proms.DC &lt;- rbind(DC.proms, nonDC.proms)
proms.DC &lt;- proms.DC[order(proms.DC$sp_BH_pval),] #Should be complete!

DE.proms &lt;- filter(promoter.contacts, DE==&quot;yes&quot;)
DE.proms &lt;- DE.proms[!duplicated(DE.proms$H2),]
nonDE.proms &lt;- filter(promoter.contacts, DE==&quot;no&quot;)
nonDE.proms &lt;- nonDE.proms[!duplicated(nonDE.proms$H2),]
proms.DE &lt;- rbind(DE.proms, nonDE.proms)
proms.DE &lt;- proms.DE[order(proms.DE$sp_BH_pval),]

DE.only.proms &lt;- filter(promoter.contacts, differential==&quot;expression&quot;)
DC.only.proms &lt;- filter(promoter.contacts, differential==&quot;contact&quot;)
DEDC.proms &lt;- filter(promoter.contacts, differential==&quot;both&quot;)
neither.proms &lt;- filter(promoter.contacts, differential==&quot;no&quot;)
DE.only.proms &lt;- DE.only.proms[!duplicated(DE.only.proms$H2),]
DC.only.proms &lt;- DC.only.proms[!duplicated(DC.only.proms$H2),]
DEDC.proms &lt;- DEDC.proms[!duplicated(DEDC.proms$H2),]
neither.proms &lt;- neither.proms[!duplicated(neither.proms$H2),]
proms.differential &lt;- rbind(DE.only.proms, DC.only.proms, DEDC.proms, neither.proms)
proms.differential &lt;- proms.differential[order(proms.differential$sp_BH_pval),]
colnames(proms.differential) &lt;- colnames(proms.DE) &lt;- colnames(proms.DC) &lt;- colnames(chrom.annos)[1:29]


##########Now, try to actually plot out overlaps of different chromHMM marks with these two different classes of bins.
#First, create a function to do the plotting, and to do it with different FDRs. At first was trying to make this work for the individual histone marks as well but those already exist so just format to work with these. Will have to pass in properly formatted dataframes, however.
chromHMMplot &lt;- function(df, nonprom.df, DC.df, DE.df, differential.df, differential.Ymax, differential.Ystart, diff.lineint, diff.textint, markcol, vpdims=c(0.3, 0.3, 0.5, 0.7), vpdims2=vpdims, vpdims3=vpdims, vpdims4=vpdims, vpdims5=vpdims, FDR=NULL){
    DEPROMS=FALSE
    DCPROMS=FALSE
    NONPROMS=FALSE
    DIFFERENTIAL=FALSE
  if(is.numeric(FDR)){
  density.proms &lt;- ggplot(data=filter(df, sp_BH_pval&lt;=FDR)) + geom_density(aes(x=get(markcol), group=true.promoter, color=true.promoter, y=..scaled..)) + ylab(&quot;Density&quot;) + xlab(paste(&quot;Length of overlap b/t bin and&quot;, markcol, &quot;Annotation&quot;, sep=&quot; &quot;)) + ggtitle(paste(&quot;Overlap between Hi-C Bins &lt;=&quot;, FDR*100, &quot;% FDR and&quot;, markcol, sep=&quot; &quot;)) + scale_colour_manual(name=&quot;Promoter Contact?&quot;, labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;))
  mypval &lt;- t.test(filter(df, true.promoter==&quot;yes&quot;, sp_BH_pval&lt;=FDR)[,markcol], filter(df, true.promoter==&quot;no&quot;, sp_BH_pval&lt;=FDR)[,markcol])$p.value
  mypval &lt;- signif(mypval, 1)
  violin.proms &lt;- ggplot(data=filter(df, sp_BH_pval&lt;=FDR)) + geom_violin(aes(y=get(markcol), group=true.promoter, color=true.promoter, x=true.promoter), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(values=c(&quot;Red&quot;, &quot;Blue&quot;), labels=c(&quot;No&quot;, &quot;Yes&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;text&quot;, x=1.5, y=3000, label=mypval, size=3)
  
  if(nrow(filter(nonprom.df, sp_BH_pval&lt;=FDR, DC==&quot;no&quot;))&gt;0){NONPROMS=TRUE
  density.nonproms &lt;- ggplot(data=filter(nonprom.df, sp_BH_pval&lt;=FDR)) + geom_density(aes(x=get(markcol), group=DC, color=DC, y=..scaled..)) + ylab(&quot;Density&quot;) + xlab(paste(&quot;Length of overlap b/t bin and&quot;, markcol, &quot;Annotation&quot;, sep=&quot; &quot;)) + ggtitle(paste(&quot;Overlap between Hi-C Bins &lt;=&quot;, FDR*100, &quot;% FDR and&quot;, markcol, sep=&quot; &quot;)) + scale_colour_manual(name=&quot;Differential Contact?&quot;, labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;))
  mypval &lt;- t.test(filter(nonprom.df, sp_BH_pval&lt;=FDR, DC==&quot;yes&quot;)[,markcol], filter(nonprom.df, sp_BH_pval&lt;=FDR, DC==&quot;no&quot;)[,markcol])$p.value
  mypval &lt;- signif(mypval, 1)
  violin.nonproms &lt;- ggplot(data=filter(nonprom.df, sp_BH_pval&lt;=FDR)) + geom_violin(aes(y=get(markcol), group=DC, color=DC, x=DC), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(name=&quot;Differential Contact?&quot;, values=c(&quot;Red&quot;, &quot;Blue&quot;), labels=c(&quot;No&quot;, &quot;Yes&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;text&quot;, x=1.5, y=3000, label=mypval, size=3)}
  
  if(nrow(filter(DC.df, sp_BH_pval&lt;=FDR, DC==&quot;no&quot;))&gt;0){DCPROMS=TRUE
  DC.density &lt;- ggplot(data=filter(DC.df, sp_BH_pval&lt;=FDR)) + geom_density(aes(x=get(markcol), group=DC, color=DC, y=..scaled..)) + ggtitle(paste(&quot;Overlap between Promoter Hi-C Bins&lt;=&quot;, FDR*100, &quot;%FDR and&quot;, markcol, sep=&quot; &quot;)) + xlab(paste(&quot;Length of overlap b/t bin and&quot;, markcol, &quot;peak&quot;, sep=&quot; &quot;)) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DC?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;))
  mypval &lt;- t.test(filter(DC.df, sp_BH_pval&lt;=FDR, DC==&quot;yes&quot;)[,markcol], filter(DC.df, sp_BH_pval&lt;=FDR, DC==&quot;no&quot;)[,markcol])$p.value
  mypval &lt;- signif(mypval, 1)
  DC.violin &lt;- ggplot(data=filter(DC.df, sp_BH_pval&lt;=FDR)) + geom_violin(aes(x=DC, y=get(markcol), group=DC, color=DC), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;text&quot;, x=1.5, y=3000, label=mypval, size=3)}
  
  if(nrow(filter(DE.df, sp_BH_pval&lt;=FDR, DE==&quot;no&quot;))&gt;0){DEPROMS=TRUE
  DE.density &lt;- ggplot(data=filter(DE.df, sp_BH_pval&lt;=FDR)) + geom_density(aes(x=get(markcol), group=DE, color=DE, y=..scaled..)) + ggtitle(paste(&quot;Overlap between Promoter Hi-C Bins&lt;=&quot;, FDR*100, &quot;%FDR and&quot;, markcol, sep=&quot; &quot;)) + xlab(paste(&quot;Length of overlap b/t bin and&quot;, markcol, &quot;peak&quot;, sep=&quot; &quot;)) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DE?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;))
  mypval &lt;- t.test(filter(DE.df, sp_BH_pval&lt;=FDR, DE==&quot;yes&quot;)[,markcol], filter(DE.df, sp_BH_pval&lt;=FDR, DE==&quot;no&quot;)[,markcol])$p.value
  mypval &lt;- signif(mypval, 1)
  DE.violin &lt;- ggplot(data=filter(DE.df, sp_BH_pval&lt;=FDR)) + geom_violin(aes(x=DE, y=get(markcol), group=DE, color=DE), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;text&quot;, x=1.5, y=3000, label=mypval, size=3)}
  
  if((nrow(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;both&quot;))&gt;0)&amp;(nrow(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;expression&quot;))&gt;0)&amp;(nrow(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;contact&quot;))&gt;0)&amp;(nrow(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;no&quot;))&gt;0&amp;(nrow(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;expression&quot;))-sum(is.na(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;expression&quot;)[,markcol]))&gt;1))){DIFFERENTIAL=TRUE
  differential.density &lt;- ggplot(data=filter(differential.df, sp_BH_pval&lt;=FDR)) + geom_density(aes(x=get(markcol), group=differential, color=differential, y=..scaled..)) + ggtitle(paste(&quot;Overlap between Promoter Hi-C Bins&lt;=&quot;, FDR*100, &quot;%FDR and&quot;, markcol, sep=&quot; &quot;)) + xlab(paste(&quot;Length of overlap b/t bin and&quot;, markcol, &quot;peak&quot;, sep=&quot; &quot;)) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;))
  NEpval &lt;- signif(t.test(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;no&quot;)[,markcol], filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;expression&quot;)[,markcol])$p.value, 1)
  NCpval &lt;- signif(t.test(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;no&quot;)[,markcol], filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;contact&quot;)[,markcol])$p.value, 1)
  NBpval &lt;- signif(t.test(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;no&quot;)[,markcol], filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;both&quot;)[,markcol])$p.value, 1)
  ECpval &lt;- signif(t.test(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;expression&quot;)[,markcol], filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;contact&quot;)[,markcol])$p.value, 1)
  EBpval &lt;- signif(t.test(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;expression&quot;)[,markcol], filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;both&quot;)[,markcol])$p.value, 1)
  CBpval &lt;- signif(t.test(filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;both&quot;)[,markcol], filter(differential.df, sp_BH_pval&lt;=FDR, differential==&quot;contact&quot;)[,markcol])$p.value, 1)
  differential.violin &lt;- ggplot(data=filter(proms.differential, sp_BH_pval&lt;=FDR)) + geom_violin(aes(x=differential, y=get(markcol), group=differential, color=differential), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, differential.Ymax)) + annotate(&quot;segment&quot;, x=1.1, y=differential.Ystart, xend=1.9, yend=differential.Ystart) + annotate(&quot;segment&quot;, x=2.1, y=differential.Ystart, xend=2.9, yend=differential.Ystart) + annotate(&quot;segment&quot;, x=3.1, y=differential.Ystart, xend=3.9, yend=differential.Ystart) + annotate(&quot;segment&quot;, x=2.1, y=differential.Ystart+diff.lineint, xend=3.9, yend=differential.Ystart+diff.lineint)  + annotate(&quot;segment&quot;, x=1.1, y=differential.Ystart+(2*diff.lineint), xend=2.9, yend=differential.Ystart+(2*diff.lineint)) + annotate(&quot;segment&quot;, x=1.1, y=differential.Ystart+(3*diff.lineint), xend=3.9, yend=differential.Ystart+(3*diff.lineint)) + annotate(&quot;text&quot;, x=3.5, y=differential.Ystart+diff.textint, label=NEpval, size=3) + annotate(&quot;text&quot;, x=2.5, y=differential.Ystart+diff.textint, label=ECpval, size=3) + annotate(&quot;text&quot;, x=1.5, y=differential.Ystart+diff.textint, label=CBpval, size=3) + annotate(&quot;text&quot;, x=3, y=differential.Ystart+diff.lineint+diff.textint, label=NCpval, size=3) + annotate(&quot;text&quot;, x=2, y=differential.Ystart+(2*diff.lineint)+diff.textint, label=EBpval, size=3) + annotate(&quot;text&quot;, x=2.5, y=differential.Ystart+(3*diff.lineint)+diff.textint, label=NBpval, size=3)
  }}
  else{#The case where we&#39;re not doing any FDR filtering.
    density.proms &lt;- ggplot(data=filter(df)) + geom_density(aes(x=get(markcol), group=true.promoter, color=true.promoter, y=..scaled..)) + ylab(&quot;Density&quot;) + xlab(paste(&quot;Length of overlap b/t bin and&quot;, markcol, &quot;Annotation&quot;, sep=&quot; &quot;)) + ggtitle(paste(&quot;Overlap between Hi-C Bins and&quot;, markcol, sep=&quot; &quot;)) + scale_colour_manual(name=&quot;Promoter Contact?&quot;, labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;))
  mypval &lt;- t.test(filter(df, true.promoter==&quot;yes&quot;)[,markcol], filter(df, true.promoter==&quot;no&quot;)[,markcol])$p.value
  mypval &lt;- signif(mypval, 1)
  violin.proms &lt;- ggplot(data=filter(df)) + geom_violin(aes(y=get(markcol), group=true.promoter, color=true.promoter, x=true.promoter), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(values=c(&quot;Red&quot;, &quot;Blue&quot;), labels=c(&quot;No&quot;, &quot;Yes&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;text&quot;, x=1.5, y=3000, label=mypval, size=3)
  
  density.nonproms &lt;- ggplot(data=filter(nonprom.df)) + geom_density(aes(x=get(markcol), group=DC, color=DC, y=..scaled..)) + ylab(&quot;Density&quot;) + xlab(paste(&quot;Length of overlap b/t bin and&quot;, markcol, &quot;Annotation&quot;, sep=&quot; &quot;)) + ggtitle(paste(&quot;Overlap between Hi-C Bins and&quot;, markcol, sep=&quot; &quot;)) + scale_colour_manual(name=&quot;Differential Contact?&quot;, labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;))
  mypval &lt;- t.test(filter(nonprom.df, DC==&quot;yes&quot;)[,markcol], filter(nonprom.df, DC==&quot;no&quot;)[,markcol])$p.value
  mypval &lt;- signif(mypval, 1)
  violin.nonproms &lt;- ggplot(data=filter(nonprom.df)) + geom_violin(aes(y=get(markcol), group=DC, color=DC, x=DC), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(name=&quot;Differential Contact?&quot;, values=c(&quot;Red&quot;, &quot;Blue&quot;), labels=c(&quot;No&quot;, &quot;Yes&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;text&quot;, x=1.5, y=3000, label=mypval, size=3)
  
  DC.density &lt;- ggplot(data=filter(DC.df)) + geom_density(aes(x=get(markcol), group=DC, color=DC, y=..scaled..)) + ggtitle(paste(&quot;Overlap between Promoter Hi-C Bins and&quot;, markcol, sep=&quot; &quot;)) + xlab(paste(&quot;Length of overlap b/t bin and&quot;, markcol, &quot;peak&quot;, sep=&quot; &quot;)) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DC?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;))
  mypval &lt;- t.test(filter(DC.df, DC==&quot;yes&quot;)[,markcol], filter(DC.df, DC==&quot;no&quot;)[,markcol])$p.value
  mypval &lt;- signif(mypval, 1)
  DC.violin &lt;- ggplot(data=filter(DC.df)) + geom_violin(aes(x=DC, y=get(markcol), group=DC, color=DC), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;text&quot;, x=1.5, y=3000, label=mypval, size=3)
  
  DE.density &lt;- ggplot(data=filter(DE.df)) + geom_density(aes(x=get(markcol), group=DE, color=DE, y=..scaled..)) + ggtitle(paste(&quot;Overlap between Promoter Hi-C Bins and&quot;, markcol, sep=&quot; &quot;)) + xlab(paste(&quot;Length of overlap b/t bin and&quot;, markcol, &quot;peak&quot;, sep=&quot; &quot;)) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DE?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;))
  mypval &lt;- t.test(filter(DE.df, DE==&quot;yes&quot;)[,markcol], filter(DE.df, DE==&quot;no&quot;)[,markcol])$p.value
  mypval &lt;- signif(mypval, 1)
  DE.violin &lt;- ggplot(data=filter(DE.df)) + geom_violin(aes(x=DE, y=get(markcol), group=DE, color=DE), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;text&quot;, x=1.5, y=3000, label=mypval, size=3)
  
  differential.density &lt;- ggplot(data=filter(differential.df)) + geom_density(aes(x=get(markcol), group=differential, color=differential, y=..scaled..)) + ggtitle(paste(&quot;Overlap between Promoter Hi-C Bins and&quot;, markcol, sep=&quot; &quot;)) + xlab(paste(&quot;Length of overlap b/t bin and&quot;, markcol, &quot;peak&quot;, sep=&quot; &quot;)) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;))
  NEpval &lt;- signif(t.test(filter(differential.df, differential==&quot;no&quot;)[,markcol], filter(differential.df, differential==&quot;expression&quot;)[,markcol])$p.value, 1)
  NCpval &lt;- signif(t.test(filter(differential.df, differential==&quot;no&quot;)[,markcol], filter(differential.df, differential==&quot;contact&quot;)[,markcol])$p.value, 1)
  NBpval &lt;- signif(t.test(filter(differential.df, differential==&quot;no&quot;)[,markcol], filter(differential.df, differential==&quot;both&quot;)[,markcol])$p.value, 1)
  ECpval &lt;- signif(t.test(filter(differential.df, differential==&quot;expression&quot;)[,markcol], filter(differential.df, differential==&quot;contact&quot;)[,markcol])$p.value, 1)
  EBpval &lt;- signif(t.test(filter(differential.df, differential==&quot;expression&quot;)[,markcol], filter(differential.df, differential==&quot;both&quot;)[,markcol])$p.value, 1)
  CBpval &lt;- signif(t.test(filter(differential.df, differential==&quot;both&quot;)[,markcol], filter(differential.df, differential==&quot;contact&quot;)[,markcol])$p.value, 1)
  differential.violin &lt;- ggplot(data=filter(proms.differential)) + geom_violin(aes(x=differential, y=get(markcol), group=differential, color=differential), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, differential.Ymax)) + annotate(&quot;segment&quot;, x=1.1, y=differential.Ystart, xend=1.9, yend=differential.Ystart) + annotate(&quot;segment&quot;, x=2.1, y=differential.Ystart, xend=2.9, yend=differential.Ystart) + annotate(&quot;segment&quot;, x=3.1, y=differential.Ystart, xend=3.9, yend=differential.Ystart) + annotate(&quot;segment&quot;, x=2.1, y=differential.Ystart+diff.lineint, xend=3.9, yend=differential.Ystart+diff.lineint)  + annotate(&quot;segment&quot;, x=1.1, y=differential.Ystart+(2*diff.lineint), xend=2.9, yend=differential.Ystart+(2*diff.lineint)) + annotate(&quot;segment&quot;, x=1.1, y=differential.Ystart+(3*diff.lineint), xend=3.9, yend=differential.Ystart+(3*diff.lineint)) + annotate(&quot;text&quot;, x=3.5, y=differential.Ystart+diff.textint, label=NEpval, size=3) + annotate(&quot;text&quot;, x=2.5, y=differential.Ystart+diff.textint, label=ECpval, size=3) + annotate(&quot;text&quot;, x=1.5, y=differential.Ystart+diff.textint, label=CBpval, size=3) + annotate(&quot;text&quot;, x=3, y=differential.Ystart+diff.lineint+diff.textint, label=NCpval, size=3) + annotate(&quot;text&quot;, x=2, y=differential.Ystart+(2*diff.lineint)+diff.textint, label=EBpval, size=3) + annotate(&quot;text&quot;, x=2.5, y=differential.Ystart+(3*diff.lineint)+diff.textint, label=NBpval, size=3)
  }
  print(density.proms)
  vp &lt;- grid::viewport(width=vpdims[1], height=vpdims[2], x=vpdims[3], y=vpdims[4])
  print(violin.proms, vp=vp)
  if(is.null(FDR)|NONPROMS){
  print(density.nonproms)
  vp &lt;- grid::viewport(width=vpdims2[1], height=vpdims2[2], x=vpdims2[3], y=vpdims2[4])
  print(violin.nonproms, vp=vp)}
  if(is.null(FDR)|DCPROMS){
  print(DC.density)
  vp &lt;- grid::viewport(width=vpdims3[1], height=vpdims3[2], x=vpdims3[3], y=vpdims3[4])
  print(DC.violin, vp=vp)}
  if(is.null(FDR)|DEPROMS){
  print(DE.density)
  vp &lt;- grid::viewport(width=vpdims4[1], height=vpdims4[2], x=vpdims4[3], y=vpdims4[4])
  print(DE.violin, vp=vp)}
  if(is.null(FDR)|DIFFERENTIAL){
  print(differential.density)
  vp &lt;- grid::viewport(width=vpdims5[1], height=vpdims5[2], x=vpdims5[3], y=vpdims5[4])
  print(differential.violin, vp=vp)}
}

#First call down there is chrom.annos, second can be nonproms.DC.df, third can be proms.DC, fourth can be proms.DE, fifth should be proms.differential, and the others will need fine tuning!
####TSSA!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssA&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 527862 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 527862 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-20.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 503851 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 503851 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-21.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 23774 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23774 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-22.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 26053 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 26053 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-23.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 26195 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 26195 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-24.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssA&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 23830 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23830 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-25.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 22317 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 22317 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-26.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1393 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1393 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-27.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1421 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1421 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-28.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1439 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1439 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-29.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssA&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 16946 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 16946 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-30.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 931 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 931 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-31.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssA&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 10784 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 10784 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-32.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 543 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 543 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-33.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###TssBiv!!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssBiv&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 532657 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 532657 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-34.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 507090 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 507090 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-35.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 25512 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 25512 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-36.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 28058 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 28058 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-37.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 28217 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 28217 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-38.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssBiv&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 24405 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 24405 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-39.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 22826 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 22826 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-40.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1534 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1534 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-41.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1576 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1576 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-42.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1598 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1598 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-43.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssBiv&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 17364 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 17364 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-44.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1054 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1054 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-45.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssBiv&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 11016 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 11016 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-46.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 600 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 600 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-47.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###BivFlnk!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;BivFlnk&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 532812 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 532812 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-48.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 507090 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 507090 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-49.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 25669 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 25669 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-50.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 28241 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 28241 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-51.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 28402 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 28402 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-52.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;BivFlnk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 24418 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 24418 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-53.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 22826 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 22826 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-54.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1548 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1548 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-55.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1590 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1590 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-56.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1613 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1613 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-57.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;BivFlnk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 17376 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 17376 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-58.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1067 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1067 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-59.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;BivFlnk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 11028 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 11028 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-60.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 615 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 615 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-61.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###EnhBiv!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;EnhBiv&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 532268 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 532268 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-62.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 506631 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 506631 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-63.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 25600 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 25600 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-64.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 28169 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 28169 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-65.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 28333 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 28333 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-66.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;EnhBiv&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 24439 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 24439 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-67.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 22834 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 22834 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-68.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1574 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1574 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-69.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1617 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1617 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-70.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1640 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1640 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-71.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;EnhBiv&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 17399 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 17399 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-72.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1083 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1083 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-73.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;EnhBiv&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 11042 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 11042 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-74.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 621 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 621 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-75.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###ReprPc!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;ReprPc&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 528518 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 528518 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-76.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 503763 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 503763 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-77.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 24549 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 24549 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-78.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 26974 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 26974 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-79.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 27125 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 27125 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-80.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;ReprPc&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 23932 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23932 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-81.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 22375 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 22375 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-82.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1482 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1482 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-83.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1523 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1523 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-84.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1543 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1543 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-85.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;ReprPc&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 17050 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 17050 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-86.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1013 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1013 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-87.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;ReprPc&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 10816 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 10816 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-88.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 579 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 579 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-89.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###ReprPCWk!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2500, 1500, 600, &quot;ReprPCWk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 522576 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 522576 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-90.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 499065 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 499065 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-91.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 23075 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23075 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-92.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 25317 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 25317 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-93.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 25457 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 25457 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-94.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2500, 1500, 600, &quot;ReprPCWk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 23261 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23261 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-95.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 21735 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 21735 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-96.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1386 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1386 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-97.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1424 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1424 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-98.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1445 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1445 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-99.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2500, 1500, 600, &quot;ReprPCWk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 16636 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 16636 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-100.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 947 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 947 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-101.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2500, 1500, 600, &quot;ReprPCWk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 10594 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 10594 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-102.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 539 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 539 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-103.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Quies!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 1000, 1500, 600, &quot;Quies&quot;, vpdims=c(0.4, 0.4, 0.4, 0.7), vpdims2=c(0.4, 0.4, 0.4, 0.7), vpdims3=c(0.3, 0.3, 0.4, 0.7), vpdims4=c(0.3, 0.3, 0.4, 0.7), vpdims5=c(0.3, 0.3, 0.4, 0.7))</code></pre>
<pre><code>Warning: Removed 437657 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 437657 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-104.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 424611 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 424611 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-105.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 9456 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 9456 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-106.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 10613 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 10613 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-107.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 10695 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 10695 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-108.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 1000, 1500, 600, &quot;Quies&quot;, vpdims=c(0.4, 0.4, 0.4, 0.7), vpdims2=c(0.4, 0.4, 0.4, 0.7), vpdims3=c(0.3, 0.3, 0.4, 0.7), vpdims4=c(0.3, 0.3, 0.4, 0.7), vpdims5=c(0.3, 0.3, 0.4, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 14099 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 14099 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-109.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 12610 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 12610 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-110.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 676 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 676 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-111.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 701 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 701 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-112.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 710 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 710 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-113.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 1000, 1500, 600, &quot;Quies&quot;, vpdims=c(0.4, 0.4, 0.4, 0.7), vpdims2=c(0.4, 0.4, 0.4, 0.7), vpdims3=c(0.3, 0.3, 0.4, 0.7), vpdims4=c(0.3, 0.3, 0.4, 0.7), vpdims5=c(0.3, 0.3, 0.4, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 11002 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 11002 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-114.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 484 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 484 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-115.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 1000, 1500, 600, &quot;Quies&quot;, vpdims=c(0.4, 0.4, 0.4, 0.7), vpdims2=c(0.4, 0.4, 0.4, 0.7), vpdims3=c(0.3, 0.3, 0.4, 0.7), vpdims4=c(0.3, 0.3, 0.4, 0.7), vpdims5=c(0.3, 0.3, 0.4, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 7695 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 7695 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-116.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 288 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 288 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-117.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###TssAFlnk!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssAFlnk&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 527076 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 527076 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-118.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 502951 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 502951 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-119.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 23870 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23870 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-120.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 26194 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 26194 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-121.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 26344 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 26344 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-122.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssAFlnk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 23793 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23793 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-123.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 22263 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 22263 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-124.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1419 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1419 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-125.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1452 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1452 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-126.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1470 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1470 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-127.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssAFlnk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 16957 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 16957 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-128.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 966 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 966 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-129.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TssAFlnk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 10789 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 10789 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-130.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 558 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 558 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-131.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###TxFlnk!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TxFlnk&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 537750 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 537750 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-132.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 511122 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 511122 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-133.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 26804 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 26804 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-134.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 29495 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 29495 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-135.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 29669 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 29669 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-136.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TxFlnk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 25055 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 25055 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-137.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 23436 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23436 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-138.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1638 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1638 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-139.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1678 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1678 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-140.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TxFlnk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 17810 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 17810 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-141.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1119 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1119 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-142.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;TxFlnk&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 11304 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 11304 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-143.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 646 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 646 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-144.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Tx!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2000, 2000, 600, &quot;Tx&quot;, vpdims=c(0.3, 0.3, 0.3, 0.3), vpdims2=c(0.3, 0.3, 0.3, 0.3), vpdims3=c(0.3, 0.3, 0.3, 0.3), vpdims4=c(0.3, 0.3, 0.3, 0.3), vpdims5=c(0.3, 0.3, 0.3, 0.3))</code></pre>
<pre><code>Warning: Removed 519622 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 519622 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-145.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 497506 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 497506 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-146.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 21537 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 21537 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-147.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 23609 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23609 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-148.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 23738 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23738 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-149.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2000, 2000, 600, &quot;Tx&quot;, vpdims=c(0.3, 0.3, 0.3, 0.3), vpdims2=c(0.3, 0.3, 0.3, 0.3), vpdims3=c(0.3, 0.3, 0.3, 0.3), vpdims4=c(0.3, 0.3, 0.3, 0.3), vpdims5=c(0.3, 0.3, 0.3, 0.3), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 22941 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 22941 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-150.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 21510 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 21510 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-151.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1245 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1245 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-152.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1266 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1266 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-153.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1284 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1284 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-154.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2000, 2000, 600, &quot;Tx&quot;, vpdims=c(0.3, 0.3, 0.3, 0.3), vpdims2=c(0.3, 0.3, 0.3, 0.3), vpdims3=c(0.3, 0.3, 0.3, 0.3), vpdims4=c(0.3, 0.3, 0.3, 0.3), vpdims5=c(0.3, 0.3, 0.3, 0.3), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 16333 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 16333 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-155.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 836 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 836 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-156.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2000, 2000, 600, &quot;Tx&quot;, vpdims=c(0.3, 0.3, 0.3, 0.3), vpdims2=c(0.3, 0.3, 0.3, 0.3), vpdims3=c(0.3, 0.3, 0.3, 0.3), vpdims4=c(0.3, 0.3, 0.3, 0.3), vpdims5=c(0.3, 0.3, 0.3, 0.3), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 10431 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 10431 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-157.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 471 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 471 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-158.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###TxWk!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2000, 2000, 600, &quot;TxWk&quot;, vpdims=c(0.3, 0.3, 0.3, 0.3), vpdims2=c(0.3, 0.3, 0.3, 0.3), vpdims3=c(0.3, 0.3, 0.3, 0.3), vpdims4=c(0.3, 0.3, 0.3, 0.3), vpdims5=c(0.3, 0.3, 0.3, 0.3))</code></pre>
<pre><code>Warning: Removed 495231 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 495231 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-159.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 477469 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 477469 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-160.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 16202 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 16202 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-161.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 17764 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 17764 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-162.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 17847 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 17847 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-163.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2000, 2000, 600, &quot;TxWk&quot;, vpdims=c(0.3, 0.3, 0.3, 0.3), vpdims2=c(0.3, 0.3, 0.3, 0.3), vpdims3=c(0.3, 0.3, 0.3, 0.3), vpdims4=c(0.3, 0.3, 0.3, 0.3), vpdims5=c(0.3, 0.3, 0.3, 0.3), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 20190 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 20190 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-164.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 18830 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 18830 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-165.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 919 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 919 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-166.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 932 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 932 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-167.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 944 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 944 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-168.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2000, 2000, 600, &quot;TxWk&quot;, vpdims=c(0.3, 0.3, 0.3, 0.3), vpdims2=c(0.3, 0.3, 0.3, 0.3), vpdims3=c(0.3, 0.3, 0.3, 0.3), vpdims4=c(0.3, 0.3, 0.3, 0.3), vpdims5=c(0.3, 0.3, 0.3, 0.3), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 14596 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 14596 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-169.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 605 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 605 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-170.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2000, 2000, 600, &quot;TxWk&quot;, vpdims=c(0.3, 0.3, 0.3, 0.3), vpdims2=c(0.3, 0.3, 0.3, 0.3), vpdims3=c(0.3, 0.3, 0.3, 0.3), vpdims4=c(0.3, 0.3, 0.3, 0.3), vpdims5=c(0.3, 0.3, 0.3, 0.3), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 9483 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 9483 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-171.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 350 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 350 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-172.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###EnhG!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;EnhG&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 536394 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 536394 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-173.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 509992 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 509992 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-174.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 26505 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 26505 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-175.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 29157 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 29157 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-176.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 29328 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 29328 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-177.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;EnhG&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 24898 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 24898 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-178.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 23276 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23276 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-179.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1620 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1620 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-180.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1656 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1656 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-181.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1680 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1680 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-182.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;EnhG&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 17687 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 17687 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-183.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1098 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1098 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-184.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;EnhG&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 11234 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 11234 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-185.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 627 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 627 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-186.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Enh!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;Enh&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 495072 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 495072 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-187.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 475503 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 475503 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-188.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 18140 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 18140 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-189.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 20007 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 20007 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-190.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 20124 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 20124 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-191.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;Enh&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 20455 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 20455 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-192.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 18943 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 18943 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-193.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1134 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1134 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-194.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1164 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1164 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-195.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1178 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1178 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-196.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;Enh&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 14931 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 14931 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-197.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 770 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 770 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-198.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;Enh&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 9732 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 9732 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-199.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 448 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 448 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-200.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###ZNF_Rpts!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;ZNF_Rpts&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 536857 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 536857 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-201.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 510289 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 510289 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-202.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 26684 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 26684 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-203.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 29385 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 29385 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-204.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 29552 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 29552 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-205.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;ZNF_Rpts&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 24962 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 24962 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-206.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 23325 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23325 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-207.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1641 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1641 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-208.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1683 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1683 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-209.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1708 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1708 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-210.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;ZNF_Rpts&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 17732 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 17732 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-211.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1125 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1125 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-212.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 6200, 2500, 1000, 400, &quot;ZNF_Rpts&quot;, vpdims=c(0.3, 0.3, 0.52, 0.7), vpdims2=c(0.3, 0.3, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 11252 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 11252 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-213.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 644 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 644 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-214.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Het!
chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2500, 1500, 600, &quot;Het&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7))</code></pre>
<pre><code>Warning: Removed 524373 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 524373 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-215.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 499281 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 499281 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-216.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 24691 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 24691 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-217.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 27207 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 27207 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-218.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 27370 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 27370 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-219.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2500, 1500, 600, &quot;Het&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.1) #At 10% FDR</code></pre>
<pre><code>Warning: Removed 23493 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 23493 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-220.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 21863 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 21863 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-221.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1538 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1538 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-222.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1576 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1576 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-223.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1600 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1600 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-224.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2500, 1500, 600, &quot;Het&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.05) #At 5% FDR</code></pre>
<pre><code>Warning: Removed 16757 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 16757 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-225.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 1051 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 1051 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-226.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chromHMMplot(chrom.annos, nonproms.DC.df, proms.DC, proms.DE, proms.differential, 10000, 2500, 1500, 600, &quot;Het&quot;, vpdims=c(0.4, 0.4, 0.52, 0.7), vpdims2=c(0.4, 0.4, 0.5, 0.7), vpdims3=c(0.3, 0.3, 0.7, 0.7), vpdims4=c(0.3, 0.3, 0.7, 0.7), vpdims5=c(0.3, 0.3, 0.5, 0.7), FDR = 0.01) #At 1% FDR</code></pre>
<pre><code>Warning: Removed 10692 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 10692 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-227.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 594 rows containing non-finite values (stat_density).</code></pre>
<pre><code>Warning: Removed 594 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/chromHMM%20Visualization-228.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="quantitative-histone-mark-enrichments" class="section level3">
<h3>Quantitative histone mark enrichments</h3>
<p>Now, I look at the quantitative enrichment of histone mark peaks from other studies on iPSCs, examining the overlap of these peaks with different classes of Hi-C significant bins.</p>
<pre class="r"><code>#####Examination of CTCF interruptions, 2x2 tables!
table(full.annotations$CTCF.interrupt, full.annotations$true.promoter, dnn = c(&quot;CTCF Interruption?&quot; , &quot;Promoter?&quot;))</code></pre>
<pre><code>                  Promoter?
CTCF Interruption?     no    yes
               no   46040  10099
               yes 209936  25995</code></pre>
<pre class="r"><code>chisq.test(table(full.annotations$CTCF.interrupt, full.annotations$true.promoter, dnn = c(&quot;CTCF Interruption?&quot; , &quot;Promoter?&quot;)))</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  table(full.annotations$CTCF.interrupt, full.annotations$true.promoter,     dnn = c(&quot;CTCF Interruption?&quot;, &quot;Promoter?&quot;))
X-squared = 2034.2, df = 1, p-value &lt; 2.2e-16</code></pre>
<pre class="r"><code>table(filter(full.annotations, true.promoter==&quot;no&quot;)$CTCF.interrupt, filter(full.annotations, true.promoter==&quot;no&quot;)$DC, dnn=c(&quot;CTCF Interruption?&quot;, &quot;DC?&quot;))</code></pre>
<pre><code>                  DC?
CTCF Interruption?     no    yes
               no   44275   1765
               yes 203281   6655</code></pre>
<pre class="r"><code>chisq.test(table(filter(full.annotations, true.promoter==&quot;no&quot;)$CTCF.interrupt, filter(full.annotations, true.promoter==&quot;no&quot;)$DC))</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  table(filter(full.annotations, true.promoter == &quot;no&quot;)$CTCF.interrupt,     filter(full.annotations, true.promoter == &quot;no&quot;)$DC)
X-squared = 52.063, df = 1, p-value = 5.376e-13</code></pre>
<pre class="r"><code>table(filter(full.annotations, true.promoter==&quot;yes&quot;)$CTCF.interrupt, filter(full.annotations, true.promoter==&quot;yes&quot;)$DC, dnn=c(&quot;CTCF Interruption?&quot;, &quot;DC?&quot;))</code></pre>
<pre><code>                  DC?
CTCF Interruption?    no   yes
               no   9747   352
               yes 25106   889</code></pre>
<pre class="r"><code>chisq.test(table(filter(full.annotations, true.promoter==&quot;yes&quot;)$CTCF.interrupt, filter(full.annotations, true.promoter==&quot;yes&quot;)$DC))</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  table(filter(full.annotations, true.promoter == &quot;yes&quot;)$CTCF.interrupt,     filter(full.annotations, true.promoter == &quot;yes&quot;)$DC)
X-squared = 0.075565, df = 1, p-value = 0.7834</code></pre>
<pre class="r"><code>table(filter(full.annotations, true.promoter==&quot;yes&quot;)$CTCF.interrupt, filter(full.annotations, true.promoter==&quot;yes&quot;)$DE, dnn=c(&quot;CTCF Interruption?&quot;, &quot;DE?&quot;))</code></pre>
<pre><code>                  DE?
CTCF Interruption?    no   yes
               no   7940  2159
               yes 20139  5856</code></pre>
<pre class="r"><code>chisq.test(table(filter(full.annotations, true.promoter==&quot;yes&quot;)$CTCF.interrupt, filter(full.annotations, true.promoter==&quot;yes&quot;)$DE, dnn=c(&quot;CTCF Interruption?&quot;, &quot;DE?&quot;)))</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  table(filter(full.annotations, true.promoter == &quot;yes&quot;)$CTCF.interrupt,     filter(full.annotations, true.promoter == &quot;yes&quot;)$DE, dnn = c(&quot;CTCF Interruption?&quot;,         &quot;DE?&quot;))
X-squared = 5.4927, df = 1, p-value = 0.0191</code></pre>
<pre class="r"><code>table(filter(full.annotations, true.promoter==&quot;yes&quot;)$CTCF.interrupt, filter(full.annotations, true.promoter==&quot;yes&quot;)$differential==&quot;both&quot;, dnn=c(&quot;CTCF Interruption?&quot;, &quot;DC &amp; DE?&quot;))</code></pre>
<pre><code>                  DC &amp; DE?
CTCF Interruption? FALSE  TRUE
               no  10031    68
               yes 25798   197</code></pre>
<pre class="r"><code>chisq.test(table(filter(full.annotations, true.promoter==&quot;yes&quot;)$CTCF.interrupt, filter(full.annotations, true.promoter==&quot;yes&quot;)$differential==&quot;both&quot;, dnn=c(&quot;CTCF Interruption?&quot;, &quot;DE?&quot;)))</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  table(filter(full.annotations, true.promoter == &quot;yes&quot;)$CTCF.interrupt,     filter(full.annotations, true.promoter == &quot;yes&quot;)$differential ==         &quot;both&quot;, dnn = c(&quot;CTCF Interruption?&quot;, &quot;DE?&quot;))
X-squared = 0.60142, df = 1, p-value = 0.438</code></pre>
<pre class="r"><code>#######Make a doubled-down data frame for single columns of quantitative enrichments. Really, will need to do differential status-level de-duplication here first, then recombine those to make this simple with ggplot2. Create one huge for looking at differential statuses, and one big one separating out promoters/non promoters (chrom.annos). Also need polarizing ones for DC/not and DE/not, without exclusivity in the case of both. chrom.annos is what was created in the previous chunk for separating out promoters/non promoters. Now, undertake the same process to create the other necessary dfs:

#First the non-promoter DF, looking at DC/not DC. Remember the promoter/not promoter was created in the first chunk (chrom.annos).
DC.non.proms &lt;- filter(full.annotations, true.promoter==&quot;no&quot;, DC==&quot;yes&quot;) %&gt;% deduplicator(.)
non.DC.non.proms &lt;- filter(full.annotations, true.promoter==&quot;no&quot;, DC==&quot;no&quot;) %&gt;% deduplicator(.)

#After deduplicating, combine so that there&#39;s one column per histone mark I&#39;m interested in, and a column showing DC status.
nonproms &lt;- data.frame(H1=c(DC.non.proms$H1, DC.non.proms$H2, non.DC.non.proms$H1, non.DC.non.proms$H2), sp_BH_pval=c(rep(DC.non.proms$sp_BH_pval, 2), rep(non.DC.non.proms$sp_BH_pval, 2)), sp_beta=c(rep(DC.non.proms$sp_beta, 2), rep(non.DC.non.proms$sp_beta, 2)), HMM=c(as.character(DC.non.proms$bin1.HMM), as.character(DC.non.proms$bin2.HMM), as.character(non.DC.non.proms$bin1.HMM), as.character(non.DC.non.proms$bin2.HMM)), DHS=c(DC.non.proms$bin1.DHSval, DC.non.proms$bin2.DHSval, non.DC.non.proms$bin1.DHSval, non.DC.non.proms$bin2.DHSval), Hac=c(DC.non.proms$bin1.Hacval, DC.non.proms$bin2.Hacval, non.DC.non.proms$bin1.Hacval, non.DC.non.proms$bin2.Hacval), Hme1=c(DC.non.proms$bin1.Hme1val, DC.non.proms$bin2.Hme1val, non.DC.non.proms$bin1.Hme1val, non.DC.non.proms$bin2.Hme1val), Hme3=c(DC.non.proms$bin1.Hme3val, DC.non.proms$bin2.Hme3val, non.DC.non.proms$bin1.Hme3val, non.DC.non.proms$bin2.Hme3val), HAR=c(DC.non.proms$bin1.HARval, DC.non.proms$bin2.HARval, non.DC.non.proms$bin1.HARval, non.DC.non.proms$bin2.HARval), DC=c(rep(DC.non.proms$DC, 2), rep(non.DC.non.proms$DC, 2)))

#Now, looking within promoter-contacting Hi-C bins, at which are DC and which are not. Should be fairly simple, and in these analyses, we are focusing on the bins making contact with a promoter (promoter.contacts df has excluded the promoters themselves here). In these cases, after the deduplication is done, it&#39;s as simple as rbinding the two filtered dfs, since they have the same number of columns and only a single value for each mark. Dedup function doesn&#39;t work here beacause only a single column; simpler case.
DC.proms &lt;- filter(promoter.contacts, DC==&quot;yes&quot;)
DC.proms &lt;- DC.proms[!duplicated(DC.proms$H2),]
nonDC.proms &lt;- filter(promoter.contacts, DC==&quot;no&quot;)
nonDC.proms &lt;- nonDC.proms[!duplicated(nonDC.proms$H2),]
proms.DC &lt;- rbind(DC.proms, nonDC.proms)
proms.DC &lt;- proms.DC[order(proms.DC$sp_BH_pval),] #Should be complete!

DE.proms &lt;- filter(promoter.contacts, DE==&quot;yes&quot;)
DE.proms &lt;- DE.proms[!duplicated(DE.proms$H2),]
nonDE.proms &lt;- filter(promoter.contacts, DE==&quot;no&quot;)
nonDE.proms &lt;- nonDE.proms[!duplicated(nonDE.proms$H2),]
proms.DE &lt;- rbind(DE.proms, nonDE.proms)
proms.DE &lt;- proms.DE[order(proms.DE$sp_BH_pval),]

DE.only.proms &lt;- filter(promoter.contacts, differential==&quot;expression&quot;)
DC.only.proms &lt;- filter(promoter.contacts, differential==&quot;contact&quot;)
DEDC.proms &lt;- filter(promoter.contacts, differential==&quot;both&quot;)
neither.proms &lt;- filter(promoter.contacts, differential==&quot;no&quot;)
DE.only.proms &lt;- DE.only.proms[!duplicated(DE.only.proms$H2),]
DC.only.proms &lt;- DC.only.proms[!duplicated(DC.only.proms$H2),]
DEDC.proms &lt;- DEDC.proms[!duplicated(DEDC.proms$H2),]
neither.proms &lt;- neither.proms[!duplicated(neither.proms$H2),]
proms.differential &lt;- rbind(DE.only.proms, DC.only.proms, DEDC.proms, neither.proms)
proms.differential &lt;- proms.differential[order(proms.differential$sp_BH_pval),]
colnames(proms.differential) &lt;- colnames(proms.DE) &lt;- colnames(proms.DC) &lt;- colnames(chrom.annos)[1:29]

#Old way of obtaining these, without removing duplicates, without focusing only on the contacts that hit promoters--very naive.
#dub.anno &lt;- data.frame(H1=c(full.annotations$H1, full.annotations$H2), sp_BH_pval=rep(full.annotations$sp_BH_pval, 2), sp_beta=rep(full.annotations$sp_beta, 2), HMM=c(as.character(full.annotations$bin1.HMM), as.character(full.annotations$bin2.HMM)), DHS=c(full.annotations$bin1.DHSval, full.annotations$bin2.DHSval), Hac=c(full.annotations$bin1.Hacval, full.annotations$bin2.Hacval), Hme1=c(full.annotations$bin1.Hme1val, full.annotations$bin2.Hme1val), Hme3=c(full.annotations$bin1.Hme3val, full.annotations$bin2.Hme3val), HAR=c(full.annotations$bin1.HARval, full.annotations$bin2.HARval), pairID=rep(full.annotations$pairID, 2), CTCF.interrupt=rep(full.annotations$CTCF.interrupt, 2), DC=rep(full.annotations$DC, 2), DE=rep(full.annotations$DE, 2), true.promoter=rep(full.annotations$true.promoter, 2), differential=rep(full.annotations$differential, 2))

###############################Plots of enrichment for different classes of hits and different marks. Seems like it would be nice to create a function for these but really it&#39;ll lead to a huge hassle with having to specify different x and y-axis limits for different marks in different sets, and drawing the segments...since I&#39;ve already done it out this way, might as well keep it.
##########DHS plots!!!
all.DHS &lt;- ggplot(data=chrom.annos) + geom_density(aes(x=DHSval, group=true.promoter, color=true.promoter, y=..scaled..)) + ggtitle(&quot;Enrichment of DHS Overlapping Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and DHS peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Promoter Contact?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 1000))
box.all.DHS &lt;- ggplot(data=chrom.annos) + geom_violin(aes(x=true.promoter, y=DHSval, group=true.promoter, color=true.promoter), draw_quantiles =c(0.25, 0.5, 0.75) ) + scale_color_manual(values=c(&quot;Red&quot;, &quot;Blue&quot;), labels=c(&quot;No&quot;, &quot;Yes&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 500)) + annotate(&quot;text&quot;, x=1.5, y=300, label=&quot;&lt; 2 e -16&quot;, size=3) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=270, yend=270)
print(all.DHS)</code></pre>
<pre><code>Warning: Removed 403531 rows containing non-finite values (stat_density).</code></pre>
<pre class="r"><code>vp &lt;- grid::viewport(width=0.4, height=0.4, x=0.5, y=0.7)
print(box.all.DHS, vp=vp)</code></pre>
<pre><code>Warning: Removed 403531 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(chrom.annos$DHS[which(chrom.annos$true.promoter==&quot;yes&quot;)], chrom.annos$DHS[which(chrom.annos$true.promoter==&quot;no&quot;)])</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  chrom.annos$DHS[which(chrom.annos$true.promoter == &quot;yes&quot;)] and chrom.annos$DHS[which(chrom.annos$true.promoter == &quot;no&quot;)]
t = 41.333, df = 34308, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 29.84468 32.81612
sample estimates:
mean of x mean of y 
  70.4425   39.1121 </code></pre>
<pre class="r"><code>noprom.DHS &lt;- ggplot(data=nonproms) + geom_density(aes(x=DHS, group=DC, color=DC, y=..scaled..)) + ggtitle(&quot;Enrichment of DHS overlapping non-promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and DHS peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Neither&quot;, &quot;Contact&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 1000))
box.noprom.DHS &lt;- ggplot(data=nonproms) + geom_violin(aes(x=DC, y=DHS, group=DC, color=DC), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Neither&quot;, &quot;Contact&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 500)) + annotate(&quot;text&quot;, x=1.5, y=300, label=&quot;&lt; 2 e -16&quot;, size=3) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=270, yend=270)
print(noprom.DHS)</code></pre>
<pre><code>Warning: Removed 398499 rows containing non-finite values (stat_density).</code></pre>
<pre class="r"><code>print(box.noprom.DHS, vp=vp)</code></pre>
<pre><code>Warning: Removed 398499 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(nonproms, DC==&quot;no&quot;)$DHS, filter(nonproms, DC==&quot;yes&quot;)$DHS)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(nonproms, DC == &quot;no&quot;)$DHS and filter(nonproms, DC == &quot;yes&quot;)$DHS
t = -12.393, df = 9763.3, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -15.27356 -11.10168
sample estimates:
mean of x mean of y 
 38.70347  51.89109 </code></pre>
<pre class="r"><code>prom.DHS &lt;- ggplot(data=proms.differential) + geom_density(aes(x=DHSval, group=differential, color=differential, y=..scaled..)) + ggtitle(&quot;Enrichment of DHS overlapping Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and DHS peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;)) + coord_cartesian(xlim=c(0, 1000))
box.prom.DHS &lt;- ggplot(data=proms.differential) + geom_violin(aes(x=differential, y=DHSval, group=differential, color=differential), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 600)) + annotate(&quot;text&quot;, x=3.5, y=300, label=&quot;5 e -12&quot;, size=3) + annotate(&quot;text&quot;, x=2.5, y=300, label=&quot;0.2&quot;, size=3) + annotate(&quot;text&quot;, x=1.5, y=400, label=&quot;4 e -2&quot;, size=3) + annotate(&quot;text&quot;, label=&quot;2 e -3&quot;, x=2, y=480, size=3) + annotate(&quot;segment&quot;, x=1.1, y=370, xend=1.9, yend=370) + annotate(&quot;segment&quot;, x=2.1, y=270, xend=2.9, yend=270) + annotate(&quot;segment&quot;, x=3.1, y=270, xend=3.9, yend=270) + annotate(&quot;segment&quot;, x=1.1, y=450, xend=2.9, yend=450) + annotate(&quot;text&quot;, x=2.5, y=550, size=3, label=&quot;1 e -5&quot;) + annotate(&quot;segment&quot;, x=1.1, y=520, xend=3.9, yend=520) + annotate(&quot;text&quot;, x=3, y=400, label=&quot;3 e -4&quot;, size=3) + annotate(&quot;segment&quot;, x=2.1, xend=3.9, y=370, yend=370)
vp &lt;- grid::viewport(width=0.4, height=0.4, x=0.5, y=0.7)
print(prom.DHS)
print(box.prom.DHS, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.differential, differential==&quot;contact&quot;)$DHSval, filter(proms.differential, differential==&quot;expression&quot;)$DHSval)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.differential, differential == &quot;contact&quot;)$DHSval and filter(proms.differential, differential == &quot;expression&quot;)$DHSval
t = 1.1969, df = 1031.7, p-value = 0.2316
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -3.670494 15.150933
sample estimates:
mean of x mean of y 
 86.71246  80.97224 </code></pre>
<pre class="r"><code>DE.DHS &lt;- ggplot(data=proms.DE) + geom_density(aes(x=DHSval, group=DE, color=DE, y=..scaled..)) + ggtitle(&quot;Enrichment of DHS overlapping DE Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and DHS peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DE?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 1000))
boxDE.DHS &lt;- ggplot(data=proms.DE) + geom_violin(aes(x=DE, y=DHSval, group=DE, color=DE), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + coord_cartesian(ylim=c(0, 1000)) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;text&quot;, x=1.5, y=400, label=&quot;6 e -13&quot;, size=3) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=350, yend=350)
vp &lt;- grid::viewport(width=0.4, height=0.4, x=0.6, y=0.5)
print(DE.DHS)
print(boxDE.DHS, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.DE, DE==&quot;yes&quot;)$DHSval, filter(proms.DE, DE==&quot;no&quot;)$DHSval)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.DE, DE == &quot;yes&quot;)$DHSval and filter(proms.DE, DE == &quot;no&quot;)$DHSval
t = 7.2042, df = 13085, p-value = 6.157e-13
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
  8.471607 14.804688
sample estimates:
mean of x mean of y 
 81.72250  70.08435 </code></pre>
<pre class="r"><code>DC.DHS &lt;- ggplot(data=proms.DC) + geom_density(aes(x=DHSval, group=DC, color=DC, y=..scaled..)) + ggtitle(&quot;Enrichment of DHS overlapping DC Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and DHS peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DC?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 1000))
boxDC.DHS &lt;- ggplot(data=proms.DC) + geom_violin(aes(x=DC, y=DHSval, group=DC, color=DC), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + coord_cartesian(ylim=c(0, 1000)) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;text&quot;, label=&quot;2 e -6&quot;, size=3, x=1.5, y=400) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=350, yend=350)#+ geom_text(aes(x=1.5, y=2000, label=&quot;0.2901&quot;), show.legend=FALSE, size=3)
print(DC.DHS)
print(boxDC.DHS, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.DC, DC==&quot;yes&quot;)$DHSval, filter(proms.DC, DC==&quot;no&quot;)$DHSval)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.DC, DC == &quot;yes&quot;)$DHSval and filter(proms.DC, DC == &quot;no&quot;)$DHSval
t = 4.832, df = 1150.1, p-value = 1.534e-06
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 12.02683 28.47069
sample estimates:
mean of x mean of y 
 90.16236  69.91360 </code></pre>
<pre class="r"><code>##############H3K27Ac plots!!!
all.Hac &lt;- ggplot(data=chrom.annos) + geom_density(aes(x=Hacval, group=true.promoter, color=true.promoter, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K27ac Overlapping Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K27ac peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Promoter Contact?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 2500))
box.all.Hac &lt;- ggplot(data=chrom.annos) + geom_violin(aes(x=true.promoter, y=Hacval, group=true.promoter, color=true.promoter)) + scale_color_manual(values=c(&quot;Red&quot;, &quot;Blue&quot;), labels=c(&quot;No&quot;, &quot;Yes&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 1000)) + annotate(&quot;text&quot;, label=&quot;&lt; 2 e -16&quot;, x=1.5, y=400, size=3) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=350, yend=350)# Throws an error when draw_quantiles is added to geom_violin, because of NAs that exist in the data so the quantile call fails? Or...?
vp &lt;- grid::viewport(width=0.4, height=0.4, x=0.5, y=0.5)
print(all.Hac)</code></pre>
<pre><code>Warning: Removed 403531 rows containing non-finite values (stat_density).</code></pre>
<pre class="r"><code>print(box.all.Hac, vp=vp)</code></pre>
<pre><code>Warning: Removed 403531 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(chrom.annos, true.promoter==&quot;yes&quot;)$Hacval, filter(chrom.annos, true.promoter==&quot;no&quot;)$Hacval)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(chrom.annos, true.promoter == &quot;yes&quot;)$Hacval and filter(chrom.annos, true.promoter == &quot;no&quot;)$Hacval
t = 29.829, df = 31989, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 79.32880 90.48739
sample estimates:
mean of x mean of y 
136.71631  51.80821 </code></pre>
<pre class="r"><code>noprom.Hac &lt;- ggplot(data=nonproms) + geom_density(aes(x=Hac, group=DC, color=DC, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K27ac overlapping non-promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K27ac peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Neither&quot;, &quot;Contact&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 2000))
box.noprom.Hac &lt;- ggplot(data=nonproms) + geom_violin(aes(x=DC, y=Hac, group=DC, color=DC)) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Neither&quot;, &quot;Contact&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 1000)) + annotate(&quot;text&quot;, label=&quot;1 e -4&quot;, x=1.5, y=400, size=3) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=350, yend=350) # Once again throws an error in draw_quantiles due to the NAs in the data (I believe).
print(noprom.Hac)</code></pre>
<pre><code>Warning: Removed 398499 rows containing non-finite values (stat_density).</code></pre>
<pre class="r"><code>print(box.noprom.Hac, vp=vp)</code></pre>
<pre><code>Warning: Removed 398499 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(nonproms, DC==&quot;yes&quot;)$Hac, filter(nonproms, DC==&quot;no&quot;)$Hac)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(nonproms, DC == &quot;yes&quot;)$Hac and filter(nonproms, DC == &quot;no&quot;)$Hac
t = 3.8524, df = 9905.1, p-value = 0.0001177
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
  6.084308 18.690560
sample estimates:
mean of x mean of y 
 63.64352  51.25608 </code></pre>
<pre class="r"><code>prom.Hac &lt;- ggplot(data=proms.differential) + geom_density(aes(x=Hacval, group=differential, color=differential, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K27ac overlapping Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K27ac peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;)) + coord_cartesian(xlim=c(0, 3500))
box.prom.Hac &lt;- ggplot(data=proms.differential) + geom_violin(aes(x=differential, y=Hacval, group=differential, color=differential), draw_quantiles = c(0.25, 0.5, 0.75))+ scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank())+ coord_cartesian(ylim=c(0, 1000)) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=430, yend=430) + annotate(&quot;segment&quot;, x=1.1, xend=2.9, y=700, yend=700) + annotate(&quot;segment&quot;, x=1.1, xend=3.9, y=880, yend=880) + annotate(&quot;segment&quot;, x=2.1, xend=2.9, y=430, yend=430) + annotate(&quot;segment&quot;, x=2.1, xend=3.9, y=560, yend=560) + annotate(&quot;segment&quot;, x=3.1, xend=3.9, y=430, yend=430) + annotate(&quot;text&quot;, label=&quot;0.4&quot;, size=3, x=1.5, y=480) + annotate(&quot;text&quot;, label=&quot;0.1&quot;, size=3, x=2.5, y=480) + annotate(&quot;text&quot;, label=&quot;1 e -3&quot;, size=3, x=3.5, y=480) + annotate(&quot;text&quot;, label=&quot;2 e -6&quot;, size=3, x=3, y=610) + annotate(&quot;text&quot;, label=&quot;0.08&quot;, size=3, x=2, y=750) + annotate(&quot;text&quot;, label=&quot;8 e -3&quot;, size=3, x=2.5, y=930)
vp &lt;- grid::viewport(width=0.4, height=0.4, x=0.5, y=0.5)
print(prom.Hac)
print(box.prom.Hac, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.differential, differential==&quot;contact&quot;)$Hacval, filter(proms.differential, differential==&quot;no&quot;)$Hacval)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.differential, differential == &quot;contact&quot;)$Hacval and filter(proms.differential, differential == &quot;no&quot;)$Hacval
t = 3.2241, df = 904.42, p-value = 0.001309
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 23.43100 96.33586
sample estimates:
mean of x mean of y 
 191.7055  131.8220 </code></pre>
<pre class="r"><code>DE.Hac &lt;- ggplot(data=proms.DE) + geom_density(aes(x=Hacval, group=DE, color=DE, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K27ac overlapping DE Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K27ac peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DE?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 3000))
boxDE.Hac &lt;- ggplot(data=proms.DE) + geom_violin(aes(x=DE, y=Hacval, group=DE, color=DE), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + coord_cartesian(ylim=c(0, 1000)) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=250, yend=250) + annotate(&quot;text&quot;, x=1.5, y=300, label=&quot;1 e -6&quot;, size=3)
print(DE.Hac)
print(boxDE.Hac, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.DE, DE==&quot;yes&quot;)$Hacval, filter(proms.DE, DE==&quot;no&quot;)$Hacval)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.DE, DE == &quot;yes&quot;)$Hacval and filter(proms.DE, DE == &quot;no&quot;)$Hacval
t = 4.8965, df = 12715, p-value = 9.876e-07
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 18.41116 42.99184
sample estimates:
mean of x mean of y 
 164.4488  133.7473 </code></pre>
<pre class="r"><code>DC.Hac &lt;- ggplot(data=proms.DC) + geom_density(aes(x=Hacval, group=DC, color=DC, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K27ac overlapping DC Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K27ac peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DC?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 2500))
boxDC.Hac &lt;- ggplot(data=proms.DC) + geom_violin(aes(x=DC, y=Hacval, group=DC, color=DC), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + coord_cartesian(ylim=c(0, 1000)) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=400, yend=400) + annotate(&quot;text&quot;, x=1.5, y=450, label=&quot;3 e -4&quot;, size=3)
print(DC.Hac)
print(boxDC.Hac, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.DC, DC==&quot;yes&quot;)$Hacval, filter(proms.DC, DC==&quot;no&quot;)$Hacval)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.DC, DC == &quot;yes&quot;)$Hacval and filter(proms.DC, DC == &quot;no&quot;)$Hacval
t = 3.6499, df = 1142.1, p-value = 0.0002742
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 28.57184 94.99826
sample estimates:
mean of x mean of y 
 196.3976  134.6126 </code></pre>
<pre class="r"><code>######H3K4Me1 plots!!!
all.Hme1 &lt;- ggplot(data=chrom.annos) + geom_density(aes(x=Hme1val, group=true.promoter, color=true.promoter, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K4me1 Overlapping Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K4me1 peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Promoter Contact?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 3000))
box.all.Hme1 &lt;- ggplot(data=chrom.annos) + geom_violin(aes(x=true.promoter, y=Hme1val, group=true.promoter, color=true.promoter), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(values=c(&quot;Red&quot;, &quot;Blue&quot;), labels=c(&quot;No&quot;, &quot;Yes&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 3000)) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=1250, yend=1250) + annotate(&quot;text&quot;, x=1.5, y=1450, label=&quot;&lt; 2 e -16&quot;, size=3)
print(all.Hme1)</code></pre>
<pre><code>Warning: Removed 403531 rows containing non-finite values (stat_density).</code></pre>
<pre class="r"><code>print(box.all.Hme1, vp=vp)</code></pre>
<pre><code>Warning: Removed 403531 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(chrom.annos, true.promoter==&quot;yes&quot;)$Hme1val, filter(chrom.annos, true.promoter==&quot;no&quot;)$Hme1val)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(chrom.annos, true.promoter == &quot;yes&quot;)$Hme1val and filter(chrom.annos, true.promoter == &quot;no&quot;)$Hme1val
t = 13.243, df = 39688, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 72.12179 97.17830
sample estimates:
mean of x mean of y 
 514.6728  430.0227 </code></pre>
<pre class="r"><code>prom.Hme1 &lt;- ggplot(data=proms.differential) + geom_density(aes(x=Hme1val, group=differential, color=differential, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K4me1 overlapping Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K4me1 peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;)) + coord_cartesian(xlim=c(0, 6000))
box.prom.Hme1 &lt;- ggplot(data=proms.differential) + geom_violin(aes(x=differential, y=Hme1val, group=differential, color=differential), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 6000)) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=1500, yend=1500) + annotate(&quot;segment&quot;, x=2.1, xend=2.9, y=1500, yend=1500) + annotate(&quot;segment&quot;, x=3.1, xend=3.9, y=1500, yend=1500) + annotate(&quot;segment&quot;, x=1.1, xend=2.9, y=3900, yend=3900) + annotate(&quot;segment&quot;, x=1.1, xend=3.9, y=5100, yend=5100) + annotate(&quot;segment&quot;, x=2.1, xend=3.9, y=2700, yend=2700) + annotate(&quot;text&quot;, x=1.5, y=1800, size=3, label=&quot;0.03&quot;) + annotate(&quot;text&quot;, x=2.5, y=1800, size=3, label=&quot;0.6&quot;) + annotate(&quot;text&quot;, x=3.5, y=1800, size=3, label=&quot;0.5&quot;) + annotate(&quot;text&quot;, x=3, y=3000, size=3, label=&quot;0.5&quot;) + annotate(&quot;text&quot;, x=2.5, y=5400, size=3, label=&quot;0.04&quot;) + annotate(&quot;text&quot;, x=2, y=4200, size=3, label=&quot;0.03&quot;)
vp &lt;- grid::viewport(width=0.4, height=0.4, x=0.5, y=0.5)
print(prom.Hme1)
print(box.prom.Hme1, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.differential, differential==&quot;both&quot;)$Hme1val, filter(proms.differential, differential==&quot;contact&quot;)$Hme1val)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.differential, differential == &quot;both&quot;)$Hme1val and filter(proms.differential, differential == &quot;contact&quot;)$Hme1val
t = 2.1512, df = 424.43, p-value = 0.03202
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
  13.61787 301.96398
sample estimates:
mean of x mean of y 
 649.8282  492.0373 </code></pre>
<pre class="r"><code>noprom.Hme1 &lt;- ggplot(data=nonproms) + geom_density(aes(x=Hme1, group=DC, color=DC, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K4me1 overlapping non-promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K4me1 peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Neither&quot;, &quot;Contact&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 4000))
box.noprom.Hme1 &lt;- ggplot(data=nonproms) + geom_violin(aes(x=DC, y=Hme1, group=DC, color=DC), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Neither&quot;, &quot;Contact&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 2200)) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=750, yend=750) + annotate(&quot;text&quot;, x=1.5, y=860, size=3, label=&quot;0.05&quot;)
print(noprom.Hme1)</code></pre>
<pre><code>Warning: Removed 398499 rows containing non-finite values (stat_density).</code></pre>
<pre class="r"><code>print(box.noprom.Hme1, vp=vp)</code></pre>
<pre><code>Warning: Removed 398499 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(nonproms, DC==&quot;no&quot;)$Hme1, filter(nonproms, DC==&quot;yes&quot;)$Hme1)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(nonproms, DC == &quot;no&quot;)$Hme1 and filter(nonproms, DC == &quot;yes&quot;)$Hme1
t = 1.9238, df = 10256, p-value = 0.0544
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3492478 37.3155341
sample estimates:
mean of x mean of y 
 430.5787  412.0956 </code></pre>
<pre class="r"><code>DE.Hme1 &lt;- ggplot(data=proms.DE) + geom_density(aes(x=Hme1val, group=DE, color=DE, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K4me1 overlapping DE Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K4me1 peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DE?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 4000))
boxDE.Hme1 &lt;- ggplot(data=proms.DE) + geom_violin(aes(x=DE, y=Hme1val, group=DE, color=DE), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + coord_cartesian(ylim=c(0, 2200)) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;text&quot;, x=1.5, y=1360, label=&quot;0.9&quot;, size=3) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=1250, yend=1250)
print(DE.Hme1)
print(boxDE.Hme1, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.DE, DE==&quot;yes&quot;)$Hme1val, filter(proms.DE, DE==&quot;no&quot;)$Hme1val)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.DE, DE == &quot;yes&quot;)$Hme1val and filter(proms.DE, DE == &quot;no&quot;)$Hme1val
t = -0.13642, df = 14318, p-value = 0.8915
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -25.55580  22.23005
sample estimates:
mean of x mean of y 
 513.1146  514.7775 </code></pre>
<pre class="r"><code>DC.Hme1 &lt;- ggplot(data=proms.DC) + geom_density(aes(x=Hme1val, group=DC, color=DC, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K4me1 overlapping DC Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K4me1 peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DC?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 5000))
boxDC.Hme1 &lt;- ggplot(data=proms.DC) + geom_violin(aes(x=DC, y=Hme1val, group=DC, color=DC), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + coord_cartesian(ylim=c(0, 2200)) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=1200, yend=1200) + annotate(&quot;text&quot;, x=1.5, y=1310, label=&quot;0.7&quot;, size=3)
print(DC.Hme1)
print(boxDC.Hme1, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-15.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.DC, DC==&quot;yes&quot;)$Hme1val, filter(proms.DC, DC==&quot;no&quot;)$Hme1val)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.DC, DC == &quot;yes&quot;)$Hme1val and filter(proms.DC, DC == &quot;no&quot;)$Hme1val
t = 0.40534, df = 1169.1, p-value = 0.6853
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -47.10186  71.63179
sample estimates:
mean of x mean of y 
 525.2002  512.9352 </code></pre>
<pre class="r"><code>####H3K4Me3 plots!!!
all.Hme3 &lt;- ggplot(data=chrom.annos) + geom_density(aes(x=Hme3val, group=true.promoter, color=true.promoter, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K4me3 Overlapping Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K4me3 peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Promoter Contact?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 6000))
box.all.Hme3 &lt;- ggplot(data=chrom.annos) + geom_violin(aes(x=true.promoter, y=Hme3val, group=true.promoter, color=true.promoter), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(values=c(&quot;Red&quot;, &quot;Blue&quot;), labels=c(&quot;No&quot;, &quot;Yes&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 4000)) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=2200, yend=2200) + annotate(&quot;text&quot;, label=&quot;&lt; 2 e -16&quot;, x=1.5, y=2400, size=3)
print(all.Hme3)</code></pre>
<pre><code>Warning: Removed 403531 rows containing non-finite values (stat_density).</code></pre>
<pre class="r"><code>print(box.all.Hme3, vp=vp)</code></pre>
<pre><code>Warning: Removed 403531 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(chrom.annos, true.promoter==&quot;yes&quot;)$Hme3val, filter(chrom.annos, true.promoter==&quot;no&quot;)$Hme3val)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(chrom.annos, true.promoter == &quot;yes&quot;)$Hme3val and filter(chrom.annos, true.promoter == &quot;no&quot;)$Hme3val
t = 45.255, df = 32792, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 305.3999 333.0516
sample estimates:
mean of x mean of y 
 531.5644  212.3386 </code></pre>
<pre class="r"><code>prom.Hme3 &lt;- ggplot(data=proms.differential) + geom_density(aes(x=Hme3val, group=differential, color=differential, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K4me3 overlapping Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K4me3 peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;))
box.prom.Hme3 &lt;- ggplot(data=proms.differential) + geom_violin(aes(x=differential, y=Hme3val, group=differential, color=differential), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 2000)) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=750, yend=750) + annotate(&quot;segment&quot;, x=2.1, xend=2.9, y=750, yend=750) + annotate(&quot;segment&quot;, x=3.1, xend=3.9, y=750, yend=750) + annotate(&quot;segment&quot;, x=2.1, xend=3.9, y=1000, yend=1000) + annotate(&quot;segment&quot;, x=1.1, xend=2.9, y=1250, yend=1250) + annotate(&quot;segment&quot;, x=1.1, xend=3.9, y=1500, yend=1500) + annotate(&quot;text&quot;, x=1.5, y=840, size=3, label=&quot;0.2&quot;) + annotate(&quot;text&quot;, x=2.5, y=840, size=3, label=&quot;0.02&quot;) + annotate(&quot;text&quot;, x=3.5, y=840, size=3, label=&quot;&lt; 2 e -16&quot;) + annotate(&quot;text&quot;, x=3, y=1090, size=3, label=&quot;9 e -8&quot;) + annotate(&quot;text&quot;, x=2.7, y=1590, size=3, label=&quot;2 e -5&quot;) + annotate(&quot;text&quot;, x=2, y=1340, size=3, label=&quot;6 e -3&quot;)
print(prom.Hme3)
vp &lt;- grid::viewport(width=0.4, height=0.4, x=0.5, y=0.5)
print(box.prom.Hme3, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-17.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.differential, differential==&quot;contact&quot;)$Hme3val, filter(proms.differential, differential==&quot;both&quot;)$Hme3val)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.differential, differential == &quot;contact&quot;)$Hme3val and filter(proms.differential, differential == &quot;both&quot;)$Hme3val
t = -1.2954, df = 469.66, p-value = 0.1958
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -303.30134   62.29743
sample estimates:
mean of x mean of y 
 755.9139  876.4158 </code></pre>
<pre class="r"><code>noprom.Hme3 &lt;- ggplot(data=nonproms) + geom_density(aes(x=Hme3, group=DC, color=DC, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K4me3 overlapping non-promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K4me3 peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Neither&quot;, &quot;Contact&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 4000))
box.noprom.Hme3 &lt;- ggplot(data=nonproms) + geom_violin(aes(x=DC, y=Hme3, group=DC, color=DC), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Neither&quot;, &quot;Contact&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 1000)) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=250, yend=250) + annotate(&quot;text&quot;, x=1.5, y=300, size=3, label=&quot;&lt; 2 e -16&quot;)
print(noprom.Hme3)</code></pre>
<pre><code>Warning: Removed 398499 rows containing non-finite values (stat_density).</code></pre>
<pre class="r"><code>print(box.noprom.Hme3, vp=vp)</code></pre>
<pre><code>Warning: Removed 398499 rows containing non-finite values (stat_ydensity).</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(nonproms, DC==&quot;no&quot;)$Hme3, filter(nonproms, DC==&quot;yes&quot;)$Hme3)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(nonproms, DC == &quot;no&quot;)$Hme3 and filter(nonproms, DC == &quot;yes&quot;)$Hme3
t = -9.1724, df = 9595.4, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -106.94291  -69.28215
sample estimates:
mean of x mean of y 
 208.9105  297.0230 </code></pre>
<pre class="r"><code>DE.Hme3 &lt;- ggplot(data=proms.DE) + geom_density(aes(x=Hme3val, group=DE, color=DE, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K4me3 overlapping DE Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K4me3 peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DE?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 4000))
boxDE.Hme3 &lt;- ggplot(data=proms.DE) + geom_violin(aes(x=DE, y=Hme3val, group=DE, color=DE), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + coord_cartesian(ylim=c(0, 1000)) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=500, yend=500) + annotate(&quot;text&quot;, x=1.5, y=550, size=3, label=&quot;&lt; 2 e -16&quot;)
print(DE.Hme3)
print(boxDE.Hme3, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-19.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.DE, DE==&quot;yes&quot;)$Hme3val, filter(proms.DE, DE==&quot;no&quot;)$Hme3val)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.DE, DE == &quot;yes&quot;)$Hme3val and filter(proms.DE, DE == &quot;no&quot;)$Hme3val
t = 8.5048, df = 12870, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 100.2563 160.3107
sample estimates:
mean of x mean of y 
 652.0436  521.7601 </code></pre>
<pre class="r"><code>DC.Hme3 &lt;- ggplot(data=proms.DC) + geom_density(aes(x=Hme3val, group=DC, color=DC, y=..scaled..)) + ggtitle(&quot;Enrichment of H3K4me3 overlapping DC Promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and H3K4me3 peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DC?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 6000))
boxDC.Hme3 &lt;- ggplot(data=proms.DC) + geom_violin(aes(x=DC, y=Hme3val, group=DC, color=DC), draw_quantiles = c(0.25, 0.5, 0.75)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + coord_cartesian(ylim=c(0, 2000)) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + annotate(&quot;segment&quot;, x=1.1, xend=1.9, y=1000, yend=1000) + annotate(&quot;text&quot;, x=1.5, y=1090, size=3, label=&quot;1 e -9&quot;)
vp &lt;- grid::viewport(width=0.4, height=0.4, x=0.6, y=0.6)
print(DC.Hme3)
print(boxDC.Hme3, vp=vp)</code></pre>
<p><img src="figure/enrichment.Rmd/Other%20quantitative%20enrichments-20.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>t.test(filter(proms.DC, DC==&quot;yes&quot;)$Hme3val, filter(proms.DC, DC==&quot;no&quot;)$Hme3val)</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  filter(proms.DC, DC == &quot;yes&quot;)$Hme3val and filter(proms.DC, DC == &quot;no&quot;)$Hme3val
t = 6.1308, df = 1145.6, p-value = 1.201e-09
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 169.4914 329.0349
sample estimates:
mean of x mean of y 
 773.6531  524.3900 </code></pre>
<pre class="r"><code>#######HAR Annotations! Seem fairly worthless but keep here in case I want to fux with them later (limits of 50 seem best for axes?)
# all.HAR &lt;- ggplot(data=dub.anno) + geom_density(aes(x=HAR, group=promoter, color=promoter, y=..scaled..)) + ggtitle(&quot;Enrichment of HAR Overlapping Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and HAR peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Promoter Contact?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) #+ coord_cartesian(xlim=c(0, 6000))
# box.all.HAR &lt;- ggplot(data=dub.anno) + geom_boxplot(aes(x=promoter, y=HAR, group=promoter, color=promoter)) + scale_color_manual(values=c(&quot;Red&quot;, &quot;Blue&quot;), labels=c(&quot;No&quot;, &quot;Yes&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) #+ coord_cartesian(ylim=c(0, 6000)) #+ geom_text(aes(x=1.5, y=3000, label=&quot;&lt;2.2 e -16&quot;), show.legend=FALSE, size=3)
# print(all.HAR)
# print(box.all.HAR, vp=vp)
# t.test(dub.anno$HAR[which(dub.anno$promoter==&quot;yes&quot;)], dub.anno$HAR[which(dub.anno$promoter==&quot;no&quot;)])
# 
# 
# prom.HAR &lt;- ggplot(data=filter(dub.anno, true.promoter==&quot;yes&quot;)) + geom_density(aes(x=HAR, group=differential, color=differential, y=..scaled..)) + ggtitle(&quot;Enrichment of HAR overlapping true.promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and HAR peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;)) + coord_cartesian(xlim=c(0, 200))
# box.prom.HAR &lt;- ggplot(data=filter(dub.anno, true.promoter==&quot;yes&quot;)) + geom_boxplot(aes(x=differential, y=HAR, group=differential, color=differential)) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 200)) #Not really clear to me why this ends at 8k!
# print(prom.HAR)
# vp &lt;- grid::viewport(width=0.3, height=0.3, x=0.5, y=0.7)
# print(box.prom.HAR, vp=vp)
# t.test(filter(dub.anno, true.promoter==&quot;yes&quot;, differential==&quot;contact&quot;)$HAR, filter(dub.anno, true.promoter==&quot;yes&quot;, differential==&quot;no&quot;)$HAR)
# vp &lt;- grid::viewport(width=0.4, height=0.4, x=0.5, y=0.6)
# 
# noprom.HAR &lt;- ggplot(data=filter(dub.anno, true.promoter==&quot;no&quot;)) + geom_density(aes(x=HAR, group=differential, color=differential, y=..scaled..)) + ggtitle(&quot;Enrichment of HAR overlapping non-promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and HAR peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;)) + coord_cartesian(xlim=c(0, 50))
# box.noprom.HAR &lt;- ggplot(data=filter(dub.anno, true.promoter==&quot;no&quot;)) + geom_boxplot(aes(x=differential, y=HAR, group=differential, color=differential)) + scale_color_manual(name=&quot;Species Differential:&quot;, labels = c(&quot;Both&quot;, &quot;Contact&quot;, &quot;Expression&quot;, &quot;Neither&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;, &quot;Green&quot;, &quot;Purple&quot;), guide=FALSE) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + coord_cartesian(ylim=c(0, 50))# + geom_text(aes(x=1.5, y=1500, label=&quot;9.99 e -14&quot;), show.legend=FALSE, size=3) Can&#39;t figure out why geom text won&#39;t work here/takes too long!!!
# print(noprom.HAR)
# print(box.noprom.HAR, vp=vp)
# t.test(filter(dub.anno, true.promoter==&quot;no&quot;, differential==&quot;no&quot;)$HAR, filter(dub.anno, true.promoter==&quot;no&quot;, differential==&quot;expression&quot;)$HAR)
# 
# DE.HAR &lt;- ggplot(data=filter(dub.anno, true.promoter==&quot;yes&quot;)) + geom_density(aes(x=HAR, group=DE, color=DE, y=..scaled..)) + ggtitle(&quot;Enrichment of HAR overlapping DE true.promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and HAR peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DE?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;))
#   boxDE.HAR &lt;- ggplot(data=filter(dub.anno, true.promoter==&quot;yes&quot;)) + geom_boxplot(aes(x=DE, y=HAR, group=DE, color=DE)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + coord_cartesian(ylim=c(0, 8000)) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + geom_text(aes(x=1.5, y=4000, label=&quot;3 e -14&quot;), show.legend=FALSE, size=3)
# print(DE.HAR)
# print(boxDE.HAR, vp=vp)
# t.test(filter(dub.anno, true.promoter==&quot;yes&quot;, DE==&quot;yes&quot;)$HAR, filter(dub.anno, true.promoter==&quot;yes&quot;, DE==&quot;no&quot;)$HAR)
# 
# DC.HAR &lt;- ggplot(data=filter(dub.anno, true.promoter==&quot;yes&quot;)) + geom_density(aes(x=HAR, group=DH, color=DH, y=..scaled..)) + ggtitle(&quot;Enrichment of HAR overlapping DC true.promoter Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and HAR peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;DC?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 8000))
# boxDC.HAR &lt;- ggplot(data=filter(dub.anno, true.promoter==&quot;yes&quot;)) + geom_boxplot(aes(x=DH, y=HAR, group=DH, color=DH)) + scale_color_manual(labels=c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;), guide=FALSE) + coord_cartesian(ylim=c(0, 8000)) + theme(axis.title.x = element_blank(), axis.text.x=element_blank(), axis.title.y=element_blank()) + geom_text(aes(x=1.5, y=4000, label=&quot;1.9 e -5&quot;), show.legend=FALSE, size=3)
# print(DC.HAR)
# print(boxDC.HAR, vp=vp)
# t.test(filter(dub.anno, true.promoter==&quot;yes&quot;, DH==&quot;yes&quot;)$HAR, filter(dub.anno, true.promoter==&quot;yes&quot;, DH==&quot;no&quot;)$HAR)
# ggplot(data=full.annotations) + geom_density(aes(x=bin1.HARval, group=promoter, color=promoter), linetype=&quot;dashed&quot;) + geom_density(aes(x=bin2.HARval, group=promoter, color=promoter)) + ggtitle(&quot;Enrichment of HAR overlapping ALL Hi-C Bins&quot;) + xlab(&quot;Length of overlap b/t bin and HAR peak&quot;) + ylab(&quot;Density&quot;) + scale_color_manual(name=&quot;Promoter Contact?&quot;, labels = c(&quot;No&quot;, &quot;Yes&quot;), values=c(&quot;Red&quot;, &quot;Blue&quot;)) + coord_cartesian(xlim=c(0, 50))</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.0 (2017-04-21)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: OS X El Capitan 10.11.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] compiler  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] bindrcpp_0.2        bedr_1.0.4          forcats_0.3.0      
 [4] purrr_0.2.4         readr_1.1.1         tibble_1.4.2       
 [7] tidyverse_1.2.1     edgeR_3.20.9        RColorBrewer_1.1-2 
[10] heatmaply_0.14.1    viridis_0.5.0       viridisLite_0.3.0  
[13] stringr_1.3.0       gplots_3.0.1        Hmisc_4.1-1        
[16] Formula_1.2-2       survival_2.41-3     lattice_0.20-35    
[19] dplyr_0.7.4         plotly_4.7.1        ggsignif_0.4.0     
[22] ggplot2_2.2.1       reshape2_1.4.3      data.table_1.10.4-3
[25] tidyr_0.8.0         plyr_1.8.4          limma_3.34.9       

loaded via a namespace (and not attached):
 [1] colorspace_1.3-2     class_7.3-14         modeltools_0.2-21   
 [4] mclust_5.4           rprojroot_1.3-2      htmlTable_1.11.2    
 [7] futile.logger_1.4.3  base64enc_0.1-3      rstudioapi_0.7      
[10] flexmix_2.3-14       mvtnorm_1.0-7        lubridate_1.7.3     
[13] xml2_1.2.0           R.methodsS3_1.7.1    codetools_0.2-15    
[16] splines_3.4.0        mnormt_1.5-5         robustbase_0.92-8   
[19] knitr_1.20           jsonlite_1.5         broom_0.4.3         
[22] cluster_2.0.6        kernlab_0.9-25       R.oo_1.21.0         
[25] httr_1.3.1           backports_1.1.2      assertthat_0.2.0    
[28] Matrix_1.2-12        lazyeval_0.2.1       cli_1.0.0           
[31] acepack_1.4.1        htmltools_0.3.6      tools_3.4.0         
[34] gtable_0.2.0         glue_1.2.0           Rcpp_0.12.16        
[37] cellranger_1.1.0     trimcluster_0.1-2    gdata_2.18.0        
[40] nlme_3.1-131.1       iterators_1.0.9      fpc_2.1-11          
[43] psych_1.7.8          testthat_2.0.0       rvest_0.3.2         
[46] gtools_3.5.0         dendextend_1.7.0     DEoptimR_1.0-8      
[49] MASS_7.3-49          scales_0.5.0         TSP_1.1-5           
[52] hms_0.4.2            parallel_3.4.0       lambda.r_1.2        
[55] yaml_2.1.18          gridExtra_2.3        rpart_4.1-13        
[58] latticeExtra_0.6-28  stringi_1.1.7        gclus_1.3.1         
[61] foreach_1.4.4        checkmate_1.8.5      seriation_1.2-3     
[64] caTools_1.17.1       rlang_0.2.0          pkgconfig_2.0.1     
[67] prabclus_2.2-6       bitops_1.0-6         evaluate_0.10.1     
[70] bindr_0.1.1          labeling_0.3         htmlwidgets_1.0     
[73] tidyselect_0.2.4     magrittr_1.5         R6_2.2.2            
[76] pillar_1.2.1         haven_1.1.1          whisker_0.3-2       
[79] foreign_0.8-69       nnet_7.3-12          modelr_0.1.1        
[82] crayon_1.3.4         futile.options_1.0.0 KernSmooth_2.23-15  
[85] rmarkdown_1.9        locfit_1.5-9.1       grid_3.4.0          
[88] readxl_1.0.0         git2r_0.21.0         digest_0.6.15       
[91] diptest_0.75-7       webshot_0.5.0        VennDiagram_1.6.19  
[94] R.utils_2.6.0        stats4_3.4.0         munsell_0.4.3       
[97] registry_0.5        </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
