<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Eres, Ittai" />

<meta name="date" content="2017-10-06" />

<title>Hi-C Data Normalization and Initial Quality Control</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-0.9/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.7.1/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/plotlyjs-1.29.2/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotlyjs-1.29.2/plotly-latest.min.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Hi-C</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Hi-C Data Normalization and Initial Quality Control</h1>
<h4 class="author"><em>Eres, Ittai</em></h4>
<h4 class="date"><em>2017-10-06</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-01-16</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 98786b1</p>
<!-- Add your analysis here -->
<div id="homer-qc-checks" class="section level3">
<h3>Homer QC Checks</h3>
<p>First, I check some of the summary statistics produced on the libraries by homer, to see if the libraries are comparable across species. This includes distance between 5’ ends of the paired reads, to determine the Hi-C random genomic fragment size in the library, as well as distribution of paired end (PE) reads as a function of the distance between them. The only real point here is to ensure that there is no strong effect seen in either metric stratifying species.</p>
<pre class="r"><code>#Read in pre-prepped tables of this information.
rand.frag &lt;- fread(&quot;~/Desktop/Hi-C/homer_QCs/petag.Local.ALL.distn&quot;, header=TRUE, stringsAsFactors = FALSE, data.table=FALSE)
dist.distn &lt;- fread(&quot;~/Desktop/Hi-C/homer_QCs/petag.Freq.ALL.distn&quot;, header=TRUE, stringsAsFactors = FALSE, data.table=FALSE)

#Create a huge long-form data frame of the same and opposite strand counts from each individual to place in a long-form dataframe for ggplot2 with the local distances and sample identities.
myfrags &lt;- rbind(rand.frag[,2:3], rand.frag[,4:5], rand.frag[,6:7], rand.frag[,8:9], rand.frag[,10:11], rand.frag[,12:13], rand.frag[,14:15], rand.frag[,16:17])

fragdf &lt;- data.frame(local_dist=rand.frag[,1], same=myfrags[,1], opposite=myfrags[,2], ID=c(rep(&quot;A_H&quot;, 20000), rep(&quot;B_H&quot;, 20000), rep(&quot;C_C&quot;, 20000), rep(&quot;D_C&quot;, 20000), rep(&quot;E_H&quot;, 20000), rep(&quot;F_H&quot;, 20000), rep(&quot;G_C&quot;, 20000), rep(&quot;H_C&quot;, 20000)))

ggplot(data=fragdf, aes(x=local_dist, group=ID)) + geom_line(aes(y=same, color=ID)) + ggtitle(&quot;Relationship b/t 5&#39; ends of paired reads, same strand&quot;) + xlab(&quot;Local Distance from 5&#39; of 1st read&quot;) + ylab(&quot;Counts per bp&quot;)</code></pre>
<p><img src="figure/initial_QC.Rmd/Homer%20QC%20Checks-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=fragdf, aes(x=local_dist, group=ID)) + geom_line(aes(y=opposite, color=ID)) + ggtitle(&quot;Relationship b/t 5&#39; ends of paired reads, opposite strand&quot;) + xlab(&quot;Local Distance from 5&#39; of 1st read&quot;) + ylab(&quot;Counts per bp&quot;)</code></pre>
<p><img src="figure/initial_QC.Rmd/Homer%20QC%20Checks-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Critically, there appears to be no clear separation between species here. If there is any effect, it would appear to be that of batch, but even this effect appears minor, and would likely be more related to total sequencing reads (batch 1 had the additional pooled lane) than to actual library prep differences.

#Rearrange data on fraction of reads from each library falling within a certain distance of one another so that it can be plotted for all individuals simultaneously in ggplot2.
dist.distn$`Distance between PE tags`[300001] &lt;- 300000000
dist.distn$binID &lt;- c(rep(seq(0, 299999000, 10000), each=10), 300000000)
colnames(dist.distn) &lt;- c(&quot;distance&quot;, &quot;A_H&quot;, &quot;B_H&quot;, &quot;C_C&quot;, &quot;D_C&quot;, &quot;E_H&quot;, &quot;F_H&quot;, &quot;G_C&quot;, &quot;H_C&quot;, &quot;binID&quot;)

#Group by to lump 10k together at a time.
group_by(dist.distn, binID) %&gt;% summarise(A_H=sum(A_H), B_H=sum(B_H), C_C=sum(C_C), D_C=sum(D_C), E_H=sum(E_H), F_H=sum(F_H), G_C=sum(G_C), H_C=sum(H_C)) -&gt; mydist

#Prepare long-form df of this information for plotting with ggplot.
distdf &lt;- data.frame(dist=rep(mydist$binID, 8), fraction=c(mydist$A_H, mydist$B_H, mydist$C_C, mydist$D_C, mydist$E_H, mydist$F_H, mydist$G_C, mydist$H_C), ID=c(rep(&quot;A_H&quot;, 30001), rep(&quot;B_H&quot;, 30001), rep(&quot;C_C&quot;, 30001), rep(&quot;D_C&quot;, 30001), rep(&quot;E_H&quot;, 30001), rep(&quot;F_H&quot;, 30001), rep(&quot;G_C&quot;, 30001), rep(&quot;H_C&quot;, 30001)), species=c(rep(&quot;H&quot;, 60002), rep(&quot;C&quot;, 60002), rep(&quot;H&quot;, 60002), rep(&quot;C&quot;, 60002)))

#Now, plot out on several different distance scales to see if there is any strong species effect.
ggplot(data=distdf, aes(x=dist, group=ID)) + geom_line(aes(y=fraction, color=ID, linetype=species)) + coord_cartesian(xlim=c(0, 50000)) + xlab(&quot;Distance between mates&quot;) + ylab(&quot;Fraction of Hi-C PE reads&quot;) + ggtitle(&quot;Hi-C Pair Distance Distributions&quot;)</code></pre>
<p><img src="figure/initial_QC.Rmd/Homer%20QC%20Checks-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=distdf, aes(x=dist, group=ID)) + geom_line(aes(y=fraction, color=ID, linetype=species)) + coord_cartesian(xlim=c(0, 100000)) + xlab(&quot;Distance between mates&quot;) + ylab(&quot;Fraction of Hi-C PE reads&quot;) + ggtitle(&quot;Hi-C Pair Distance Distributions&quot;)</code></pre>
<p><img src="figure/initial_QC.Rmd/Homer%20QC%20Checks-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=distdf, aes(x=dist, group=ID)) + geom_line(aes(y=fraction, color=ID, linetype=species)) + coord_cartesian(xlim=c(0, 500000)) + xlab(&quot;Distance between mates&quot;) + ylab(&quot;Fraction of Hi-C PE reads&quot;) + ggtitle(&quot;Hi-C Pair Distance Distributions&quot;)</code></pre>
<p><img src="figure/initial_QC.Rmd/Homer%20QC%20Checks-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Once again, there appears to be no clear separation between species here. There doesn&#39;t even appear to be a batch effect!</code></pre>
<p>From these quality control checks we can infer that the libraries are comparable across species, as neither of these basic metrics for Hi-C library quality differs significantly between the species.</p>
</div>
<div id="initial-data-read-in-and-quality-control" class="section level3">
<h3>Initial Data read-in and quality control</h3>
<p>First, I read in the data and normalize it with cyclic pairwise loess normalization. Then I look at histograms of the distributions of the contact frequencies on an individual-by-individual basis, to see if they are comparable. I also look at a plot of the mean vs. the variance as an additional quality control metric; this is typically done on RNA-seq count data. In that case, it’s done to check if the normalized data are distributed in a particular fashion (e.g. a poisson model would have a linear mean-variance relationship). In the limma/voom pipeline, count data are log-cpm transformed and a loess trend line of variance vs. mean is then fit to create weights for individual genes to be fed into linear modeling with limma. Since this is not a QC metric typically used for Hi-C data, the only thing I hope to see is no particularly strong relationship between variance and mean.</p>
<pre class="r"><code>###Read in data, normalize it with cyclic loess (pairwise), clean it from any rows that have missing values.
full.data &lt;- fread(&quot;~/Desktop/final.10kb.homer.df&quot;, header=TRUE, stringsAsFactors = FALSE, data.table=FALSE, showProgress = FALSE)
full.contacts &lt;- full.data[complete.cases(full.data[,304:311]), 304:311]
full.contacts &lt;- as.data.frame(normalizeCyclicLoess(full.contacts, span=1/4, iterations=3, method=&quot;pairs&quot;))
full.data &lt;- full.data[complete.cases(full.data[,304:311]),]
full.data[,304:311] &lt;- full.contacts

###First, a quick look at histograms of the distributions of the significant Hi-C hits from homer, in both humans and chimps.
ggplot(data=melt(full.contacts[,c(1:2, 5:6)]), aes(x=value)) + geom_histogram(binwidth=0.0009, aes(group=variable)) + facet_wrap(~as.factor(variable)) + ggtitle(&quot;Distributions of Homer-Significant Hi-C Contacts, Humans&quot;) + xlab(&quot;Homer-Normalized Interaction Frequency&quot;) + coord_cartesian(xlim=c(-6.6, 6.6), ylim=c(0, 4500)) #Human Dist&#39;ns</code></pre>
<pre><code>No id variables; using all as measure variables</code></pre>
<p><img src="figure/initial_QC.Rmd/Initial%20Hi-C%20Data%20Read-in%20and%20QC-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=melt(full.contacts[,c(3:4, 7:8)]), aes(x=value)) + geom_histogram(binwidth=0.0009, aes(group=variable)) + facet_wrap(~as.factor(variable)) + ggtitle(&quot;Distributions of Homer-Significant Hi-C Contacts, Chimps&quot;) + xlab(&quot;Homer-Normalized Interaction Frequency&quot;) + coord_cartesian(xlim=c(-6.6, 6.6), ylim=c(0, 4500)) #Chimp Dist&#39;ns</code></pre>
<pre><code>No id variables; using all as measure variables</code></pre>
<p><img src="figure/initial_QC.Rmd/Initial%20Hi-C%20Data%20Read-in%20and%20QC-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Both sets of distributions look fairly bimodal, with chimps in particular showing a peak around 0.

###As an initial quality control metric, take a look at the mean vs. variance plot for the normalized data. Also go ahead and add on columns for mean frequencies and variances both within and across the species, while they&#39;re being calculated here anyway:
select(full.data, &quot;A-21792_norm&quot;, &quot;B-28126_norm&quot;, &quot;E-28815_norm&quot;, &quot;F-28834_norm&quot;) %&gt;% apply(., 1, mean) -&gt; full.data$Hmean
select(full.data, &quot;C-3649_norm&quot;, &quot;D-40300_norm&quot;, &quot;G-3624_norm&quot;, &quot;H-3651_norm&quot;) %&gt;% apply(., 1, mean) -&gt; full.data$Cmean
select(full.data, &quot;A-21792_norm&quot;, &quot;B-28126_norm&quot;, &quot;E-28815_norm&quot;, &quot;F-28834_norm&quot;, &quot;C-3649_norm&quot;, &quot;D-40300_norm&quot;, &quot;G-3624_norm&quot;, &quot;H-3651_norm&quot;) %&gt;% apply(., 1, mean) -&gt; full.data$ALLmean #Get means across species.
select(full.data, &quot;A-21792_norm&quot;, &quot;B-28126_norm&quot;, &quot;E-28815_norm&quot;, &quot;F-28834_norm&quot;) %&gt;% apply(., 1, var) -&gt; full.data$Hvar #Get human variances.
select(full.data, &quot;C-3649_norm&quot;, &quot;D-40300_norm&quot;, &quot;G-3624_norm&quot;, &quot;H-3651_norm&quot;) %&gt;% apply(., 1, var) -&gt; full.data$Cvar #Get chimp variances.
select(full.data, &quot;A-21792_norm&quot;, &quot;B-28126_norm&quot;, &quot;E-28815_norm&quot;, &quot;F-28834_norm&quot;, &quot;C-3649_norm&quot;, &quot;D-40300_norm&quot;, &quot;G-3624_norm&quot;, &quot;H-3651_norm&quot;) %&gt;% apply(., 1, var) -&gt; full.data$ALLvar #Get variance across species.
{plot(full.data$ALLmean, full.data$ALLvar, main=&quot;Mean vs. Variance in Homer-corrected contact frequency&quot;, xlab=&quot;Mean Hi-C Contact Frequency&quot;, ylab=&quot;Variance in Hi-C Contact Frequency&quot;)
  abline(lm(full.data$ALLmean~full.data$ALLvar), col=&quot;red&quot;)}</code></pre>
<p><img src="figure/initial_QC.Rmd/Initial%20Hi-C%20Data%20Read-in%20and%20QC-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Though there is a slight downward trend, on the whole, there is not a strong correlation between the contact frequency and the variance across individuals for it.</code></pre>
<p>On the whole, the distributions look fairly comparable across individuals and species, and the mean vs. variance plots show weak correlation at best between the two metrics. Now knowing the data are comparable in this sense, the next question would be whether they are different enough between the species to separate them from one another with unsupervised clustering and PCA methods.</p>
</div>
<div id="data-clustering-with-principal-components-analysis-pca-and-correlation-heatmap-clustering." class="section level3">
<h3>Data clustering with Principal Components Analysis (PCA) and correlation heatmap clustering.</h3>
<p>Now I use PCA, breaking down the interaction frequency values into principal components that best represent the variability within the data. This technique uses orthogonal transformation to convert the many dimensions in variability of this dataset into a lower-dimensional picture that can be used to separate out functional groups in the data. In this case, the expectation is that the strongest principal component, representing the plurality of the variance, will separate out humans and chimps along its axis, grouping the two species independently, as we expect their interaction frequency values to cluster together. I then also compute pairwise pearson correlations between each of the individuals across all Hi-C contacts, and use unsupervised hierarchical clustering to create a heatmap that once again will group individuals based on similarity in interaction frequency values. Again, I would expect this technique to separate out the species very distinctly from one another.</p>
<pre class="r"><code>###Now do principal components analysis (PCA) on these data to see how they separate:
meta.data &lt;- data.frame(&quot;SP&quot;=c(&quot;H&quot;, &quot;H&quot;, &quot;C&quot;, &quot;C&quot;, &quot;H&quot;, &quot;H&quot;, &quot;C&quot;, &quot;C&quot;), &quot;SX&quot;=c(&quot;F&quot;, &quot;M&quot;, &quot;M&quot;, &quot;F&quot;, &quot;M&quot;, &quot;F&quot;, &quot;M&quot;, &quot;F&quot;), &quot;Batch&quot;=c(1, 1, 1, 1, 2, 2, 2, 2), &quot;PE_reads&quot;=c(1084472930, 1103077950, 1015696574, 1047650944, 980287418, 1037054332, 930089380, 964085606), &quot;tags_per_BP&quot;=c(0.351291, 0.357315, 0.342325, 0.353097, 0.317553, 0.335937, 0.313494, 0.324936), &quot;inter_reads&quot;=c(0.1825411, 0.1734168, 0.1559306, 0.2010798, 0.1711258, 0.1479523, 0.1604700, 0.1712287)) #need the metadata first to make this interesting; obtain some of this information from my study design and other parts of it from homer&#39;s outputs
pca &lt;- prcomp(t(full.contacts), scale=TRUE, center=TRUE)
ggplot(data=as.data.frame(pca$x), aes(x=PC1, y=PC2, shape=as.factor(meta.data$SP), color=as.factor(meta.data$Batch), size=2)) + geom_point() +labs(title=&quot;PCA on normalized Hi-C Contact Frequency&quot;) + guides(color=guide_legend(order=1), size=FALSE, shape=guide_legend(order=2)) + xlab(paste(&quot;PC1 (&quot;, 100*summary(pca)$importance[2,1], &quot;% of variance)&quot;)) + ylab((paste(&quot;PC2 (&quot;, 100*summary(pca)$importance[2,2], &quot;% of variance)&quot;))) + labs(color=&quot;Batch&quot;, shape=&quot;Species&quot;) + scale_shape_manual(labels=c(&quot;Chimp&quot;, &quot;Human&quot;), values=c(16, 17)) #PCA shows the species separating nicely along PC1, which explains the highest proportion of the variance (as seen from scree plot below). This is nice and suggests that there ARE true differences in the data between species we are likely to detect here. None of the other PCs correlate with the number of PE reads, the tags per base pair statistic, or the fraction of interchromosomal reads, all from the homer output (not shown).</code></pre>
<p><img src="figure/initial_QC.Rmd/Clustering%20w/%20PCA%20and%20Heatmaps-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(summary(pca)$importance[2,], xlab=&quot;PCs&quot;, ylab=&quot;Proportion of Variance Explained&quot;, main=&quot;PCA on normalized Hi-C contact frequency&quot;) #Scree plot showing all the PCs and the proportion of the variance they explain.</code></pre>
<p><img src="figure/initial_QC.Rmd/Clustering%20w/%20PCA%20and%20Heatmaps-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Heatmap clustering of these data as another quality control metric:
corheat &lt;- cor(full.contacts, use=&quot;complete.obs&quot;, method=&quot;pearson&quot;) #Corheat for the full data set, and heatmap
colnames(corheat) &lt;- c(&quot;A_HF&quot;, &quot;B_HM&quot;, &quot;C_CM&quot;, &quot;D_CF&quot;, &quot;E_HM&quot;, &quot;F_HF&quot;, &quot;G_CM&quot;, &quot;H_CF&quot;)
rownames(corheat) &lt;- colnames(corheat)
heatmaply(corheat, main=&quot;Pairwise Pearson Correlation @ 10kb&quot;, k_row=2, k_col=2, symm=TRUE, margins=c(50, 50, 30, 30))</code></pre>
<pre><code>We recommend that you use the dev version of ggplot2 with `ggplotly()`
Install it with: `devtools::install_github(&#39;hadley/ggplot2&#39;)`
We recommend that you use the dev version of ggplot2 with `ggplotly()`
Install it with: `devtools::install_github(&#39;hadley/ggplot2&#39;)`
We recommend that you use the dev version of ggplot2 with `ggplotly()`
Install it with: `devtools::install_github(&#39;hadley/ggplot2&#39;)`</code></pre>
<div id="912658383e91" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="912658383e91">{"x":{"data":[{"x":[4.5,7.125,null,7.125,7.125,null,7.125,6.25,null,6.25,6.25,null,6.25,5.5,null,5.5,5.5,null,5.5,5,null,5,5,null,5.5,6,null,6,6,null,6.25,7,null,7,7,null,7.125,8,null,8,8],"y":[1.18517552954421,1.18517552954421,null,1.18517552954421,1.06687081321649,null,1.06687081321649,1.06687081321649,null,1.06687081321649,1.03441233125822,null,1.03441233125822,1.03441233125822,null,1.03441233125822,1.00124047394332,null,1.00124047394332,1.00124047394332,null,1.00124047394332,0,null,1.00124047394332,1.00124047394332,null,1.00124047394332,0,null,1.03441233125822,1.03441233125822,null,1.03441233125822,0,null,1.06687081321649,1.06687081321649,null,1.06687081321649,0],"text":["y: 1.1851755","y: 1.1851755",null,"y: 1.1851755","y: 1.1851755",null,"y: 1.0668708","y: 1.0668708",null,"y: 1.0668708","y: 1.0668708",null,"y: 1.0344123","y: 1.0344123",null,"y: 1.0344123","y: 1.0344123",null,"y: 1.0012405","y: 1.0012405",null,"y: 1.0012405","y: 1.0012405",null,"y: 1.0012405","y: 1.0012405",null,"y: 1.0012405","y: 1.0012405",null,"y: 1.0344123","y: 1.0344123",null,"y: 1.0344123","y: 1.0344123",null,"y: 1.0668708","y: 1.0668708",null,"y: 1.0668708","y: 1.0668708"],"type":"scatter","mode":"lines","line":{"width":2.26771653543307,"color":"rgba(57,190,177,1)","dash":"solid"},"hoveron":"points","name":"(#39BEB1,solid,1)","legendgroup":"(#39BEB1,solid,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4.5,1.875,null,1.875,1.875,null,1.875,1,null,1,1,null,1.875,2.75,null,2.75,2.75,null,2.75,2,null,2,2,null,2.75,3.5,null,3.5,3.5,null,3.5,3,null,3,3,null,3.5,4,null,4,4],"y":[1.18517552954421,1.18517552954421,null,1.18517552954421,1.00188827852063,null,1.00188827852063,1.00188827852063,null,1.00188827852063,0,null,1.00188827852063,1.00188827852063,null,1.00188827852063,0.961508716398257,null,0.961508716398257,0.961508716398257,null,0.961508716398257,0,null,0.961508716398257,0.961508716398257,null,0.961508716398257,0.9392561944276,null,0.9392561944276,0.9392561944276,null,0.9392561944276,0,null,0.9392561944276,0.9392561944276,null,0.9392561944276,0],"text":["y: 1.1851755","y: 1.1851755",null,"y: 1.1851755","y: 1.1851755",null,"y: 1.0018883","y: 1.0018883",null,"y: 1.0018883","y: 1.0018883",null,"y: 1.0018883","y: 1.0018883",null,"y: 1.0018883","y: 1.0018883",null,"y: 0.9615087","y: 0.9615087",null,"y: 0.9615087","y: 0.9615087",null,"y: 0.9615087","y: 0.9615087",null,"y: 0.9615087","y: 0.9615087",null,"y: 0.9392562","y: 0.9392562",null,"y: 0.9392562","y: 0.9392562",null,"y: 0.9392562","y: 0.9392562",null,"y: 0.9392562","y: 0.9392562"],"type":"scatter","mode":"lines","line":{"width":2.26771653543307,"color":"rgba(228,149,165,1)","dash":"solid"},"hoveron":"points","name":"(#E495A5,solid,1)","legendgroup":"(#E495A5,solid,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4.5,1.875,1,2.75,2,3.5,3,4,7.125,6.25,5.5,5,6,7,8],"y":[1.18517552954421,1.00188827852063,0,0.961508716398257,0,0.9392561944276,0,0,1.06687081321649,1.03441233125822,1.00124047394332,0,0,0,0],"text":["y: 1.1851755","y: 1.0018883","y: 0.0000000","y: 0.9615087","y: 0.0000000","y: 0.9392562","y: 0.0000000","y: 0.0000000","y: 1.0668708","y: 1.0344123","y: 1.0012405","y: 0.0000000","y: 0.0000000","y: 0.0000000","y: 0.0000000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"transparent","opacity":1,"size":null,"symbol":null,"line":{"width":1.88976377952756,"color":"transparent"}},"hoveron":"points","name":"(NA,1)","legendgroup":"(NA,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"type":"scatter","mode":"markers","marker":{"fillcolor":"rgba(31,119,180,1)","color":"rgba(214,39,40,1)","line":{"color":"transparent"}},"xaxis":"x2","yaxis":"y","frame":null},{"x":[1,2,3,4,5,6,7,8],"y":[1,2,3,4,5,6,7,8],"z":[[1,0.141533380681047,0.158839274622791,0.153050453403804,0.0589406966114147,0.0230409524307309,0,0.00437708703382176],[0.141533380681047,1,0.175877488292864,0.176062646102501,0.0740342739152109,0.0453483816077108,0.0212883522049347,0.0243494183113591],[0.158839274622791,0.175877488292864,1,0.194560650524792,0.0962416337564241,0.0539072528015847,0.0322678232360692,0.0305706707847016],[0.153050453403804,0.176062646102501,0.194560650524792,1,0.0969371194063134,0.0551784141031276,0.0295367439769798,0.033060053894094],[0.0589406966114147,0.0740342739152109,0.0962416337564241,0.0969371194063134,1,0.143101641294004,0.117561206778976,0.118110209324742],[0.0230409524307309,0.0453483816077108,0.0539072528015847,0.0551784141031276,0.143101641294004,1,0.124001712649372,0.104693672326057],[0,0.0212883522049347,0.0322678232360692,0.0295367439769798,0.117561206778976,0.124001712649372,1,0.085223977912317],[0.00437708703382176,0.0243494183113591,0.0305706707847016,0.033060053894094,0.118110209324742,0.104693672326057,0.085223977912317,1]],"text":[["value: 1.0000000<br />column: D_CF<br />row: D_CF","value: 0.2921335<br />column: H_CF<br />row: D_CF","value: 0.3064034<br />column: C_CM<br />row: D_CF","value: 0.3016301<br />column: G_CM<br />row: D_CF","value: 0.2240299<br />column: E_HM<br />row: D_CF","value: 0.1944281<br />column: F_HF<br />row: D_CF","value: 0.1754292<br />column: B_HM<br />row: D_CF","value: 0.1790384<br />column: A_HF<br />row: D_CF"],["value: 0.2921335<br />column: D_CF<br />row: H_CF","value: 1.0000000<br />column: H_CF<br />row: H_CF","value: 0.3204526<br />column: C_CM<br />row: H_CF","value: 0.3206053<br />column: G_CM<br />row: H_CF","value: 0.2364757<br />column: E_HM<br />row: H_CF","value: 0.2128221<br />column: F_HF<br />row: H_CF","value: 0.1929829<br />column: B_HM<br />row: H_CF","value: 0.1955070<br />column: A_HF<br />row: H_CF"],["value: 0.3064034<br />column: D_CF<br />row: C_CM","value: 0.3204526<br />column: H_CF<br />row: C_CM","value: 1.0000000<br />column: C_CM<br />row: C_CM","value: 0.3358582<br />column: G_CM<br />row: C_CM","value: 0.2547872<br />column: E_HM<br />row: C_CM","value: 0.2198795<br />column: F_HF<br />row: C_CM","value: 0.2020363<br />column: B_HM<br />row: C_CM","value: 0.2006368<br />column: A_HF<br />row: C_CM"],["value: 0.3016301<br />column: D_CF<br />row: G_CM","value: 0.3206053<br />column: H_CF<br />row: G_CM","value: 0.3358582<br />column: C_CM<br />row: G_CM","value: 1.0000000<br />column: G_CM<br />row: G_CM","value: 0.2553607<br />column: E_HM<br />row: G_CM","value: 0.2209277<br />column: F_HF<br />row: G_CM","value: 0.1997843<br />column: B_HM<br />row: G_CM","value: 0.2026895<br />column: A_HF<br />row: G_CM"],["value: 0.2240299<br />column: D_CF<br />row: E_HM","value: 0.2364757<br />column: H_CF<br />row: E_HM","value: 0.2547872<br />column: C_CM<br />row: E_HM","value: 0.2553607<br />column: G_CM<br />row: E_HM","value: 1.0000000<br />column: E_HM<br />row: E_HM","value: 0.2934266<br />column: F_HF<br />row: E_HM","value: 0.2723667<br />column: B_HM<br />row: E_HM","value: 0.2728194<br />column: A_HF<br />row: E_HM"],["value: 0.1944281<br />column: D_CF<br />row: F_HF","value: 0.2128221<br />column: H_CF<br />row: F_HF","value: 0.2198795<br />column: C_CM<br />row: F_HF","value: 0.2209277<br />column: G_CM<br />row: F_HF","value: 0.2934266<br />column: E_HM<br />row: F_HF","value: 1.0000000<br />column: F_HF<br />row: F_HF","value: 0.2776774<br />column: B_HM<br />row: F_HF","value: 0.2617565<br />column: A_HF<br />row: F_HF"],["value: 0.1754292<br />column: D_CF<br />row: B_HM","value: 0.1929829<br />column: H_CF<br />row: B_HM","value: 0.2020363<br />column: C_CM<br />row: B_HM","value: 0.1997843<br />column: G_CM<br />row: B_HM","value: 0.2723667<br />column: E_HM<br />row: B_HM","value: 0.2776774<br />column: F_HF<br />row: B_HM","value: 1.0000000<br />column: B_HM<br />row: B_HM","value: 0.2457024<br />column: A_HF<br />row: B_HM"],["value: 0.1790384<br />column: D_CF<br />row: A_HF","value: 0.1955070<br />column: H_CF<br />row: A_HF","value: 0.2006368<br />column: C_CM<br />row: A_HF","value: 0.2026895<br />column: G_CM<br />row: A_HF","value: 0.2728194<br />column: E_HM<br />row: A_HF","value: 0.2617565<br />column: F_HF<br />row: A_HF","value: 0.2457024<br />column: B_HM<br />row: A_HF","value: 1.0000000<br />column: A_HF<br />row: A_HF"]],"colorscale":[[0,"#440154"],[0.00437708703382176,"#440256"],[0.0212883522049347,"#46095C"],[0.0230409524307309,"#460A5D"],[0.0243494183113591,"#460A5D"],[0.0295367439769798,"#470C5F"],[0.0305706707847016,"#470D60"],[0.0322678232360692,"#470D60"],[0.033060053894094,"#470D60"],[0.0453483816077108,"#471265"],[0.0539072528015847,"#481668"],[0.0551784141031276,"#481668"],[0.0589406966114147,"#481769"],[0.0740342739152109,"#481C6E"],[0.085223977912317,"#482071"],[0.0962416337564241,"#482475"],[0.0969371194063134,"#482475"],[0.104693672326057,"#482677"],[0.117561206778976,"#472A7A"],[0.118110209324742,"#472A7A"],[0.124001712649372,"#472D7B"],[0.141533380681047,"#46327E"],[0.143101641294004,"#46327E"],[0.153050453403804,"#453581"],[0.158839274622791,"#453882"],[0.175877488292864,"#433D84"],[0.176062646102501,"#433D84"],[0.194560650524792,"#414287"],[1,"#FDE725"]],"type":"heatmap","showscale":false,"autocolorscale":false,"showlegend":false,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[1],"y":[1],"name":"99_61296fe47d049705a2b30862cdc826bd","type":"scatter","mode":"markers","opacity":0,"hoverinfo":"none","showlegend":false,"marker":{"color":[0,1],"colorscale":[[0,"#470C5F"],[0.0625,"#482173"],[0.125,"#453581"],[0.1875,"#3F4788"],[0.25,"#38588C"],[0.3125,"#31688E"],[0.375,"#2A768E"],[0.4375,"#25858E"],[0.5,"#20938C"],[0.5625,"#1FA287"],[0.625,"#2AB07F"],[0.6875,"#41BE71"],[0.75,"#60CA60"],[0.8125,"#85D54A"],[0.875,"#ADDC30"],[0.9375,"#D7E219"],[1,"#FDE725"]],"colorbar":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"thickness":23.04,"title":null,"titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"tickmode":"array","ticktext":["0.2","0.4","0.6","0.8","1.0"],"tickvals":[0,0.25,0.5,0.75,1],"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"ticklen":2,"len":0.5}},"xaxis":"x","yaxis":"y2","frame":null},{"x":[1.18517552954421,1.18517552954421,null,1.18517552954421,1.06687081321649,null,1.06687081321649,1.06687081321649,null,1.06687081321649,1.03441233125822,null,1.03441233125822,1.03441233125822,null,1.03441233125822,1.00124047394332,null,1.00124047394332,1.00124047394332,null,1.00124047394332,0,null,1.00124047394332,1.00124047394332,null,1.00124047394332,0,null,1.03441233125822,1.03441233125822,null,1.03441233125822,0,null,1.06687081321649,1.06687081321649,null,1.06687081321649,0],"y":[4.5,7.125,null,7.125,7.125,null,7.125,6.25,null,6.25,6.25,null,6.25,5.5,null,5.5,5.5,null,5.5,5,null,5,5,null,5.5,6,null,6,6,null,6.25,7,null,7,7,null,7.125,8,null,8,8],"text":["y: 1.1851755","y: 1.1851755",null,"y: 1.1851755","y: 1.1851755",null,"y: 1.0668708","y: 1.0668708",null,"y: 1.0668708","y: 1.0668708",null,"y: 1.0344123","y: 1.0344123",null,"y: 1.0344123","y: 1.0344123",null,"y: 1.0012405","y: 1.0012405",null,"y: 1.0012405","y: 1.0012405",null,"y: 1.0012405","y: 1.0012405",null,"y: 1.0012405","y: 1.0012405",null,"y: 1.0344123","y: 1.0344123",null,"y: 1.0344123","y: 1.0344123",null,"y: 1.0668708","y: 1.0668708",null,"y: 1.0668708","y: 1.0668708"],"type":"scatter","mode":"lines","line":{"width":2.26771653543307,"color":"rgba(57,190,177,1)","dash":"solid"},"hoveron":"points","name":"(#39BEB1,solid,1)","legendgroup":"(#39BEB1,solid,1)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[1.18517552954421,1.18517552954421,null,1.18517552954421,1.00188827852063,null,1.00188827852063,1.00188827852063,null,1.00188827852063,0,null,1.00188827852063,1.00188827852063,null,1.00188827852063,0.961508716398257,null,0.961508716398257,0.961508716398257,null,0.961508716398257,0,null,0.961508716398257,0.961508716398257,null,0.961508716398257,0.9392561944276,null,0.9392561944276,0.9392561944276,null,0.9392561944276,0,null,0.9392561944276,0.9392561944276,null,0.9392561944276,0],"y":[4.5,1.875,null,1.875,1.875,null,1.875,1,null,1,1,null,1.875,2.75,null,2.75,2.75,null,2.75,2,null,2,2,null,2.75,3.5,null,3.5,3.5,null,3.5,3,null,3,3,null,3.5,4,null,4,4],"text":["y: 1.1851755","y: 1.1851755",null,"y: 1.1851755","y: 1.1851755",null,"y: 1.0018883","y: 1.0018883",null,"y: 1.0018883","y: 1.0018883",null,"y: 1.0018883","y: 1.0018883",null,"y: 1.0018883","y: 1.0018883",null,"y: 0.9615087","y: 0.9615087",null,"y: 0.9615087","y: 0.9615087",null,"y: 0.9615087","y: 0.9615087",null,"y: 0.9615087","y: 0.9615087",null,"y: 0.9392562","y: 0.9392562",null,"y: 0.9392562","y: 0.9392562",null,"y: 0.9392562","y: 0.9392562",null,"y: 0.9392562","y: 0.9392562"],"type":"scatter","mode":"lines","line":{"width":2.26771653543307,"color":"rgba(228,149,165,1)","dash":"solid"},"hoveron":"points","name":"(#E495A5,solid,1)","legendgroup":"(#E495A5,solid,1)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[1.18517552954421,1.00188827852063,0,0.961508716398257,0,0.9392561944276,0,0,1.06687081321649,1.03441233125822,1.00124047394332,0,0,0,0],"y":[4.5,1.875,1,2.75,2,3.5,3,4,7.125,6.25,5.5,5,6,7,8],"text":["y: 1.1851755","y: 1.0018883","y: 0.0000000","y: 0.9615087","y: 0.0000000","y: 0.9392562","y: 0.0000000","y: 0.0000000","y: 1.0668708","y: 1.0344123","y: 1.0012405","y: 0.0000000","y: 0.0000000","y: 0.0000000","y: 0.0000000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"transparent","opacity":1,"size":null,"symbol":null,"line":{"width":1.88976377952756,"color":"transparent"}},"hoveron":"points","name":"(NA,1)","legendgroup":"(NA,1)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null}],"layout":{"xaxis":{"domain":[0,0.8],"type":"linear","autorange":false,"tickmode":"array","range":[0.5,8.5],"ticktext":["D_CF","H_CF","C_CM","G_CM","E_HM","F_HF","B_HM","A_HF"],"tickvals":[1,2,3,4,5,6,7,8],"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":13.2835201328352},"tickangle":-45,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y2","title":"","titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"hoverformat":".2f"},"xaxis2":{"domain":[0.8,1],"showticklabels":false,"showgrid":false,"zeroline":false,"anchor":"y2","type":"linear","autorange":false,"tickmode":"array","range":[0,1.18517552954421],"ticktext":["0.0","0.3","0.6","0.9"],"tickvals":[0,0.3,0.6,0.9],"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"gridcolor":null,"gridwidth":0,"title":"","titlefont":{"color":null,"family":null,"size":0},"hoverformat":".2f"},"yaxis2":{"domain":[0,0.8],"type":"linear","autorange":false,"tickmode":"array","range":[0.5,8.5],"ticktext":["D_CF","H_CF","C_CM","G_CM","E_HM","F_HF","B_HM","A_HF"],"tickvals":[1,2,3,4,5,6,7,8],"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":13.2835201328352},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"hoverformat":".2f"},"yaxis":{"domain":[0.8,1],"showticklabels":false,"showgrid":false,"zeroline":false,"anchor":"x","type":"linear","autorange":false,"tickmode":"array","range":[0,1.18517552954421],"ticktext":["0.0","0.3","0.6","0.9"],"tickvals":[0,0.3,0.6,0.9],"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"gridcolor":null,"gridwidth":0,"title":"","titlefont":{"color":null,"family":null,"size":0},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.8,"y0":0.8,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.8,"y0":0,"y1":0.8},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.8,"x1":1,"y0":0,"y1":0.8}],"margin":{"t":30,"r":30,"b":50,"l":50},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"y":1,"yanchor":"top"},"hovermode":"closest","paper_bgcolor":"rgba(255,255,255,1)","title":"Pairwise Pearson Correlation @ 10kb"},"attrs":{"91263c3b1e3b":{"xend":{},"yend":{},"colour":{},"linetype":{},"size":{},"x":{},"y":{},"type":"ggplotly"},"91264d3dd261":{"colour":{},"shape":{},"size":{},"x":{},"y":{}},"91266e035400":{"alpha":1,"sizes":[10,100]},"9126527f7b5f":{"fill":{},"x":{},"y":{},"type":"ggplotly"},"912668b42598":{"xend":{},"yend":{},"colour":{},"linetype":{},"size":{},"x":{},"y":{},"type":"ggplotly"},"9126b1053dc":{"colour":{},"shape":{},"size":{},"x":{},"y":{}}},"source":"A","config":{"modeBarButtonsToAdd":[],"cloud":false,"displaylogo":false,"modeBarButtonsToRemove":["sendDataToCloud","select2d","lasso2d","autoScale2d","hoverClosestCartesian","hoverCompareCartesian","sendDataToCloud"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1}},"subplot":true,"base_url":"https://plot.ly"},"evals":[],"jsHooks":{"render":[{"code":"function(el, x) { var ctConfig = crosstalk.var('plotlyCrosstalkOpts').set({\"on\":\"plotly_click\",\"persistent\":false,\"dynamic\":false,\"selectize\":false,\"opacityDim\":0.2,\"selected\":{\"opacity\":1}}); }","data":null}]}}</script>
<pre class="r"><code>#We can see that all the humans cluster together, as do all the chimps. This is very good, despite the fact that the correlations are fairly low.</code></pre>
</div>
<div id="variance-exploration" class="section level3">
<h3>Variance exploration</h3>
<p>Now I look at several metrics to assess variance in the data, checking which species hits were discovered as significant in and how many individuals a given hit was discovered in. I look at this to understand if there are some cutoffs that should be made to reduce the noisiness of the data and maximize the significance of further findings.</p>
<pre class="r"><code>###Add columns to full.data to indicate species of discovery and number of individuals discovered in. These pieces of information are good to know about each of the hits generally, but can also be used to make decisions about filtering out certain classes of hits if variance is associated with any of these metrics.  
humNAs &lt;- rowSums(is.na(full.data[,1:149])) #148 is when there is no human info. 37 NAs per individual.
chimpNAs &lt;- rowSums(is.na(full.data[,150:297])) #Same as above of course.
full.data$found_in_H &lt;- (4-humNAs/37) #Set a new column identifying number of humans hit was found in
full.data$found_in_C &lt;- (4-chimpNAs/37) #Set a new column identifying number of chimps hit was found in
full.data$disc_species &lt;- ifelse(full.data$found_in_C&gt;0&amp;full.data$found_in_H&gt;0, &quot;B&quot;, ifelse(full.data$found_in_C==0, &quot;H&quot;, &quot;C&quot;)) #Set a column identifying which species (H, C, or Both) the hit in question was identified in. Works with nested ifelses.
full.data$tot_indiv_IDd &lt;- full.data$found_in_C+full.data$found_in_H #Add a column specifying total number of individuals homer found the significant hit in.

###Take a look at what proportion of the significant hits were discovered in either of the species (or both of them).
sum(full.data$disc_species==&quot;H&quot;) #~1.2 million</code></pre>
<pre><code>[1] 1186069</code></pre>
<pre class="r"><code>sum(full.data$disc_species==&quot;C&quot;) #~1.1 million</code></pre>
<pre><code>[1] 1057817</code></pre>
<pre class="r"><code>sum(full.data$disc_species==&quot;B&quot;) #~690 k</code></pre>
<pre><code>[1] 688624</code></pre>
<pre class="r"><code>#This is reassuring, as similar numbers of discoveries in both species suggests comparable power for detection.

###Now I look at variance in interaction frequency as a function of the number of individuals in which a pair was independently called as significant. Essentially, I&#39;m checking here to see if there is some kind of cutoff I should make for the significant hits--i.e., if the variance looks to drop off significantly after a hit is discovered in at least 2 individuals, this suggests possibly filtering out any hit only discovered in 1 individual.
myplot &lt;- data.frame(disc_species=full.data$disc_species, found_in_H=full.data$found_in_H, found_in_C=full.data$found_in_C, tot_indiv_IDd=full.data$tot_indiv_IDd, Hvar=full.data$Hvar, Cvar=full.data$Cvar, ALLvar=full.data$ALLvar)
ggplot(data=myplot, aes(group=tot_indiv_IDd, x=tot_indiv_IDd, y=ALLvar)) + geom_boxplot() + ggtitle(&quot;Number of Individuals a Hi-C Hit is Discovered in vs. Variance&quot;) + xlab(&quot;# Individuals where Hi-C contact called significant&quot;) + ylab(&quot;Variance in Interaction Frequency&quot;)</code></pre>
<p><img src="figure/initial_QC.Rmd/Variance%20QC-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=myplot, aes(group=tot_indiv_IDd, x=tot_indiv_IDd, y=ALLvar)) + geom_boxplot() + scale_y_continuous(limits=c(0, 1)) + ggtitle(&quot;Number of Individuals a Hi-C Hit is Discovered in vs. Variance&quot;) + xlab(&quot;# Individuals where Hi-C contact called significant&quot;) + ylab(&quot;Variance in Interaction Frequency&quot;)</code></pre>
<pre><code>Warning: Removed 737679 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/initial_QC.Rmd/Variance%20QC-2.png" width="672" style="display: block; margin: auto;" /> From this, it would appear that variance drops off drastically after discovery in at least 4 individuals. I observed a similar pattern when looking at number of individuals a hit is discovered in vs. species-specific variances, as well as similar reductions when looking at variance in hits within species vs. number of individuals discovered in (not shown). Hence I now repeat all of the QC analyses done above, but this time conditioned upon discovery in 4 individuals. Moving forward I will ultimately filter out all the hits that were not found in at least 4 individuals, though I sometimes check the full dataset as well just for sanity’s sake.</p>
</div>
<div id="initial-quality-control-analyses-conditioned-upon-discovery-in-at-least-4-individuals" class="section level3">
<h3>Initial Quality Control Analyses Conditioned Upon Discovery in at Least 4 Individuals</h3>
<p>Given the results just seen for variance, I now repeat all the analyses I carried out previously on the data conditioned upon homer discovering a hit as significant in at least 4 individuals.</p>
<pre class="r"><code>#Repeating all QC with the condition of discovery in 4 individuals:
data.4 &lt;- filter(full.data, tot_indiv_IDd&gt;=4) #Enforcing this condition brings us down from ~3 million hits to ~350k hits.

###See how distributions look with condition of 4 individuals.
ggplot(data=melt(data.4[,c(304:305, 308:309)]), aes(x=value)) + geom_histogram(binwidth=0.0009, aes(group=variable)) + facet_wrap(~as.factor(variable)) + ggtitle(&quot;Distributions of Homer-Significant Hi-C Contacts, Humans | 4&quot;) + xlab(&quot;Homer-Normalized Interaction Frequency&quot;) + coord_cartesian(xlim=c(-6.6, 6.6), ylim=c(0, 450))#Human Dist&#39;ns</code></pre>
<pre><code>No id variables; using all as measure variables</code></pre>
<p><img src="figure/initial_QC.Rmd/QC%20%7C%204%20Individuals-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data=melt(data.4[,c(306:307, 310:311)]), aes(x=value)) + geom_histogram(binwidth=0.0009, aes(group=variable)) + facet_wrap(~as.factor(variable)) + ggtitle(&quot;Distributions of Homer-Significant Hi-C Contacts, Chimps | 4&quot;) + xlab(&quot;Homer-Normalized Interaction Frequency&quot;) + coord_cartesian(xlim=c(-6.6, 6.6), ylim=c(0, 450)) #Chimp Dist&#39;ns</code></pre>
<pre><code>No id variables; using all as measure variables</code></pre>
<p><img src="figure/initial_QC.Rmd/QC%20%7C%204%20Individuals-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Overall the distributions look fairly similar to before, with the bimodality seen before being much less stark here. Most of the negative interaction frequency significant hits have dropped out, and there is still a strong mass around 0 (especially in the chimp distributions). Tha majority of the mass of the hits now seems to hover around 1.8 or so.

###Mean vs. variance plot with condition of 4 individuals:
{plot(data.4$ALLmean, data.4$ALLvar, main=&quot;Mean vs. Variance in Homer-corrected contact frequency | 4&quot;, xlab = &quot;Mean Hi-C Contact Frequency&quot;, ylab=&quot;Variance in Hi-C Contact Frequency&quot;)
  abline(lm(data.4$ALLmean~data.4$ALLvar), col=&quot;red&quot;)}</code></pre>
<p><img src="figure/initial_QC.Rmd/QC%20%7C%204%20Individuals-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Looks very similar to the same plot above for the whole dataset; with slightly fewer outliers in the top left of the plot and a different trend in the red line modeling the mean regressed upon the variance. Here, the line is fairly flat, perhaps trending slightly upwards, whereas the entire dataset sees it have a moderate negative slope. Overall, would really prefer there to be no relationship here, but, again, this is not an ideal metric for Hi-C data, so it&#39;s not a particularly worrisome result.

###Now to do PCA on the data conditioned upon discovery in 4 individuals--would expect many of the same patterns here, perhaps with even stronger separation due to only grabbing hits that are shared by 4 individuals:
pca4 &lt;- prcomp(t(data.4[,304:311]), scale=TRUE, center=TRUE)
ggplot(data=as.data.frame(pca4$x), aes(x=PC1, y=PC2, shape=as.factor(meta.data$SP), color=as.factor(meta.data$Batch), size=2)) + geom_point() +labs(title=&quot;PCA on normalized Hi-C Contact Frequency | 4&quot;) + guides(color=guide_legend(order=1, title=&quot;Batch&quot;), size=FALSE, shape=guide_legend(order=2, title=&quot;Species&quot;)) + xlab(paste(&quot;PC1 (&quot;, 100*summary(pca4)$importance[2,1], &quot;% of variance)&quot;)) + ylab((paste(&quot;PC2 (&quot;, 100*summary(pca4)$importance[2,2], &quot;% of variance)&quot;))) + labs(color=&quot;Batch&quot;, shape=&quot;Species&quot;) + scale_shape_manual(labels=c(&quot;Chimp&quot;, &quot;Human&quot;), values=c(16, 17)) #Metadata hasn&#39;t changed so use the same metadata from above. PCA shows a similar separation along species lines for PC1, with this time it representing 24.171% of the variance (up from 20.49% earlier). This is rather unsurprising, since we have subset down to hits that are more shared than the total set we had before.</code></pre>
<p><img src="figure/initial_QC.Rmd/QC%20%7C%204%20Individuals-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(summary(pca4)$importance[2,], xlab=&quot;PCs&quot;, ylab=&quot;Proportion of Variance Explained&quot;, main=&quot;PCA on normalized Hi-C contact frequency | 4&quot;) #Scree plot showing all the PCs and the proportion of the variance they explain. This is similar to above, with no PCs beyond a 7th one, and the first PC taking the vast majority of the variance while the others hover near 12%. No variables in experimentation, library prep, sequencing batch, or anything else seem to correlate with these PCs.</code></pre>
<p><img src="figure/initial_QC.Rmd/QC%20%7C%204%20Individuals-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###Heatmap clustering of these data after conditioning upon discovery in 4 individuals. Would expect to see the same clustering, with perhaps higher correlation values due to conditioning upon discovery in 4 individuals.
corheat4 &lt;- cor(data.4[,304:311], use=&quot;complete.obs&quot;, method=&quot;pearson&quot;) #Corheat for the full data set, and heatmap
colnames(corheat4) &lt;- c(&quot;A_HF&quot;, &quot;B_HM&quot;, &quot;C_CM&quot;, &quot;D_CF&quot;, &quot;E_HM&quot;, &quot;F_HF&quot;, &quot;G_CM&quot;, &quot;H_CF&quot;)
rownames(corheat4) &lt;- colnames(corheat4)
heatmaply(corheat4, main=&quot;Pairwise Pearson Correlation Clustering @ 10kb | 4&quot;, k_row=2, k_col=2, symm=TRUE, margins=c(50, 50, 30, 30))</code></pre>
<pre><code>We recommend that you use the dev version of ggplot2 with `ggplotly()`
Install it with: `devtools::install_github(&#39;hadley/ggplot2&#39;)`</code></pre>
<pre><code>We recommend that you use the dev version of ggplot2 with `ggplotly()`
Install it with: `devtools::install_github(&#39;hadley/ggplot2&#39;)`
We recommend that you use the dev version of ggplot2 with `ggplotly()`
Install it with: `devtools::install_github(&#39;hadley/ggplot2&#39;)`</code></pre>
<div id="91261fbb2cbe" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="91261fbb2cbe">{"x":{"data":[{"x":[4.625,7.125,null,7.125,7.125,null,7.125,6.25,null,6.25,6.25,null,6.25,5.5,null,5.5,5.5,null,5.5,5,null,5,5,null,5.5,6,null,6,6,null,6.25,7,null,7,7,null,7.125,8,null,8,8],"y":[0.785701721253597,0.785701721253597,null,0.785701721253597,0.463357918963573,null,0.463357918963573,0.463357918963573,null,0.463357918963573,0.43605698231264,null,0.43605698231264,0.43605698231264,null,0.43605698231264,0.42863517329007,null,0.42863517329007,0.42863517329007,null,0.42863517329007,0,null,0.42863517329007,0.42863517329007,null,0.42863517329007,0,null,0.43605698231264,0.43605698231264,null,0.43605698231264,0,null,0.463357918963573,0.463357918963573,null,0.463357918963573,0],"text":["y: 0.7857017","y: 0.7857017",null,"y: 0.7857017","y: 0.7857017",null,"y: 0.4633579","y: 0.4633579",null,"y: 0.4633579","y: 0.4633579",null,"y: 0.4360570","y: 0.4360570",null,"y: 0.4360570","y: 0.4360570",null,"y: 0.4286352","y: 0.4286352",null,"y: 0.4286352","y: 0.4286352",null,"y: 0.4286352","y: 0.4286352",null,"y: 0.4286352","y: 0.4286352",null,"y: 0.4360570","y: 0.4360570",null,"y: 0.4360570","y: 0.4360570",null,"y: 0.4633579","y: 0.4633579",null,"y: 0.4633579","y: 0.4633579"],"type":"scatter","mode":"lines","line":{"width":2.26771653543307,"color":"rgba(57,190,177,1)","dash":"solid"},"hoveron":"points","name":"(#39BEB1,solid,1)","legendgroup":"(#39BEB1,solid,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4.625,2.125,null,2.125,2.125,null,2.125,1,null,1,1,null,2.125,3.25,null,3.25,3.25,null,3.25,2.5,null,2.5,2.5,null,2.5,2,null,2,2,null,2.5,3,null,3,3,null,3.25,4,null,4,4],"y":[0.785701721253597,0.785701721253597,null,0.785701721253597,0.432195917034341,null,0.432195917034341,0.432195917034341,null,0.432195917034341,0,null,0.432195917034341,0.432195917034341,null,0.432195917034341,0.41317667692587,null,0.41317667692587,0.41317667692587,null,0.41317667692587,0.388881441090047,null,0.388881441090047,0.388881441090047,null,0.388881441090047,0,null,0.388881441090047,0.388881441090047,null,0.388881441090047,0,null,0.41317667692587,0.41317667692587,null,0.41317667692587,0],"text":["y: 0.7857017","y: 0.7857017",null,"y: 0.7857017","y: 0.7857017",null,"y: 0.4321959","y: 0.4321959",null,"y: 0.4321959","y: 0.4321959",null,"y: 0.4321959","y: 0.4321959",null,"y: 0.4321959","y: 0.4321959",null,"y: 0.4131767","y: 0.4131767",null,"y: 0.4131767","y: 0.4131767",null,"y: 0.3888814","y: 0.3888814",null,"y: 0.3888814","y: 0.3888814",null,"y: 0.3888814","y: 0.3888814",null,"y: 0.3888814","y: 0.3888814",null,"y: 0.4131767","y: 0.4131767",null,"y: 0.4131767","y: 0.4131767"],"type":"scatter","mode":"lines","line":{"width":2.26771653543307,"color":"rgba(228,149,165,1)","dash":"solid"},"hoveron":"points","name":"(#E495A5,solid,1)","legendgroup":"(#E495A5,solid,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[4.625,2.125,1,3.25,2.5,2,3,4,7.125,6.25,5.5,5,6,7,8],"y":[0.785701721253597,0.432195917034341,0,0.41317667692587,0.388881441090047,0,0,0,0.463357918963573,0.43605698231264,0.42863517329007,0,0,0,0],"text":["y: 0.7857017","y: 0.4321959","y: 0.0000000","y: 0.4131767","y: 0.3888814","y: 0.0000000","y: 0.0000000","y: 0.0000000","y: 0.4633579","y: 0.4360570","y: 0.4286352","y: 0.0000000","y: 0.0000000","y: 0.0000000","y: 0.0000000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"transparent","opacity":1,"size":null,"symbol":null,"line":{"width":1.88976377952756,"color":"transparent"}},"hoveron":"points","name":"(NA,1)","legendgroup":"(NA,1)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"type":"scatter","mode":"markers","marker":{"fillcolor":"rgba(31,119,180,1)","color":"rgba(214,39,40,1)","line":{"color":"transparent"}},"xaxis":"x2","yaxis":"y","frame":null},{"x":[1,2,3,4,5,6,7,8],"y":[1,2,3,4,5,6,7,8],"z":[[1,0.379685516908789,0.391446336388182,0.361992873909541,0.0461251867498212,0.0155614272263319,0,0.00017012853667054],[0.379685516908789,1,0.425527364000444,0.389556408641003,0.0798689070508902,0.0292629274509691,0.0224690711017004,0.0179052496862342],[0.391446336388182,0.425527364000444,1,0.410996368446768,0.086321816358197,0.0379843472126727,0.0243148545167096,0.0247829525901117],[0.361992873909541,0.389556408641003,0.410996368446768,1,0.0710846787781944,0.033263880667018,0.0160220049006587,0.0208268898066415],[0.0461251867498212,0.0798689070508902,0.086321816358197,0.0710846787781944,1,0.369361180898482,0.362583753059596,0.345006834000355],[0.0155614272263319,0.0292629274509691,0.0379843472126727,0.033263880667018,0.369361180898482,1,0.355934297657991,0.335254462351322],[0,0.0224690711017004,0.0243148545167096,0.0160220049006587,0.362583753059596,0.355934297657991,1,0.31539523387547],[0.00017012853667054,0.0179052496862342,0.0247829525901117,0.0208268898066415,0.345006834000355,0.335254462351322,0.31539523387547,1]],"text":[["value: 1.0000000<br />column: H_CF<br />row: H_CF","value: 0.7032486<br />column: G_CM<br />row: H_CF","value: 0.7088748<br />column: C_CM<br />row: H_CF","value: 0.6947846<br />column: D_CF<br />row: H_CF","value: 0.5436771<br />column: E_HM<br />row: H_CF","value: 0.5290557<br />column: B_HM<br />row: H_CF","value: 0.5216113<br />column: F_HF<br />row: H_CF","value: 0.5216927<br />column: A_HF<br />row: H_CF"],["value: 0.7032486<br />column: H_CF<br />row: G_CM","value: 1.0000000<br />column: G_CM<br />row: G_CM","value: 0.7251788<br />column: C_CM<br />row: G_CM","value: 0.7079707<br />column: D_CF<br />row: G_CM","value: 0.5598197<br />column: E_HM<br />row: G_CM","value: 0.5356104<br />column: B_HM<br />row: G_CM","value: 0.5323603<br />column: F_HF<br />row: G_CM","value: 0.5301770<br />column: A_HF<br />row: G_CM"],["value: 0.7088748<br />column: H_CF<br />row: C_CM","value: 0.7251788<br />column: G_CM<br />row: C_CM","value: 1.0000000<br />column: C_CM<br />row: C_CM","value: 0.7182273<br />column: D_CF<br />row: C_CM","value: 0.5629067<br />column: E_HM<br />row: C_CM","value: 0.5397826<br />column: B_HM<br />row: C_CM","value: 0.5332433<br />column: F_HF<br />row: C_CM","value: 0.5334672<br />column: A_HF<br />row: C_CM"],["value: 0.6947846<br />column: H_CF<br />row: D_CF","value: 0.7079707<br />column: G_CM<br />row: D_CF","value: 0.7182273<br />column: C_CM<br />row: D_CF","value: 1.0000000<br />column: D_CF<br />row: D_CF","value: 0.5556174<br />column: E_HM<br />row: D_CF","value: 0.5375244<br />column: B_HM<br />row: D_CF","value: 0.5292761<br />column: F_HF<br />row: D_CF","value: 0.5315747<br />column: A_HF<br />row: D_CF"],["value: 0.5436771<br />column: H_CF<br />row: E_HM","value: 0.5598197<br />column: G_CM<br />row: E_HM","value: 0.5629067<br />column: C_CM<br />row: E_HM","value: 0.5556174<br />column: D_CF<br />row: E_HM","value: 1.0000000<br />column: E_HM<br />row: E_HM","value: 0.6983095<br />column: B_HM<br />row: E_HM","value: 0.6950673<br />column: F_HF<br />row: E_HM","value: 0.6866587<br />column: A_HF<br />row: E_HM"],["value: 0.5290557<br />column: H_CF<br />row: B_HM","value: 0.5356104<br />column: G_CM<br />row: B_HM","value: 0.5397826<br />column: C_CM<br />row: B_HM","value: 0.5375244<br />column: D_CF<br />row: B_HM","value: 0.6983095<br />column: E_HM<br />row: B_HM","value: 1.0000000<br />column: B_HM<br />row: B_HM","value: 0.6918863<br />column: F_HF<br />row: B_HM","value: 0.6819933<br />column: A_HF<br />row: B_HM"],["value: 0.5216113<br />column: H_CF<br />row: F_HF","value: 0.5323603<br />column: G_CM<br />row: F_HF","value: 0.5332433<br />column: C_CM<br />row: F_HF","value: 0.5292761<br />column: D_CF<br />row: F_HF","value: 0.6950673<br />column: E_HM<br />row: F_HF","value: 0.6918863<br />column: B_HM<br />row: F_HF","value: 1.0000000<br />column: F_HF<br />row: F_HF","value: 0.6724928<br />column: A_HF<br />row: F_HF"],["value: 0.5216927<br />column: H_CF<br />row: A_HF","value: 0.5301770<br />column: G_CM<br />row: A_HF","value: 0.5334672<br />column: C_CM<br />row: A_HF","value: 0.5315747<br />column: D_CF<br />row: A_HF","value: 0.6866587<br />column: E_HM<br />row: A_HF","value: 0.6819933<br />column: B_HM<br />row: A_HF","value: 0.6724928<br />column: F_HF<br />row: A_HF","value: 1.0000000<br />column: A_HF<br />row: A_HF"]],"colorscale":[[0,"#440154"],[0.00017012853667054,"#440154"],[0.0155614272263319,"#46075A"],[0.0160220049006587,"#46075A"],[0.0179052496862342,"#46085B"],[0.0208268898066415,"#46095C"],[0.0224690711017004,"#46095D"],[0.0243148545167096,"#460A5D"],[0.0247829525901117,"#460A5D"],[0.0292629274509691,"#460C5F"],[0.033263880667018,"#470D60"],[0.0379843472126727,"#470F62"],[0.0461251867498212,"#471365"],[0.0710846787781944,"#481B6D"],[0.0798689070508902,"#481E6F"],[0.086321816358197,"#482071"],[0.31539523387547,"#33638D"],[0.335254462351322,"#31688E"],[0.345006834000355,"#2F6B8E"],[0.355934297657991,"#2E6E8E"],[0.361992873909541,"#2E6F8E"],[0.362583753059596,"#2E6F8E"],[0.369361180898482,"#2D718E"],[0.379685516908789,"#2C738E"],[0.389556408641003,"#2B758E"],[0.391446336388182,"#2A768E"],[0.410996368446768,"#297B8E"],[0.425527364000444,"#277F8E"],[1,"#FDE725"]],"type":"heatmap","showscale":false,"autocolorscale":false,"showlegend":false,"xaxis":"x","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[1],"y":[1],"name":"99_1a45221797029f3db6e482f9f4dd9033","type":"scatter","mode":"markers","opacity":0,"hoverinfo":"none","showlegend":false,"marker":{"color":[0,1],"colorscale":[[0,"#471063"],[0.0434782608695653,"#481E6F"],[0.0869565217391305,"#472C7A"],[0.130434782608696,"#443983"],[0.173913043478261,"#404588"],[0.217391304347826,"#3B518B"],[0.260869565217391,"#365D8D"],[0.304347826086957,"#31678E"],[0.347826086956522,"#2C718E"],[0.391304347826087,"#287C8E"],[0.434782608695652,"#25858E"],[0.478260869565217,"#21908D"],[0.521739130434783,"#1F9A8A"],[0.565217391304348,"#20A386"],[0.608695652173913,"#27AD81"],[0.652173913043478,"#35B779"],[0.695652173913044,"#47C06F"],[0.739130434782609,"#5CC863"],[0.782608695652174,"#74D055"],[0.826086956521739,"#8FD644"],[0.869565217391304,"#AADC32"],[0.91304347826087,"#C7E020"],[0.956521739130435,"#E3E418"],[1,"#FDE725"]],"colorbar":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"thickness":23.04,"title":null,"titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"tickmode":"array","ticktext":["0.6","0.7","0.8","0.9","1.0"],"tickvals":[0.130434782608696,0.347826086956522,0.565217391304348,0.782608695652174,1],"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"ticklen":2,"len":0.5}},"xaxis":"x","yaxis":"y2","frame":null},{"x":[0.785701721253597,0.785701721253597,null,0.785701721253597,0.463357918963573,null,0.463357918963573,0.463357918963573,null,0.463357918963573,0.43605698231264,null,0.43605698231264,0.43605698231264,null,0.43605698231264,0.42863517329007,null,0.42863517329007,0.42863517329007,null,0.42863517329007,0,null,0.42863517329007,0.42863517329007,null,0.42863517329007,0,null,0.43605698231264,0.43605698231264,null,0.43605698231264,0,null,0.463357918963573,0.463357918963573,null,0.463357918963573,0],"y":[4.625,7.125,null,7.125,7.125,null,7.125,6.25,null,6.25,6.25,null,6.25,5.5,null,5.5,5.5,null,5.5,5,null,5,5,null,5.5,6,null,6,6,null,6.25,7,null,7,7,null,7.125,8,null,8,8],"text":["y: 0.7857017","y: 0.7857017",null,"y: 0.7857017","y: 0.7857017",null,"y: 0.4633579","y: 0.4633579",null,"y: 0.4633579","y: 0.4633579",null,"y: 0.4360570","y: 0.4360570",null,"y: 0.4360570","y: 0.4360570",null,"y: 0.4286352","y: 0.4286352",null,"y: 0.4286352","y: 0.4286352",null,"y: 0.4286352","y: 0.4286352",null,"y: 0.4286352","y: 0.4286352",null,"y: 0.4360570","y: 0.4360570",null,"y: 0.4360570","y: 0.4360570",null,"y: 0.4633579","y: 0.4633579",null,"y: 0.4633579","y: 0.4633579"],"type":"scatter","mode":"lines","line":{"width":2.26771653543307,"color":"rgba(57,190,177,1)","dash":"solid"},"hoveron":"points","name":"(#39BEB1,solid,1)","legendgroup":"(#39BEB1,solid,1)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[0.785701721253597,0.785701721253597,null,0.785701721253597,0.432195917034341,null,0.432195917034341,0.432195917034341,null,0.432195917034341,0,null,0.432195917034341,0.432195917034341,null,0.432195917034341,0.41317667692587,null,0.41317667692587,0.41317667692587,null,0.41317667692587,0.388881441090047,null,0.388881441090047,0.388881441090047,null,0.388881441090047,0,null,0.388881441090047,0.388881441090047,null,0.388881441090047,0,null,0.41317667692587,0.41317667692587,null,0.41317667692587,0],"y":[4.625,2.125,null,2.125,2.125,null,2.125,1,null,1,1,null,2.125,3.25,null,3.25,3.25,null,3.25,2.5,null,2.5,2.5,null,2.5,2,null,2,2,null,2.5,3,null,3,3,null,3.25,4,null,4,4],"text":["y: 0.7857017","y: 0.7857017",null,"y: 0.7857017","y: 0.7857017",null,"y: 0.4321959","y: 0.4321959",null,"y: 0.4321959","y: 0.4321959",null,"y: 0.4321959","y: 0.4321959",null,"y: 0.4321959","y: 0.4321959",null,"y: 0.4131767","y: 0.4131767",null,"y: 0.4131767","y: 0.4131767",null,"y: 0.3888814","y: 0.3888814",null,"y: 0.3888814","y: 0.3888814",null,"y: 0.3888814","y: 0.3888814",null,"y: 0.3888814","y: 0.3888814",null,"y: 0.4131767","y: 0.4131767",null,"y: 0.4131767","y: 0.4131767"],"type":"scatter","mode":"lines","line":{"width":2.26771653543307,"color":"rgba(228,149,165,1)","dash":"solid"},"hoveron":"points","name":"(#E495A5,solid,1)","legendgroup":"(#E495A5,solid,1)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[0.785701721253597,0.432195917034341,0,0.41317667692587,0.388881441090047,0,0,0,0.463357918963573,0.43605698231264,0.42863517329007,0,0,0,0],"y":[4.625,2.125,1,3.25,2.5,2,3,4,7.125,6.25,5.5,5,6,7,8],"text":["y: 0.7857017","y: 0.4321959","y: 0.0000000","y: 0.4131767","y: 0.3888814","y: 0.0000000","y: 0.0000000","y: 0.0000000","y: 0.4633579","y: 0.4360570","y: 0.4286352","y: 0.0000000","y: 0.0000000","y: 0.0000000","y: 0.0000000"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"transparent","opacity":1,"size":null,"symbol":null,"line":{"width":1.88976377952756,"color":"transparent"}},"hoveron":"points","name":"(NA,1)","legendgroup":"(NA,1)","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null}],"layout":{"xaxis":{"domain":[0,0.8],"type":"linear","autorange":false,"tickmode":"array","range":[0.5,8.5],"ticktext":["H_CF","G_CM","C_CM","D_CF","E_HM","B_HM","F_HF","A_HF"],"tickvals":[1,2,3,4,5,6,7,8],"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":13.2835201328352},"tickangle":-45,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y2","title":"","titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"hoverformat":".2f"},"xaxis2":{"domain":[0.8,1],"showticklabels":false,"showgrid":false,"zeroline":false,"anchor":"y2","type":"linear","autorange":false,"tickmode":"array","range":[0,0.785701721253597],"ticktext":["0.0","0.2","0.4","0.6"],"tickvals":[0,0.2,0.4,0.6],"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"gridcolor":null,"gridwidth":0,"title":"","titlefont":{"color":null,"family":null,"size":0},"hoverformat":".2f"},"yaxis2":{"domain":[0,0.8],"type":"linear","autorange":false,"tickmode":"array","range":[0.5,8.5],"ticktext":["H_CF","G_CM","C_CM","D_CF","E_HM","B_HM","F_HF","A_HF"],"tickvals":[1,2,3,4,5,6,7,8],"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":13.2835201328352},"tickangle":-0,"showline":true,"linecolor":"rgba(0,0,0,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"hoverformat":".2f"},"yaxis":{"domain":[0.8,1],"showticklabels":false,"showgrid":false,"zeroline":false,"anchor":"x","type":"linear","autorange":false,"tickmode":"array","range":[0,0.785701721253597],"ticktext":["0.0","0.2","0.4","0.6"],"tickvals":[0,0.2,0.4,0.6],"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"gridcolor":null,"gridwidth":0,"title":"","titlefont":{"color":null,"family":null,"size":0},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.8,"y0":0.8,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.8,"y0":0,"y1":0.8},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.8,"x1":1,"y0":0,"y1":0.8}],"margin":{"t":30,"r":30,"b":50,"l":50},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"y":1,"yanchor":"top"},"hovermode":"closest","paper_bgcolor":"rgba(255,255,255,1)","title":"Pairwise Pearson Correlation Clustering @ 10kb | 4"},"attrs":{"912648e26224":{"xend":{},"yend":{},"colour":{},"linetype":{},"size":{},"x":{},"y":{},"type":"ggplotly"},"91266994ede":{"colour":{},"shape":{},"size":{},"x":{},"y":{}},"912656d62592":{"alpha":1,"sizes":[10,100]},"91264524afec":{"fill":{},"x":{},"y":{},"type":"ggplotly"},"91269d658e2":{"xend":{},"yend":{},"colour":{},"linetype":{},"size":{},"x":{},"y":{},"type":"ggplotly"},"912657656279":{"colour":{},"shape":{},"size":{},"x":{},"y":{}}},"source":"A","config":{"modeBarButtonsToAdd":[],"cloud":false,"displaylogo":false,"modeBarButtonsToRemove":["sendDataToCloud","select2d","lasso2d","autoScale2d","hoverClosestCartesian","hoverCompareCartesian","sendDataToCloud"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1}},"subplot":true,"base_url":"https://plot.ly"},"evals":[],"jsHooks":{"render":[{"code":"function(el, x) { var ctConfig = crosstalk.var('plotlyCrosstalkOpts').set({\"on\":\"plotly_click\",\"persistent\":false,\"dynamic\":false,\"selectize\":false,\"opacityDim\":0.2,\"selected\":{\"opacity\":1}}); }","data":null}]}}</script>
<pre class="r"><code>#Indeed, we can again see that all the humans cluster together, as do all the chimps. This is very good, and given this new condition, we also see higher within-species correlations for the significant hits, in the 0.6-0.7 range (was much closer to 0.3 before).

###This initial QC suggest quality of the data is high, and has given us some information on how to filter the data before further intrepretation. When working locally, can be helpful to write out both the full dataset and the one conditioned upon 4 individuals, in order to do sanity checks on both later on. Note that a call of identical on something fread in after fwriting it out will still return FALSE with this as floating point can&#39;t represent decimals exactly (see https://stackoverflow.com/questions/9508518/why-are-these-numbers-not-equal or https://cran.r-project.org/doc/FAQ/R-FAQ.html#Why-doesn_0027t-R-think-these-numbers-are-equal_003f):
fwrite(full.data, &quot;~/Desktop/Hi-C/full.data.10.Nov17&quot;, quote = TRUE, sep = &quot;\t&quot;, row.names = FALSE, col.names = TRUE, na=&quot;NA&quot;, showProgress = FALSE)
fwrite(data.4, &quot;~/Desktop/Hi-C/data.4.10.Nov17&quot;, quote=FALSE, sep=&quot;\t&quot;, row.names=FALSE, col.names=TRUE, na=&quot;NA&quot;, showProgress = FALSE)</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.0 (2017-04-21)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: OS X El Capitan 10.11.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] compiler  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] bindrcpp_0.2        bedr_1.0.4          forcats_0.2.0      
 [4] purrr_0.2.4         readr_1.1.1         tibble_1.3.4       
 [7] tidyverse_1.2.1     edgeR_3.20.1        RColorBrewer_1.1-2 
[10] heatmaply_0.13.0    viridis_0.4.0       viridisLite_0.2.0  
[13] stringr_1.2.0       gplots_3.0.1        Hmisc_4.0-3        
[16] Formula_1.2-2       survival_2.41-3     lattice_0.20-35    
[19] dplyr_0.7.4         plotly_4.7.1        ggplot2_2.2.1      
[22] reshape2_1.4.2      data.table_1.10.4-3 tidyr_0.7.2        
[25] plyr_1.8.4          limma_3.34.3       

loaded via a namespace (and not attached):
  [1] colorspace_1.3-2     class_7.3-14         modeltools_0.2-21   
  [4] mclust_5.4           rprojroot_1.2        htmlTable_1.11.0    
  [7] futile.logger_1.4.3  base64enc_0.1-3      rstudioapi_0.7      
 [10] flexmix_2.3-14       mvtnorm_1.0-6        lubridate_1.7.1     
 [13] xml2_1.1.1           codetools_0.2-15     splines_3.4.0       
 [16] R.methodsS3_1.7.1    mnormt_1.5-5         robustbase_0.92-8   
 [19] knitr_1.17           jsonlite_1.5         broom_0.4.3         
 [22] cluster_2.0.6        kernlab_0.9-25       R.oo_1.21.0         
 [25] shiny_1.0.5          httr_1.3.1           backports_1.1.1     
 [28] assertthat_0.2.0     Matrix_1.2-12        lazyeval_0.2.1      
 [31] cli_1.0.0            acepack_1.4.1        htmltools_0.3.6     
 [34] tools_3.4.0          gtable_0.2.0         glue_1.2.0          
 [37] Rcpp_0.12.14         cellranger_1.1.0     trimcluster_0.1-2   
 [40] gdata_2.18.0         nlme_3.1-131         crosstalk_1.0.0     
 [43] iterators_1.0.8      fpc_2.1-10           psych_1.7.8         
 [46] testthat_1.0.2       rvest_0.3.2          mime_0.5            
 [49] gtools_3.5.0         dendextend_1.6.0     DEoptimR_1.0-8      
 [52] MASS_7.3-47          scales_0.5.0         TSP_1.1-5           
 [55] hms_0.4.0            parallel_3.4.0       lambda.r_1.2        
 [58] yaml_2.1.15          gridExtra_2.3        rpart_4.1-11        
 [61] latticeExtra_0.6-28  stringi_1.1.6        gclus_1.3.1         
 [64] foreach_1.4.3        checkmate_1.8.5      seriation_1.2-2     
 [67] caTools_1.17.1       rlang_0.1.4          pkgconfig_2.0.1     
 [70] prabclus_2.2-6       bitops_1.0-6         evaluate_0.10.1     
 [73] bindr_0.1            labeling_0.3         htmlwidgets_0.9     
 [76] magrittr_1.5         R6_2.2.2             haven_1.1.0         
 [79] whisker_0.3-2        foreign_0.8-69       nnet_7.3-12         
 [82] modelr_0.1.1         crayon_1.3.4         futile.options_1.0.0
 [85] KernSmooth_2.23-15   rmarkdown_1.8        locfit_1.5-9.1      
 [88] grid_3.4.0           readxl_1.0.0         git2r_0.19.0        
 [91] digest_0.6.12        diptest_0.75-7       webshot_0.5.0       
 [94] xtable_1.8-2         VennDiagram_1.6.18   httpuv_1.3.5        
 [97] R.utils_2.6.0        stats4_3.4.0         munsell_0.4.3       
[100] registry_0.5        </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
